# Performance of Bisulfite Sequencing Methods in Invertebrates

[![issues](https://img.shields.io/github/issues/hputnam/Meth_Compare.svg)](https://img.shields.io/github/issues/hputnam/Meth_Compare)
[![NSF-1921149](https://img.shields.io/badge/NSF-1921149-blue.svg)](https://nsf.gov/awardsearch/showAward?AWD_ID=1921149)
[![NSF-1921149](https://img.shields.io/badge/NSF-1921465-blue.svg)](https://nsf.gov/awardsearch/showAward?AWD_ID=1921465)
[![OSF](https://img.shields.io/badge/OSF-x5waz-blueviolet.svg)](https://osf.io/x5waz/)


## Repository Structure

- [metadata](https://github.com/hputnam/Meth_Compare/tree/master/metadata): Sample metadata
- [lab-methods](https://github.com/hputnam/Meth_Compare/tree/master/lab-methods): Commercial protocols, kit information, and lab notebook entries for DNA extraction, enrichment, and library preparation
- [genome-feature-tracks](https://github.com/hputnam/Meth_Compare/tree/master/genome-feature-files): Genome feature tracks generated for *M. capitata* and *P. acuta*
- [code](https://github.com/hputnam/Meth_Compare/tree/master/code): Bash scripts, R Markdown files, and Jupyter notebooks used to analyze data
- [analyses](https://github.com/hputnam/Meth_Compare/tree/master/analyses): Individual subdirectories for each analysis along with intermediate output
- [output](https://github.com/hputnam/Meth_Compare/tree/master/output): Finalized figures and tables from each analysis







## Abstract

The elucidation of potential genome to phenome pathways is essential for our understanding of biology, ecology, and evolution. Our understanding of phenotypic plasticity, a key feature of these pathways, has recently broadened to include epigenetic mechanisms, such as DNA methylation. DNA methylation and its role in gene expression regulation are increasingly investigated in marine invertebrates, as they provide mechanisms by which sessile organisms can respond to rapidly changing climate factors such as ocean acidification and heatwaves. However, studies using bisulfite sequencing have been limited due to cost. Further, data interpretation is dependent on the bisulfite sequencing method chosen and potential biases of methods have not been thoroughly compared for non-model marine invertebrates. Given this interest, we compare three common methods of quantifying DNA methylation with single base pair resolution: Whole Genome Bisulfite Sequencing (WGBS), Reduced Representation Bisulfite Sequencing (RRBS), and Methyl Binding Domain Bisulfite Sequencing (MBDBS), in the reef building corals *Montipora capitata* and *Pocillopora acuta*. Methylation level is consistently higher across all methods in *M. capitata* than *P. acuta* and the majority of CpGs (> 70%, > 82%, respectively) show low methylation in all methods. For common loci within a method, there is high correlation in methylation quantification for *M. capitata* (r > 0.83), but *P. acuta* displays lower and more variable correlations (r = 0.73-0.92). Correlations of common loci were generally lower between methods, with contrasting patterns between species. For *M. capitata*, WGBS v. MBDBS had the highest correlation (r = 0.82), whereas for *P. acuta*, WGBS v. RRBS had the highest correlation (r = 0.64). Common locus analysis identified discordant values of methylation (i.e., CpG with > 50% difference in methylation call) between methods. The greatest discordance in the methylation quantification of common loci between methods was found in *P. acuta* with 11% and 15% of common CpG sites being called at least 50% different in MBDBS compared to WGBS and RRBS, respectively, in contrast to only 0.4% and 5%, respectively in *M. capitata*. Highly methylated CpGs are more abundant in the MBDBS approach for both species, due to the enrichment of more highly methylated CpGs by MBD. In both species, using MBDBS generates the greatest bias for detecting CpGs in coding regions.  This is likely to be advantageous when the goal is to compare gene body methylation that is dominant in invertebrates. Functionally, gene ontology slim terms were equally represented in the genes with CpG data between methods, but the proportion of CpGs in each gene was biased differently between the species. Specifically, when comparing CpG coverage of one-to-one orthologs only between the two species, RRBS generated data for a greater proportion of possible CpGs compared to MBDBS for *M. capitata*, but for *P. acuta*, MBDBS covered a greater proportion of CpGs on average across orthologs than RRBS. While RRBS and MBDBS will have some limitations compared to whole methylome analysis, these reduced representation approaches provide cost effective methods to assess epigenetic mechanisms with high sample sizes. High throughput epigenetic analyses will be essential for improving our understanding of the role and importance of DNA methylation to phenotypic plasticity in critical marine invertebrate taxa.

## Results

- Tables
	- [Table 1](): Mapping Statistics. Values are Million reads (*code [here]()*).
- Figures
	- [Figure 1](): Comparison of CpG site coverage at different sequencing depths across library preparation methods. (A) Fraction of CpG sites in the genome covered at sequencing coverage (X) by M. capitata samples from each library preparation (see table 1 for number of reads). (B) Fraction of CpG sites in the genome covered at each sequencing depth (X) across different numbers of read pairs (M reads) for M. capitata samples from each library preparation. (C) Fraction of CpG sites in the genome covered at sequencing coverage (X) by P. acuta samples from each library preparation. (D) Fraction of CpG sites in the genome covered at each sequencing depth (X) across different numbers of read pairs (M reads) for P. acuta samples from each library preparation. For (A,B), Loess smoothing was applied and 95% confidence intervals are shown by shaded areas. For (F,G), results obtained from downsampling analyses are shown for sequencing depths ranging from 1–50X for library sizes of 50–200 M read pairs. All samples were pooled for the downsampling analyses (*code [here]()*).
	- [Figure 2](): The number of unique CpGs (vertical bars) for each intersection set (connected dots) for A) M. capitata and B) P. acuta. Color key shows intersection sets in venn diagram format with CpGs covered by all three methods (black), by any two methods (blue), by only one method (red), or CpGs not covered by any method (yellow). The CpG set size for the genome and for each method is shown in the horizontal bars (gray) (*code [here]()*).
	- [Figure 3](https://github.com/hputnam/Meth_Compare/blob/master/Output/Union-CpG-Type-Multipanel.pdf): Percent of highly methylated (≥ 50%; darkest shade), moderately methylated (10-50%; medium shade), and lowly methylated CpGs (< 10%; lightest shade) detected by each method A) for M. capitata and B) P. acuta (*code [here](https://github.com/hputnam/Meth_Compare/blob/master/code/Characterizing-CpG-Methylation-5x-Union-Summary-Plots.Rmd)*).
	- [Figure 4](https://github.com/hputnam/Meth_Compare/blob/master/Output/Union-CpG-Features-Multipanel.pdf): Percent of CpGs detected by sequencing methods in coding sequences (CDS), introns, 1 kb flanking regions upstream (Upstream Flank) or downstream of genes (Downstream Flank), and intergenic regions A) for M. capitata and B) P. acuta (*code [here](https://github.com/hputnam/Meth_Compare/blob/master/code/Characterizing-CpG-Methylation-5x-Union-Summary-Plots.Rmd)*).
	- [Figure 5](): Matrix of pairwise scatter plots for CpG loci covered at > 5x across all samples for A) M. capitata (n=4,666 common loci) and B) P. acuta (n=93,714 common loci). The red lines represent linear regression fits and the green lines are polynomial regression fits. Pearson correlation coefficients for each pairwise comparison are presented in the upper right boxes. Methods are color coded on the X and Y axes (WGBS = green, MBDBS = purple, and RRBS = orange) and replicate samples are indicated on the diagonal along with histograms of % CpG methylation (*code [here]()*).
	- [Figure 6](): Mean proportion (n=3 samples per method) of CpGs per gene that have at least 5x coverage in all of the one-to-one orthologous genes, as identified by OrthoFinder (Supplemental File X) for A) Montipora capitata and B) Pocillopora acuta. The heatmaps are ordered by orthogroup for the groups with one to one gene matches (*code [here]()*).
- [Supplementary Tables and Figures]()
