# Performance of Bisulfite Sequencing Methods in Invertebrates





[![issues](https://img.shields.io/github/issues/hputnam/Meth_Compare.svg)](https://img.shields.io/github/issues/hputnam/Meth_Compare)
[![NSF-1921149](https://img.shields.io/badge/NSF-1921149-blue.svg)](https://nsf.gov/awardsearch/showAward?AWD_ID=1921149) 
[![NSF-1921149](https://img.shields.io/badge/NSF-1921465-blue.svg)](https://nsf.gov/awardsearch/showAward?AWD_ID=1921465)
[![OSF](https://img.shields.io/badge/OSF-x5waz-blueviolet.svg)](https://osf.io/x5waz/)


##Abstract 
The elucidation of potential genome to phenome pathways is essential for our understanding of biology, ecology, and evolution. Our understanding of phenotypic plasticity, a key feature of these pathways, has recently broadened to include epigenetic mechanisms, such as DNA methylation. DNA methylation and its role in gene expression regulation are increasingly investigated in marine invertebrates, as they provide mechanisms by which sessile organisms can respond to rapidly changing climate factors such as ocean acidification and heatwaves. However, studies using bisulfite sequencing have been limited due to cost. Further, data interpretation is dependent on the bisulfite sequencing method chosen and potential biases of methods have not been thoroughly compared for non-model marine invertebrates. Given this interest, we compare three common methods of quantifying DNA methylation with single base pair resolution: Whole Genome Bisulfite Sequencing (WGBS), Reduced Representation Bisulfite Sequencing (RRBS), and Methyl Binding Domain Bisulfite Sequencing (MBDBS), in the reef building corals *Montipora capitata* and *Pocillopora acuta*. Methylation level is consistently higher across all methods in *M. capitata* than *P. acuta* and the majority of CpGs (> 70%, > 82%, respectively) show low methylation in all methods. For common loci within a method, there is high correlation in methylation quantification for *M. capitata* (r > 0.83), but *P. acuta* displays lower and more variable correlations (r = 0.73-0.92). Correlations of common loci were generally lower between methods, with contrasting patterns between species. For *M. capitata*, WGBS v. MBDBS had the highest correlation (r = 0.82), whereas for *P. acuta*, WGBS v. RRBS had the highest correlation (r = 0.64). Common locus analysis identified discordant values of methylation (i.e., CpG with > 50% difference in methylation call) between methods. The greatest discordance in the methylation quantification of common loci between methods was found in *P. acuta* with 11% and 15% of common CpG sites being called at least 50% different in MBDBS compared to WGBS and RRBS, respectively, in contrast to only 0.4% and 5%, respectively in *M. capitata*. Highly methylated CpGs are more abundant in the MBDBS approach for both species, due to the enrichment of more highly methylated CpGs by MBD. In both species, using MBDBS generates the greatest bias for detecting CpGs in coding regions.  This is likely to be advantageous when the goal is to compare gene body methylation that is dominant in invertebrates. Functionally, gene ontology slim terms were equally represented in the genes with CpG data between methods, but the proportion of CpGs in each gene was biased differently between the species. Specifically, when comparing CpG coverage of one-to-one orthologs only between the two species, RRBS generated data for a greater proportion of possible CpGs compared to MBDBS for *M. capitata*, but for *P. acuta*, MBDBS covered a greater proportion of CpGs on average across orthologs than RRBS. While RRBS and MBDBS will have some limitations compared to whole methylome analysis, these reduced representation approaches provide cost effective methods to assess epigenetic mechanisms with high sample sizes. High throughput epigenetic analyses will be essential for improving our understanding of the role and importance of DNA methylation to phenotypic plasticity in critical marine invertebrate taxa.