# Supplemental Material

**[Supplemental_Material.pdf](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/Supplemental_Material.pdf)**: PDF of all supplementary material

### Supplementary Figures
- [SF1.jpg](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/SF1.png): Supplementary Figure 1: Summary of sequencing depth and alignments for all libraries
- [SF2.png](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/SF2.png): Supplementary Figure 2: Bisulfite conversion efficiency
- [SF3.jpg](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/SF3.jpg): Percent of highly methylated (≥ 50%; darkest shade), moderately methylated (10-50%; medium shade), and lowly methylated CpGs (< 10%; lightest shade) detected by each method for (**A**) *M. capitata* and (**B**) *P. acuta*. Principal Coordinate Analyses associated with perMANOVA and beta-dispersion tests related to Supplementary Table 6 that show differences in proportion of CpGs that are highly (≥ 50%), moderately (10-50%), or lowly (≤ 10%) methylated in (**C**) *M. capitata* and (**D**) *P. acuta*. WGBS is represented by green circles, RRBS by purple triangles, and MBDBS by orange diamonds. Percent variation explained by each PCoA axis is included in the axis label. Ellipses depict 95% confidence intervals for each sequencing method. All eigenvectors are significant at the α = 0.05 level.
- [SF4.jpg](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/SF4.jpg): Supplementary Figure 4: PCA of CpG methylation for loci covered at 5x read depth in all samples for (**A**) *M. capitata* and (**B**) *P. acuta*.
- [SF5.png](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/SF5.png): Estimated fraction of CpG sites in the genome covered by at least 5 reads at different sequencing depths (number of M read pairs) for (**A**) *M. capitata* and (**B**) *P. acuta* samples for each bisulfite sequencing method. ‘Observed’ (opaque line and dots) denotes the fraction of genome-wide CpG loci covered by at least 5 reads determined from pooled data that was subsampled at 50M, 100M, 150M, and 200M reads. ‘Estimated’ (translucent line and dots) denotes the fraction of genome-wide CpG loci covered by at least 5 reads estimated by michaelis-menten modelling of the ‘observed’ data with standard error shown by shaded areas. All samples within a bisulfite sequencing method were pooled for the downsampling analyses.
- [SF6.png](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/SF6.png): Mean proportion (n=3 samples per method) of CpGs per gene that have at least 5x coverage in all of the one-to-one orthologous genes, as identified by OrthoFinder (Putnam et al., 2020) for (**A**) *M. capitata* and (**B**) *P. acuta*. 

### Supplementary Tables
- [ST1](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/ST1.csv): Supplementary Table 1: Trimming statistics
- [ST2](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/ST2.csv): Supplementary Table 2: Alignment statistics
- [ST3](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/ST3.csv): Supplementary Table 3: C1 alignment statistics
- [ST4](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/ST4.csv): Supplementary Table 4: Lambda conversion efficiency and alignment statistics
- [ST5](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/ST5.csv): Supplementary Table 5: Bisulfite conversion efficiency comparisons
- [ST6](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/ST6.csv): Supplementary Table 6: Results from PERMANOVA and beta-dispersion tests for genomic location
- [ST7](https://github.com/hputnam/Meth_Compare/blob/master/output/supplemental-material/ST7.csv): Supplementary Table 7: Contingency test results for _M. capitata_ and _P. acuta_
