---
title: "Circos Meth Compare"
author: "HM Putnam"
date: "7/19/2020"
output: html_document
---

Setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Load Libraries
```{r}
library(circlize) 
library(tidyverse)
library(ComplexHeatmap)
```

Generate Circos plots
#https://jokergoo.github.io/circlize_book/book/initialize-genomic-plot.html#customize-chromosome-track
#The input data for circos.genomicInitialize() is also a data frame with at least three columns. The first column is genomic category (for cytoband data, it is chromosome name), and the next two columns are positions in each genomic category.

Montipora capitata
```{r}
#read in the 5x coverage bedgraph of CpG for all 9 samples, 3 reps x 3 methods 
Mcap <- read.table("../data/Mcap_union_5x.bedgraph", sep = "\t", header = TRUE, na.strings = "N/A")
head(Mcap)
str(Mcap)

#load in all genes gff from a genome
Mcap.cytoband <- read.table("../genome-feature-files/Mcap.GFFannotation.genes.gff", sep = "\t",  
                            header = FALSE)

#remove extra coloumns from gene gff
Mcap.cytoband.df <- Mcap.cytoband[,c(1,4,5,6,9)]
str(Mcap.cytoband.df)
head(Mcap.cytoband.df)

n_distinct(Mcap.cytoband$V1)

Mcap.gene.cytoband <- Mcap.cytoband[,c(1,4,5,6,7)]
Mcap.gene.cytoband$V4 <-as.numeric(Mcap.gene.cytoband$V4)
Mcap.gene.cytoband$V5 <- as.numeric(Mcap.gene.cytoband$V5)
Mcap.gene.cytoband$V6 <- as.numeric(0.5)
Mcap.gene.cytoband$V7 <- as.numeric(1.0)

#examine scaffolds for the number of genes in a scaffold 
Mcap.sizes <- Mcap.cytoband %>% 
  group_by(V1) %>% 
  summarise(count=n()) %>%
  arrange(-count)

Mcap.gene.cytoband <- Mcap.gene.cytoband %>%
  filter(V1 == 2)

Mcap_union <- Mcap %>%
  filter(chrom== 2)
str(Mcap_union)

Mcap.gene.cytoband.head <- c(2,0,1,0,1)

Mcap.scaffold.lengths <- read.table("../data/Mcap.genome_assembly-sequence-lengths.txt", sep = "\t",  header = FALSE) %>%
  filter(V1 == 2)
Mcap.gene.cytoband.tail <- c(2,Mcap.scaffold.lengths[1,2]-1,Mcap.scaffold.lengths[1,2],0,1)

Mcap.gene.cytoband <- rbind(Mcap.gene.cytoband.head, Mcap.gene.cytoband)
Mcap.gene.cytoband <- rbind(Mcap.gene.cytoband,Mcap.gene.cytoband.tail)
str(Mcap.gene.cytoband)

#scale the data from 0-1
Mcap_union.WGBS <- Mcap_union %>% mutate(avg = rowMeans(.[4:6], na.rm=TRUE))
Mcap_union.WGBS <- Mcap_union.WGBS[,c(1,2,3,13)]
Mcap_union.WGBS$avg <- Mcap_union.WGBS$avg/100
Mcap_union.WGBS$avg[is.nan(Mcap_union.WGBS$avg)] <- NA
Mcap_union.RRBS <- Mcap_union %>% mutate(avg = rowMeans(.[7:9], na.rm=TRUE))
Mcap_union.RRBS <- Mcap_union.RRBS[,c(1,2,3,13)]
Mcap_union.RRBS$avg <- Mcap_union.RRBS$avg/100
Mcap_union.RRBS$avg[is.nan(Mcap_union.RRBS$avg)] <- NA
Mcap_union.MBDBS <- Mcap_union %>% mutate(avg = rowMeans(.[10:12], na.rm=TRUE))
Mcap_union.MBDBS <- Mcap_union.MBDBS[,c(1,2,3,13)]
Mcap_union.MBDBS$avg <- Mcap_union.MBDBS$avg/100
Mcap_union.MBDBS$avg[is.nan(Mcap_union.MBDBS$avg)] <- NA

```

Pocillopora acuta
```{r}
#read in the 5x coverage bedgraph of CpG for all 9 samples, 3 reps x 3 methods 
Pact <- read.table("../data/Pact_union_5x.bedgraph", sep = "\t", header = TRUE, na.strings = "N/A")
head(Pact)
str(Pact)

#load in all genes gff from a genome
Pact.cytoband <- read.table("../genome-feature-files/Pact.GFFannotation.genes.gff", sep = "\t",  
                            header = FALSE)

#remove extra coloumns from gene gff
Pact.cytoband.df <- Pact.cytoband[,c(1,4,5,6,9)]
str(Pact.cytoband.df)
head(Pact.cytoband.df)

n_distinct(Pact.cytoband$V1)

Pact.gene.cytoband <- Pact.cytoband[,c(1,4,5,6,7)]
Pact.gene.cytoband$V4 <- as.numeric(Pact.gene.cytoband$V4)
Pact.gene.cytoband$V5 <- as.numeric(Pact.gene.cytoband$V5)
Pact.gene.cytoband$V6 <- as.numeric(0.5)
Pact.gene.cytoband$V7 <- as.numeric(1.0)
str(Pact.gene.cytoband)

#examine scaffolds for the number of genes in a scaffold 
Pact.sizes <- Pact.cytoband %>% 
  group_by(V1) %>% 
  summarise(count=n()) %>%
  arrange(-count)

Pact.gene.cytoband <- Pact.gene.cytoband %>%
  filter(V1 == "scaffold502_cov107" )
str(Pact.gene.cytoband)

Pact.gene.cytoband$V1

Pact_union <- Pact %>%
  filter(chrom== "scaffold502_cov107")
str(Pact_union)

Pact_union$chrom <- 1

Pact.gene.cytoband.head <- c(1,0,1,0,1)

Pact.scaffold.lengths <- read.table("../data/Pact.genome_assembly-sequence-lengths.txt", sep = "\t",  stringsAsFactors=FALSE, header = FALSE) %>%
  filter(V1 == "scaffold502_cov107")
Pact.gene.cytoband.tail <- c(1,Pact.scaffold.lengths[1,2]-1,Pact.scaffold.lengths[1,2],0,1)

str(Pact.gene.cytoband)
Pact.gene.cytoband$V1 <- 1

Pact.gene.cytoband <- rbind(Pact.gene.cytoband.head, Pact.gene.cytoband)
str(Pact.gene.cytoband)
Pact.gene.cytoband <- rbind(Pact.gene.cytoband,Pact.gene.cytoband.tail)
str(Pact.gene.cytoband)

#scale the data from 0-1
Pact_union.WGBS <- Pact_union %>% mutate(avg = rowMeans(.[4:6], na.rm=TRUE))
Pact_union.WGBS <- Pact_union.WGBS[,c(1,2,3,13)]
Pact_union.WGBS$avg <- Pact_union.WGBS$avg/100
Pact_union.WGBS$avg[is.nan(Pact_union.WGBS$avg)] <- NA
Pact_union.RRBS <- Pact_union %>% mutate(avg = rowMeans(.[7:9], na.rm=TRUE))
Pact_union.RRBS <- Pact_union.RRBS[,c(1,2,3,13)]
Pact_union.RRBS$avg <- Pact_union.RRBS$avg/100
Pact_union.RRBS$avg[is.nan(Pact_union.RRBS$avg)] <- NA
Pact_union.MBDBS <- Pact_union %>% mutate(avg = rowMeans(.[10:12], na.rm=TRUE))
Pact_union.MBDBS <- Pact_union.MBDBS[,c(1,2,3,13)]
Pact_union.MBDBS$avg <- Pact_union.MBDBS$avg/100
Pact_union.MBDBS$avg[is.nan(Pact_union.MBDBS$avg)] <- NA

```


```{r}
#plot top scaffolds with the greatest number of gene

pdf('../Output/Circos_plots/Circos_topscaff.pdf', width=6, height=10)
par(mfrow=c(2,1))
circos.clear()

#Montipora
set.seed(123)
circos.par("track.height" = 0.2, start.degree = 90, gap.degree = 20, cell.padding = c(0.01,1,0.01,1))

circos.initializeWithIdeogram(Mcap.gene.cytoband, species = NULL, sort.chr = TRUE)
circos.genomicTrack(Mcap_union.WGBS, stack=FALSE, ylim=c(0,1), track.height = 0.1,
                    panel.fun = function(region, value, ...) {
                      circos.genomicPoints(region, value, pch = 16, cex = 0.2, col="#31A354")
                    })
circos.yaxis(side="left", labels = TRUE, tick=TRUE, at=c(0,0.5,1))
circos.genomicTrack(Mcap_union.RRBS, stack=FALSE, ylim=c(0,1), track.height = 0.1,
                    panel.fun = function(region, value, ...) {
                      circos.genomicPoints(region, value, pch = 16, cex = 0.2, col="#756BB1")
                    })
circos.yaxis(side="left", labels = TRUE, tick=TRUE, at=c(0,0.5,1))
circos.genomicTrack(Mcap_union.MBDBS, stack=FALSE, ylim=c(0,1), track.height = 0.1,
                    panel.fun = function(region, value, ...) {
                      circos.genomicPoints(region, value, pch = 16, cex = 0.2, col="#E6550D")
                    })
circos.yaxis(side="left", labels = TRUE, tick=TRUE, at=c(0,0.5,1))
lgd = Legend(at = c("WGBS", "RRBS", "MBDBS"), legend_gp=gpar(fill=c("#31A354","#756BB1","#E6550D")))

draw(lgd,  just = c("bottom"))


#Pocillopora
circos.clear()

set.seed(123)
circos.par("track.height" = 0.2, start.degree = 90, gap.degree = 20, cell.padding = c(0.01,1,0.01,1))

circos.initializeWithIdeogram(Pact.gene.cytoband, species = NULL, sort.chr = TRUE)
circos.genomicTrack(Pact_union.WGBS, stack=FALSE, ylim=c(0,1), track.height = 0.1,
                    panel.fun = function(region, value, ...) {
                      circos.genomicPoints(region, value, pch = 16, cex = 0.2, col="#31A354")
                    })
circos.yaxis(side="left", labels = TRUE, tick=TRUE, at=c(0,0.5,1))
circos.genomicTrack(Pact_union.RRBS, stack=FALSE, ylim=c(0,1), track.height = 0.1,
                    panel.fun = function(region, value, ...) {
                      circos.genomicPoints(region, value, pch = 16, cex = 0.2, col="#756BB1")
                    })
circos.yaxis(side="left", labels = TRUE, tick=TRUE, at=c(0,0.5,1))
circos.genomicTrack(Pact_union.MBDBS, stack=FALSE, ylim=c(0,1), track.height = 0.1,
                    panel.fun = function(region, value, ...) {
                      circos.genomicPoints(region, value, pch = 16, cex = 0.2, col="#E6550D")
                    })
circos.yaxis(side="left", labels = TRUE, tick=TRUE, at=c(0,0.5,1))
lgd = Legend(at = c("WGBS", "RRBS", "MBDBS"), legend_gp=gpar(fill=c("#31A354","#756BB1","#E6550D")))

draw(lgd,  just = c("bottom"))



dev.off()

```

