# Library Prep Methods


## RRBS Library Prep
https://meschedl.github.io/MESPutnam_Open_Lab_Notebook/RRBS-Meth-Comp/

## MBDBS Enrichment
https://meschedl.github.io/MESPutnam_Open_Lab_Notebook/MBD-Meth-Comp/

## WGBS and MBDBS Library Prep
https://meschedl.github.io/MESPutnam_Open_Lab_Notebook/WGBS-MC-1/
https://meschedl.github.io/MESPutnam_Open_Lab_Notebook/meth-comp-PMS/
https://meschedl.github.io/MESPutnam_Open_Lab_Notebook/Redos-PMS/


---

# Sequence Data


values provided by GeneWiz

```
63a1cc9e23dcedd71a66dac00f36467d  Meth10_R1_001.fastq.gz
ac966619cb5a2483870f973b696a38a1  Meth10_R2_001.fastq.gz
06278c40d97f7a3efdae47218921ac6a  Meth11_R1_001.fastq.gz
07e71cb470c1622ced336478df70204a  Meth11_R2_001.fastq.gz
692992366c6709c1c50419bde55397cb  Meth12_R1_001.fastq.gz
27803c9f39f0456ebc56aaa07620b7c5  Meth12_R2_001.fastq.gz
9988a9eefd526231f6608a4549d2cfa7  Meth13_R1_001.fastq.gz
04db1696a84174d2a146fdd2e2b8a1ca  Meth13_R2_001.fastq.gz
131cb0c436de334bf17c375d49e02977  Meth14_R1_001.fastq.gz
fc7f543911fffce636899e05ec29dee3  Meth14_R2_001.fastq.gz
448f06b9bf1e19ee535ab74ce0f744fa  Meth15_R1_001.fastq.gz
6bd07498df6b72f060b8e2c7ce624e01  Meth15_R2_001.fastq.gz
46a49f6745c7478346f25ed7f509a704  Meth16_R1_001.fastq.gz
a6a1dd2355ce0e6e625c24f23373dc86  Meth16_R2_001.fastq.gz
f0a894a39a8a46d8729cd953c2531035  Meth17_R1_001.fastq.gz
bd0c3257b16d91e181b688d64d7d0b54  Meth17_R2_001.fastq.gz
d1b6b17e76c2634af2d9918f9ef7d9e9  Meth18_R1_001.fastq.gz
e9ad11546163f7e22ed7aa427fa7d0c1  Meth18_R2_001.fastq.gz
2560348651ae8a94fb351d0e54eff54e  Meth1_R1_001.fastq.gz
45faf1e9cbf5a664e5c6818817e85ef1  Meth1_R2_001.fastq.gz
a4963c92a3b6f62f2480f4f4bd72ec8f  Meth2_R1_001.fastq.gz
beac9fbd4b2e460bc7f8375e3e6caaa8  Meth2_R2_001.fastq.gz
1ae956f0fb4023165aaa4c58df9d437c  Meth3_R1_001.fastq.gz
6c6a24173eb24f2db81290b21edf17c0  Meth3_R2_001.fastq.gz
2ad73235560c58d8e68c3f75ada90840  Meth4_R1_001.fastq.gz
ad106065fbdbcf9030cd804296290e18  Meth4_R2_001.fastq.gz
8b966a3413ad40a087dcbf68204e84d6  Meth5_R1_001.fastq.gz
779e2fee610f06fe006c2ef8a4535114  Meth5_R2_001.fastq.gz
e481ea3fd24c09dafdfad82dec5d2d0a  Meth6_R1_001.fastq.gz
3d876fd2cd176450c8a63bf739ac5857  Meth6_R2_001.fastq.gz
d5199e48ed87773f67c4e3d8d9db3fce  Meth7_R1_001.fastq.gz
5d8467eb30d47bb00d8133fa4a743acf  Meth7_R2_001.fastq.gz
9b59ce33ab17158f2fea64f2d0d0ade7  Meth8_R1_001.fastq.gz
37c495ced527f3e30ae2f3388a9146bf  Meth8_R2_001.fastq.gz
c8ec0b73721a067729124c043904dc0e  Meth9_R1_001.fastq.gz
b896f80209fbaaee4ca5316f435839ff  Meth9_R2_001.fastq.gz
```


Raw data on Mox - from where trimming was performed.(why does this say where trimming was performed @sr320?)

```
63a1cc9e23dcedd71a66dac00f36467d  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth10_R1_001.fastq.gz
ac966619cb5a2483870f973b696a38a1  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth10_R2_001.fastq.gz
06278c40d97f7a3efdae47218921ac6a  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth11_R1_001.fastq.gz
07e71cb470c1622ced336478df70204a  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth11_R2_001.fastq.gz
692992366c6709c1c50419bde55397cb  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth12_R1_001.fastq.gz
27803c9f39f0456ebc56aaa07620b7c5  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth12_R2_001.fastq.gz
9988a9eefd526231f6608a4549d2cfa7  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth13_R1_001.fastq.gz
04db1696a84174d2a146fdd2e2b8a1ca  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth13_R2_001.fastq.gz
131cb0c436de334bf17c375d49e02977  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth14_R1_001.fastq.gz
fc7f543911fffce636899e05ec29dee3  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth14_R2_001.fastq.gz
448f06b9bf1e19ee535ab74ce0f744fa  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth15_R1_001.fastq.gz
6bd07498df6b72f060b8e2c7ce624e01  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth15_R2_001.fastq.gz
46a49f6745c7478346f25ed7f509a704  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth16_R1_001.fastq.gz
a6a1dd2355ce0e6e625c24f23373dc86  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth16_R2_001.fastq.gz
f0a894a39a8a46d8729cd953c2531035  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth17_R1_001.fastq.gz
bd0c3257b16d91e181b688d64d7d0b54  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth17_R2_001.fastq.gz
d1b6b17e76c2634af2d9918f9ef7d9e9  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth18_R1_001.fastq.gz
e9ad11546163f7e22ed7aa427fa7d0c1  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth18_R2_001.fastq.gz
2560348651ae8a94fb351d0e54eff54e  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth1_R1_001.fastq.gz
45faf1e9cbf5a664e5c6818817e85ef1  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth1_R2_001.fastq.gz
a4963c92a3b6f62f2480f4f4bd72ec8f  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth2_R1_001.fastq.gz
beac9fbd4b2e460bc7f8375e3e6caaa8  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth2_R2_001.fastq.gz
1ae956f0fb4023165aaa4c58df9d437c  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth3_R1_001.fastq.gz
6c6a24173eb24f2db81290b21edf17c0  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth3_R2_001.fastq.gz
2ad73235560c58d8e68c3f75ada90840  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth4_R1_001.fastq.gz
ad106065fbdbcf9030cd804296290e18  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth4_R2_001.fastq.gz
8b966a3413ad40a087dcbf68204e84d6  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth5_R1_001.fastq.gz
779e2fee610f06fe006c2ef8a4535114  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth5_R2_001.fastq.gz
e481ea3fd24c09dafdfad82dec5d2d0a  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth6_R1_001.fastq.gz
3d876fd2cd176450c8a63bf739ac5857  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth6_R2_001.fastq.gz
d5199e48ed87773f67c4e3d8d9db3fce  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth7_R1_001.fastq.gz
5d8467eb30d47bb00d8133fa4a743acf  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth7_R2_001.fastq.gz
9b59ce33ab17158f2fea64f2d0d0ade7  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth8_R1_001.fastq.gz
37c495ced527f3e30ae2f3388a9146bf  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth8_R2_001.fastq.gz
c8ec0b73721a067729124c043904dc0e  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth9_R1_001.fastq.gz
b896f80209fbaaee4ca5316f435839ff  /gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth9_R2_001.fastq.gz






```

# Quality Trimming

```
#!/bin/bash
## Job Name
#SBATCH --job-name=TrimGfroger
## Allocation Definition
#SBATCH --account=srlab
#SBATCH --partition=srlab
## Resources
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=4-23:30:00
## Memory per node
#SBATCH --mem=100G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=strigg@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/strigg/analyses/20200311


#TrimGalore on WGBS and MBD-BS
/gscratch/srlab/programs/TrimGalore-0.4.5/trim_galore \
--output_dir /gscratch/scrubbed/strigg/analyses/20200311/WGBS_MBD \
--paired \
--fastqc_args \
"--outdir /gscratch/scrubbed/strigg/analyses/20200311/WGBS_MBD/FASTQC \
--threads 28" \
--illumina \
--clip_R1 10 \
--clip_R2 10 \
--three_prime_clip_R1 10 \
--three_prime_clip_R2 10 \
--path_to_cutadapt /gscratch/srlab/programs/miniconda3/bin/cutadapt \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth10_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth10_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth11_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth11_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth12_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth12_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth1_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth1_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth2_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth2_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth3_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth3_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth7_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth7_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth8_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth8_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth9_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth9_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth16_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth16_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth17_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth17_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth18_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth18_R2_001.fastq.gz

#TrimGalore on RRBS
/gscratch/srlab/programs/TrimGalore-0.4.5/trim_galore \
--output_dir /gscratch/scrubbed/strigg/analyses/20200311/RRBS \
--fastqc_args \
"--outdir /gscratch/scrubbed/strigg/analyses/20200311/RRBS/FASTQC \
--threads 28" \
--non_directional \
--rrbs \
--paired \
--path_to_cutadapt /gscratch/srlab/programs/miniconda3/bin/cutadapt \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth13_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth13_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth14_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth14_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth15_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth15_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth4_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth4_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth5_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth5_R2_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth6_R1_001.fastq.gz \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth6_R2_001.fastq.gz

#run multiqc for WGBS and MBD samples

/gscratch/srlab/strigg/bin/anaconda3/bin/multiqc \
/gscratch/scrubbed/strigg/analyses/20200311/WGBS_MBD/FASTQC/.

#run multiqc for RRBS samples

/gscratch/srlab/strigg/bin/anaconda3/bin/multiqc \
/gscratch/scrubbed/strigg/analyses/20200311/RRBS/FASTQC/.
```
---



**Trimmed data on Mox - from where trimming was performed**
`/gscratch/scrubbed/strigg/analyses/20200311/RRBS`


```
#RRBS:
2e87e45941d8a4c523e6080fc460d3f3  Meth13_R1_001_val_1.fq.gz
e9548420a9512df45a08cf3b735cc95f  Meth13_R2_001_val_2.fq.gz
4ed014c23ba4c28681d5b4af17e95346  Meth14_R1_001_val_1.fq.gz
8937fd8eb2ef852e103a79b48231a9ab  Meth14_R2_001_val_2.fq.gz
49ece06a7e8a4a8a936fddcdbd9c3921  Meth15_R1_001_val_1.fq.gz
e1048fea898bc32cb03ff801534183d9  Meth15_R2_001_val_2.fq.gz
7a40c8cfafabb02e78bb8efe9001bf53  Meth4_R1_001_val_1.fq.gz
900a3858aab3571a8329c2e5952b265e  Meth4_R2_001_val_2.fq.gz
8c5ac01b3aeee1b4a40e991885259007  Meth5_R1_001_val_1.fq.gz
d6e026bb59b10a11ad9b51b8acdd18a7  Meth5_R2_001_val_2.fq.gz
2bafaff063eb4b8c960e345634291fb4  Meth6_R1_001_val_1.fq.gz
6ea2c605b839ea9ca91ecb524e7184d7  Meth6_R2_001_val_2.fq.gz
```

`/gscratch/scrubbed/strigg/analyses/20200311/WGBS_MBD`

```
#WGBS_MBD:
04e14c4ec8cd7ef280b97d3ecca51b1a  Meth10_R1_001_val_1.fq.gz
7eed0ec3df5d8fd42984f517e5633ef6  Meth10_R2_001_val_2.fq.gz
84f59f4b67562be14bd0692c89ee3066  Meth11_R1_001_val_1.fq.gz
91e1151899fa6de016d4bce233be0819  Meth11_R2_001_val_2.fq.gz
645233a25517dcb1e620e794aaf1e4bc  Meth12_R1_001_val_1.fq.gz
f5f0339ac5f056d8dd3cf6747cfcbff6  Meth12_R2_001_val_2.fq.gz
c4e449e7210609e56be0348d1cacaf85  Meth16_R1_001_val_1.fq.gz
047d811b05a0df7a93344c110ce5f155  Meth16_R2_001_val_2.fq.gz
36aba1f7007b9fb4c83a76810e4b1306  Meth17_R1_001_val_1.fq.gz
a006f7dbf9baa7239272506b1ea560be  Meth17_R2_001_val_2.fq.gz
cdde1f3bc4e4020fb4025ea1dc7a0c2e  Meth18_R1_001_val_1.fq.gz
06a030b4c6ae092d748a08a4c226e674  Meth18_R2_001_val_2.fq.gz
f41790ce58777f20ee742cba75692065  Meth1_R1_001_val_1.fq.gz
ab16c162efedbcb6dce8e678affa4ee1  Meth1_R2_001_val_2.fq.gz
e78a80a91c6d0e0928299d22a7aeba33  Meth2_R1_001_val_1.fq.gz
7e94bfa7f8c6c0794e4958e5628fe45f  Meth2_R2_001_val_2.fq.gz
27895574a941878cd26a1400b04ee3cd  Meth3_R1_001_val_1.fq.gz
1926fcfc1e4172e26b83cad82d87fb67  Meth3_R2_001_val_2.fq.gz
1c1bb5648e064bd9f927e4f481a7d8e5  Meth7_R1_001_val_1.fq.gz
0edb6d01e4f6561d3eb24c4b2ff62035  Meth7_R2_001_val_2.fq.gz
efc6cc59733842eabadd21536c2f33bb  Meth8_R1_001_val_1.fq.gz
8c5a88ab86e9673b4e1c82f1452b6ad1  Meth8_R2_001_val_2.fq.gz
04829778554df5986ae415fcda3b7e81  Meth9_R1_001_val_1.fq.gz
a4c8dd40dece0e44397abebd447d7ec1  Meth9_R2_001_val_2.fq.gz
```

---
# Genome Preparation

**File info**     
```
859M Jun  8  2018 Mcap.genome_assembly.fa
d51b958bd7adc4ba83a959f7c0a1774c  Mcap.genome_assembly.fa
```

```
344M Mar  5 15:46 Pocillopora_acuta_genome_v1.fasta
4664217f0afd33b80614b09368be2c00  Pocillopora_acuta_genome_v1.fasta
```





---

```
#!/bin/bash
## Job Name
#SBATCH --job-name=fr-01
## Allocation Definition
#SBATCH --account=srlab
#SBATCH --partition=srlab
## Resources
## Nodes (We only get 1, so this is fixed)
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=6-00:00:00
## Memory per node
#SBATCH --mem=100G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=sr320@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/sr320/030520-fr01/


# Directories and programs
bismark_dir="/gscratch/srlab/programs/Bismark-0.21.0"
bowtie2_dir="/gscratch/srlab/programs/bowtie2-2.3.4.1-linux-x86_64/"
samtools="/gscratch/srlab/programs/samtools-1.9/samtools"

source /gscratch/srlab/programs/scripts/paths.sh


${bismark_dir}/bismark_genome_preparation \
--verbose \
--parallel 28 \
--path_to_aligner ${bowtie2_dir} \
/gscratch/srlab/sr320/data/froger/Mcap_Genome/


${bismark_dir}/bismark_genome_preparation \
--verbose \
--parallel 28 \
--path_to_aligner ${bowtie2_dir} \
/gscratch/srlab/sr320/data/froger/Pact_Genome/
```



# Bismark Alignment   




**Trim files used**

```
/gscratch/scrubbed/sr320/031520-TG-bs/Mcap_trim/
/gscratch/scrubbed/sr320/031520-TG-bs/Pact_trim/
```

```
[sr320@mox2 Mcap_trim]$ ls
Meth10_R1_001_val_1.fq.gz  Meth13_R1_001_val_1.fq.gz  Meth16_R1_001_val_1.fq.gz
Meth10_R2_001_val_2.fq.gz  Meth13_R2_001_val_2.fq.gz  Meth16_R2_001_val_2.fq.gz
Meth11_R1_001_val_1.fq.gz  Meth14_R1_001_val_1.fq.gz  Meth17_R1_001_val_1.fq.gz
Meth11_R2_001_val_2.fq.gz  Meth14_R2_001_val_2.fq.gz  Meth17_R2_001_val_2.fq.gz
Meth12_R1_001_val_1.fq.gz  Meth15_R1_001_val_1.fq.gz  Meth18_R1_001_val_1.fq.gz
Meth12_R2_001_val_2.fq.gz  Meth15_R2_001_val_2.fq.gz  Meth18_R2_001_val_2.fq.gz

```

```
[sr320@mox2 Pact_trim]$ ls
Meth1_R1_001_val_1.fq.gz  Meth4_R1_001_val_1.fq.gz  Meth7_R1_001_val_1.fq.gz
Meth1_R2_001_val_2.fq.gz  Meth4_R2_001_val_2.fq.gz  Meth7_R2_001_val_2.fq.gz
Meth2_R1_001_val_1.fq.gz  Meth5_R1_001_val_1.fq.gz  Meth8_R1_001_val_1.fq.gz
Meth2_R2_001_val_2.fq.gz  Meth5_R2_001_val_2.fq.gz  Meth8_R2_001_val_2.fq.gz
Meth3_R1_001_val_1.fq.gz  Meth6_R1_001_val_1.fq.gz  Meth9_R1_001_val_1.fq.gz
Meth3_R2_001_val_2.fq.gz  Meth6_R2_001_val_2.fq.gz  Meth9_R2_001_val_2.fq.gz
```










---

```
[sr320@mox2 031520-TG-bs]$ cat 0313_1100.sh
#!/bin/bash
## Job Name
#SBATCH --job-name=tg-bismark
##  This script is meant to align trim galore data to both genomes
##  And generate files for downstream analyses
## Allocation Definition
#SBATCH --account=srlab
#SBATCH --partition=srlab
## Resources
## Nodes (We only get 1, so this is fixed)
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=10-00:00:00
## Memory per node
#SBATCH --mem=500G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=sr320@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/sr320/031520-TG-bs


# Directories and programs
bismark_dir="/gscratch/srlab/programs/Bismark-0.21.0"
bowtie2_dir="/gscratch/srlab/programs/bowtie2-2.3.4.1-linux-x86_64/"
samtools="/gscratch/srlab/programs/samtools-1.9/samtools"

source /gscratch/srlab/programs/scripts/paths.sh




# Aligning to genomes. Will need trim reads in distinct directories
# Need to confirm filenames jive with code.
# AND set output directory

genome_folder="/gscratch/srlab/sr320/data/froger/Mcap_Genome/"
reads_dir="/gscratch/scrubbed/sr320/031520-TG-bs/Mcap_trim/"


find ${reads_dir}*_R1_001_val_1.fq.gz \
| xargs basename -s _R1_001_val_1.fq.gz | xargs -I{} ${bismark_dir}/bismark \
--path_to_bowtie ${bowtie2_dir} \
-genome ${genome_folder} \
-p 4 \
-score_min L,0,-0.6 \
--non_directional \
-1 ${reads_dir}{}_R1_001_val_1.fq.gz \
-2 ${reads_dir}{}_R2_001_val_2.fq.gz \
-o Mcap_tg



genome_folder="/gscratch/srlab/sr320/data/froger/Pact_Genome/"
reads_dir="/gscratch/scrubbed/sr320/031520-TG-bs/Pact_trim/"


find ${reads_dir}*_R1_001_val_1.fq.gz \
| xargs basename -s _R1_001_val_1.fq.gz | xargs -I{} ${bismark_dir}/bismark \
--path_to_bowtie ${bowtie2_dir} \
-genome ${genome_folder} \
-p 4 \
-score_min L,0,-0.6 \
--non_directional \
-1 ${reads_dir}{}_R1_001_val_1.fq.gz \
-2 ${reads_dir}{}_R2_001_val_2.fq.gz \
-o Pact_tg



--------------------------------------------------------

# From here we extract methylation and create downstream amendable files.
# will split samples into dedup and nodedup directories for reasson
# RRBS data does NOT need to be deduplicated
# First the dedups which in our case would be

mkdir Mcap_tg
mkdir Mcap_tg/dedup
cp Mcap_tg/Meth10*bam Mcap_tg/dedup/
cp Mcap_tg/Meth11*bam Mcap_tg/dedup/
cp Mcap_tg/Meth12*bam Mcap_tg/dedup/
cp Mcap_tg/Meth16*bam Mcap_tg/dedup/
cp Mcap_tg/Meth17*bam Mcap_tg/dedup/
cp Mcap_tg/Meth18*bam Mcap_tg/dedup/

mkdir Mcap_tg/nodedup
cp Mcap_tg/Meth13*bam Mcap_tg/nodedup/
cp Mcap_tg/Meth14*bam Mcap_tg/nodedup/
cp Mcap_tg/Meth15*bam Mcap_tg/nodedup/


cd Mcap_tg/dedup


find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${bismark_dir}/deduplicate_bismark \
--bam \
--paired \
{}.bam



${bismark_dir}/bismark_methylation_extractor \
--bedGraph --counts --scaffolds \
--multicore 14 \
--buffer_size 75% \
*deduplicated.bam



# Bismark processing report

${bismark_dir}/bismark2report

#Bismark summary report

${bismark_dir}/bismark2summary



# Sort files for methylkit and IGV

find *deduplicated.bam | \
xargs basename -s .bam | \
xargs -I{} ${samtools} \
sort --threads 28 {}.bam \
-o {}.sorted.bam

# Index sorted files for IGV
# The "-@ 16" below specifies number of CPU threads to use.

find *.sorted.bam | \
xargs basename -s .sorted.bam | \
xargs -I{} ${samtools} \
index -@ 28 {}.sorted.bam


genome_folder="/gscratch/srlab/sr320/data/froger/Mcap_Genome/"


find *deduplicated.bismark.cov.gz \
| xargs basename -s deduplicated.bismark.cov.gz \
| xargs -I{} ${bismark_dir}/coverage2cytosine \
--genome_folder ${genome_folder} \
-o {} \
--merge_CpG \
--zero_based \
{}deduplicated.bismark.cov.gz


#creating bedgraphs post merge

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4}}' \
  > "${STEM}"_10x.bedgraph
done



for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4}}' \
  > "${STEM}"_5x.bedgraph
done


#creating tab files with raw count for glms

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_10x.tab
done


for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_5x.tab
done


------------------------


cd ../nodedup


${bismark_dir}/bismark_methylation_extractor \
--bedGraph --counts --scaffolds \
--multicore 14 \
--buffer_size 75% \
*.bam



# Bismark processing report

${bismark_dir}/bismark2report

#Bismark summary report

${bismark_dir}/bismark2summary



# Sort files for methylkit and IGV

find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${samtools} \
sort --threads 28 {}.bam \
-o {}.sorted.bam

# Index sorted files for IGV
# The "-@ 16" below specifies number of CPU threads to use.

find *.sorted.bam | \
xargs basename -s .sorted.bam | \
xargs -I{} ${samtools} \
index -@ 28 {}.sorted.bam




find *bismark.cov.gz \
| xargs basename -s bismark.cov.gz \
| xargs -I{} ${bismark_dir}/coverage2cytosine \
--genome_folder ${genome_folder} \
-o {} \
--merge_CpG \
--zero_based \
{}bismark.cov.gz


#creating bedgraphs post merge

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4}}' \
  > "${STEM}"_10x.bedgraph
done



for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4}}' \
  > "${STEM}"_5x.bedgraph
done


#creating tab files with raw count for glms

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_10x.tab
done


for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_5x.tab
done

-----------------------------------------------------------------------------


cd ../../

mkdir Pact_tg
mkdir Pact_tg/dedup
cp Pact_tg/Meth1*bam Pact_tg/dedup/
cp Pact_tg/Meth2*bam Pact_tg/dedup/
cp Pact_tg/Meth3*bam Pact_tg/dedup/
cp Pact_tg/Meth7*bam Pact_tg/dedup/
cp Pact_tg/Meth8*bam Pact_tg/dedup/
cp Pact_tg/Meth9*bam Pact_tg/dedup/

mkdir Pact_tg/nodedup
cp Pact_tg/Meth4*bam Pact_tg/nodedup/
cp Pact_tg/Meth5*bam Pact_tg/nodedup/
cp Pact_tg/Meth6*bam Pact_tg/nodedup/


cd Pact_tg/dedup


find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${bismark_dir}/deduplicate_bismark \
--bam \
--paired \
{}.bam



${bismark_dir}/bismark_methylation_extractor \
--bedGraph --counts --scaffolds \
--multicore 14 \
--buffer_size 75% \
*deduplicated.bam



# Bismark processing report

${bismark_dir}/bismark2report

#Bismark summary report

${bismark_dir}/bismark2summary



# Sort files for methylkit and IGV

find *deduplicated.bam | \
xargs basename -s .bam | \
xargs -I{} ${samtools} \
sort --threads 28 {}.bam \
-o {}.sorted.bam

# Index sorted files for IGV
# The "-@ 16" below specifies number of CPU threads to use.

find *.sorted.bam | \
xargs basename -s .sorted.bam | \
xargs -I{} ${samtools} \
index -@ 28 {}.sorted.bam


genome_folder="/gscratch/srlab/sr320/data/froger/Pact_Genome/"


find *deduplicated.bismark.cov.gz \
| xargs basename -s deduplicated.bismark.cov.gz \
| xargs -I{} ${bismark_dir}/coverage2cytosine \
--genome_folder ${genome_folder} \
-o {} \
--merge_CpG \
--zero_based \
{}deduplicated.bismark.cov.gz


#creating bedgraphs post merge

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4}}' \
  > "${STEM}"_10x.bedgraph
done



for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4}}' \
  > "${STEM}"_5x.bedgraph
done


#creating tab files with raw count for glms

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_10x.tab
done


for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_5x.tab
done


------------------------


cd ../nodedup


${bismark_dir}/bismark_methylation_extractor \
--bedGraph --counts --scaffolds \
--multicore 14 \
--buffer_size 75% \
*.bam



# Bismark processing report

${bismark_dir}/bismark2report

#Bismark summary report

${bismark_dir}/bismark2summary



# Sort files for methylkit and IGV

find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${samtools} \
sort --threads 28 {}.bam \
-o {}.sorted.bam

# Index sorted files for IGV
# The "-@ 16" below specifies number of CPU threads to use.

find *.sorted.bam | \
xargs basename -s .sorted.bam | \
xargs -I{} ${samtools} \
index -@ 28 {}.sorted.bam



find *bismark.cov.gz \
| xargs basename -s bismark.cov.gz \
| xargs -I{} ${bismark_dir}/coverage2cytosine \
--genome_folder ${genome_folder} \
-o {} \
--merge_CpG \
--zero_based \
{}bismark.cov.gz


#creating bedgraphs post merge

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4}}' \
  > "${STEM}"_10x.bedgraph
done



for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4}}' \
  > "${STEM}"_5x.bedgraph
done


#creating tab files with raw count for glms

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_10x.tab
done


for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_5x.tab
done

```

---

**Output from Bismark Aligment ++**

https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/031520-TG-bs/
