---
title: "Chacterizing CpG Methylation (5x Union Data)"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this script, I'll create summary tables and figures to characterize CpG methylation in *M. capitata* and *P. acuta* 5x union bedgraphs for WGBS, RRBS, and MBD-BS.

# Install packages

```{r}
#install.packages("RColorBrewer")
require(RColorBrewer) #Load RColorBrewer

#install.packages("dichromat")
require(dichromat)

#install.packages("broom")
require(broom)
```

# Session information

```{r}
sessionInfo()
```

# CpG distributions

## Mcap

### Import file counts

```{r}
McapAll <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-averages-counts.txt", header = FALSE, col.names = c("totalLines", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
McapAll <- McapAll[-4,] #Remove last row (total lines for all files)
tail(McapAll) #Confirm import
```

```{r}
McapMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-Meth-counts.txt", header = FALSE, col.names = c("Meth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
McapMeth <- McapMeth[-4,] #Remove last row (total lines for all files)
tail(McapMeth) #Confirm import
```

```{r}
McapSparseMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-sparseMeth-counts.txt", header = FALSE, col.names = c("sparseMeth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
McapSparseMeth <- McapSparseMeth[-4,] #Remove last row (total lines for all files)
tail(McapSparseMeth) #Confirm import
```

```{r}
McapUnMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-unMeth-counts.txt", header = FALSE, col.names = c("unMeth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
McapUnMeth <- McapUnMeth[-4,] #Remove last row (total lines for all files)
tail(McapUnMeth) #Confirm import
```

### Create summary table

```{r}
McapCpGType <- cbind(McapAll, McapMeth, McapSparseMeth, McapUnMeth) #Mash tables together by column
McapCpGType <- McapCpGType[,-c(2,4,6,8)] #Remove filename columns
tail(McapCpGType) #Confirm table mashing
```

```{r}
McapCpGType$percentMeth <- (McapCpGType$Meth / McapCpGType$totalLines) * 100 #Calculate percent methylated loci
McapCpGType$percentSparseMeth <- (McapCpGType$sparseMeth / McapCpGType$totalLines) * 100 #Calculate percent sparsely methylated loci
McapCpGType$percentUnMeth <- (McapCpGType$unMeth / McapCpGType$totalLines) * 100 #Calculate percent unmethylated loci
McapCpGType <- McapCpGType[,c(1, 2, 5, 3, 6, 4, 7)] #Reorganize columns
tail(McapCpGType) #Confirm calculations
```

```{r}
write.table(McapCpGType, "../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union-CpG-Type.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

### Conduct chi-squared tests of homogeneity

The null hypothesis is that distributions of different CpG types is the same between sequencing methods

```{r}
McapCpGTypeStatTest <- McapCpGType[,c(2,4,6)] #Separate out count data for statistical testing
row.names(McapCpGTypeStatTest) <- c("WGBS", "RRBS", "MBDBS") #Add row names
head(McapCpGTypeStatTest) #Confirm changes
```

#### WGBS vs. RRBS

```{r}
McapCpGTypeWGRR <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(McapCpGTypeStatTest)) { #For each CpG type
  Method1CpG <- McapCpGTypeStatTest[1,i] #Variable for # CpG type for method 1
  Method2CpG <- McapCpGTypeStatTest[2,i] #Variable for # CpG type for method 2
  Method1NotCpG <- sum(McapCpGTypeStatTest[1,-i]) #Variable for # other CpG types for method 1
  Method2NotCpG <- sum(McapCpGTypeStatTest[2,-i]) #Variable for # other CpG types for method 2
  ct <- matrix(c(Method1CpG,Method2CpG,Method1NotCpG,Method2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(McapCpGTypeStatTest[i])), paste0("Not", colnames(McapCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c(as.character(row.names(McapCpGTypeStatTest)[1:2])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(McapCpGTypeStatTest)[i]) #Add CpG type to results
  McapCpGTypeWGRR <- rbind(McapCpGTypeWGRR, ctResults) #Add test statistics to master table
}
```

```{r}
McapCpGTypeWGRR$p.adj <- p.adjust(McapCpGTypeWGRR$p.value, method = "fdr") #Correct p-value using FDR
McapCpGTypeWGRR$comparison <- rep("WGBS vs. RRBS", times = 3) #Add methods compared
head(McapCpGTypeWGRR) #Confirm changes
```

#### WGBS vs. MBD-BS

```{r}
McapCpGTypeWGMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(McapCpGTypeStatTest)) { #For each CpG type
  Method1CpG <- McapCpGTypeStatTest[1,i] #Variable for # CpG type for method 1
  Method2CpG <- McapCpGTypeStatTest[3,i] #Variable for # CpG type for method 2
  Method1NotCpG <- sum(McapCpGTypeStatTest[1,-i]) #Variable for # other CpG types for method 1
  Method2NotCpG <- sum(McapCpGTypeStatTest[3,-i]) #Variable for # other CpG types for method 2
  ct <- matrix(c(Method1CpG,Method2CpG,Method1NotCpG,Method2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(McapCpGTypeStatTest[i])), paste0("Not", colnames(McapCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c(as.character(row.names(McapCpGTypeStatTest)[c(1,3)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(McapCpGTypeStatTest)[i]) #Add CpG type to results
  McapCpGTypeWGMB <- rbind(McapCpGTypeWGMB, ctResults) #Add test statistics to master table
}
```

```{r}
McapCpGTypeWGMB$p.adj <- p.adjust(McapCpGTypeWGMB$p.value, method = "fdr") #Correct p-value using FDR
McapCpGTypeWGMB$comparison <- rep("WGBS vs. MBDBS", times = 3) #Add methods compared
head(McapCpGTypeWGMB) #Confirm changes
```

#### RRBS vs. MBD-BS

```{r}
McapCpGTypeRRMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(McapCpGTypeStatTest)) { #For each CpG type
  Method1CpG <- McapCpGTypeStatTest[2,i] #Variable for # CpG type for method 1
  Method2CpG <- McapCpGTypeStatTest[3,i] #Variable for # CpG type for method 2
  Method1NotCpG <- sum(McapCpGTypeStatTest[2,-i]) #Variable for # other CpG types for method 1
  Method2NotCpG <- sum(McapCpGTypeStatTest[3,-i]) #Variable for # other CpG types for method 2
  ct <- matrix(c(Method1CpG,Method2CpG,Method1NotCpG,Method2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(McapCpGTypeStatTest[i])), paste0("Not", colnames(McapCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c(as.character(row.names(McapCpGTypeStatTest)[c(2,3)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(McapCpGTypeStatTest)[i]) #Add CpG type to results
  McapCpGTypeRRMB <- rbind(McapCpGTypeRRMB, ctResults) #Add test statistics to master table
}
```

```{r}
McapCpGTypeRRMB$p.adj <- p.adjust(McapCpGTypeRRMB$p.value, method = "fdr") #Correct p-value using FDR
McapCpGTypeRRMB$comparison <- rep("RRBS vs. MBDBS", times = 3) #Add methods compared
head(McapCpGTypeRRMB) #Confirm changes
```

#### Save statistical test results

```{r}
McapCpGTypeStatResults <- rbind(McapCpGTypeWGRR,
                                McapCpGTypeWGMB,
                                McapCpGTypeRRMB) #Combine tables
tail(McapCpGTypeStatResults) #Confirm changes
```

```{r}
write.table(McapCpGTypeStatResults, "../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union-CpG-Type-StatResults.txt", quote = FALSE, row.names = FALSE) #Save table
```

### Create plot

```{r}
McapCpGTypePercents <- McapCpGType[,c(3,5,7)] #Keep only percent information
head(McapCpGTypePercents) #Confirm changes
```

```{r}
plotColors <- c(rev(RColorBrewer::brewer.pal(3, "Greens")), 
                rev(RColorBrewer::brewer.pal(3, "Blues")), 
                rev(RColorBrewer::brewer.pal(3, "Reds"))) #Create vector of WGBS, RRBS, and MBD colors
barplot(t(McapCpGTypePercents), col = dichromat(c(rev(RColorBrewer::brewer.pal(3, "Greens"))))) #Check greens
barplot(t(McapCpGTypePercents), col = dichromat(c(rev(RColorBrewer::brewer.pal(3, "Blues"))))) #Check blues
barplot(t(McapCpGTypePercents), col = dichromat(c(rev(RColorBrewer::brewer.pal(3, "Reds"))))) #Check reds
```

```{r}
#pdf("../Output/Mcap_union-CpG-Type.pdf", height = 8.5, width = 11)
par(mar = c(3,5,1,1), oma = c(2, 1, 1, 1)) #Change figure boundaries

barplot(t(McapCpGTypePercents), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("WGBS", "RRBS", "MBD-BS"), cex.names = 1.2,
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs. 

for (i in 1:ncol(t(McapCpGTypePercents))){
    xx <- t(McapCpGTypePercents)
    xx[,-i] <- NA
    colnames(xx)[-i] <- NA
    barplot(xx, col = c(plotColors[(3*i)-2], plotColors[(3*i)-1], plotColors[(3*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "Proportion CpGs", line = 3, cex = 1.5) #Add y-axis label
mtext(side = 1, "Sequencing Method", line = 3, cex = 1.5) #Add x-axis label

#segments(x0 = 1, y0 = 106, x1 = 3, y1 = 106, lwd = 1.5) #Add line connecting WGBS and MBD-BSSeq
#segments(x0 = 1, y0 = 106, x1 = 1, y1 = 103, lwd = 1.5) #Add left segment from connecting line to WGBS
#segments(x0 = 3, y0 = 106, x1 = 3, y1 = 103, lwd = 1.5) #Add right segment from connecting line to MBD
#text("p = 0.019", x = 2, y = 108, cex = 1.2) #Add p-value

#dev.off() #Turn off plotting device
```





## Pact

### Import file counts

```{r}
PactAll <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-averages-counts.txt", header = FALSE, col.names = c("totalLines", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
PactAll <- PactAll[-4,] #Remove last row (total lines for all files)
tail(PactAll) #Confirm import
```

```{r}
PactMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-Meth-counts.txt", header = FALSE, col.names = c("Meth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
PactMeth <- PactMeth[-4,] #Remove last row (total lines for all files)
tail(PactMeth) #Confirm import
```

```{r}
PactSparseMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-sparseMeth-counts.txt", header = FALSE, col.names = c("sparseMeth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
PactSparseMeth <- PactSparseMeth[-4,] #Remove last row (total lines for all files)
tail(PactSparseMeth) #Confirm import
```

```{r}
PactUnMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-unMeth-counts.txt", header = FALSE, col.names = c("unMeth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
PactUnMeth <- PactUnMeth[-4,] #Remove last row (total lines for all files)
tail(PactUnMeth) #Confirm import
```

### Create summary table

```{r}
PactCpGType <- cbind(PactAll, PactMeth, PactSparseMeth, PactUnMeth) #Mash tables together by column
PactCpGType <- PactCpGType[,-c(2,4,6,8)] #Remove filename columns
tail(PactCpGType) #Confirm table mashing
```

```{r}
PactCpGType$percentMeth <- (PactCpGType$Meth / PactCpGType$totalLines) * 100 #Calculate percent methylated loci
PactCpGType$percentSparseMeth <- (PactCpGType$sparseMeth / PactCpGType$totalLines) * 100 #Calculate percent sparsely methylated loci
PactCpGType$percentUnMeth <- (PactCpGType$unMeth / PactCpGType$totalLines) * 100 #Calculate percent unmethylated loci
PactCpGType <- PactCpGType[,c(1, 2, 5, 3, 6, 4, 7)] #Reorganize columns
tail(PactCpGType) #Confirm calculations
```

```{r}
write.table(PactCpGType, "../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-CpG-Type.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

### Conduct chi-squared tests of homogeneity

The null hypothesis is that distributions of different CpG types is the same between sequencing methods

```{r}
PactCpGTypeStatTest <- PactCpGType[,c(2,4,6)] #Separate out count data for statistical testing
row.names(PactCpGTypeStatTest) <- c("WGBS", "RRBS", "MBDBS") #Add row names
head(PactCpGTypeStatTest) #Confirm changes
```

#### WGBS vs. RRBS

```{r}
PactCpGTypeWGRR <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactCpGTypeStatTest)) { #For each CpG type
  Method1CpG <- PactCpGTypeStatTest[1,i] #Variable for # CpG type for method 1
  Method2CpG <- PactCpGTypeStatTest[2,i] #Variable for # CpG type for method 2
  Method1NotCpG <- sum(PactCpGTypeStatTest[1,-i]) #Variable for # other CpG types for method 1
  Method2NotCpG <- sum(PactCpGTypeStatTest[2,-i]) #Variable for # other CpG types for method 2
  ct <- matrix(c(Method1CpG,Method2CpG,Method1NotCpG,Method2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactCpGTypeStatTest[i])), paste0("Not", colnames(PactCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c(as.character(row.names(PactCpGTypeStatTest)[1:2])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(PactCpGTypeStatTest)[i]) #Add CpG type to results
  PactCpGTypeWGRR <- rbind(PactCpGTypeWGRR, ctResults) #Add test statistics to master table
}
```

```{r}
PactCpGTypeWGRR$p.adj <- p.adjust(PactCpGTypeWGRR$p.value, method = "fdr") #Correct p-value using FDR
PactCpGTypeWGRR$comparison <- rep("WGBS vs. RRBS", times = 3) #Add methods compared
head(PactCpGTypeWGRR) #Confirm changes
```

#### WGBS vs. MBD-BS

```{r}
PactCpGTypeWGMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactCpGTypeStatTest)) { #For each CpG type
  Method1CpG <- PactCpGTypeStatTest[1,i] #Variable for # CpG type for method 1
  Method2CpG <- PactCpGTypeStatTest[3,i] #Variable for # CpG type for method 2
  Method1NotCpG <- sum(PactCpGTypeStatTest[1,-i]) #Variable for # other CpG types for method 1
  Method2NotCpG <- sum(PactCpGTypeStatTest[3,-i]) #Variable for # other CpG types for method 2
  ct <- matrix(c(Method1CpG,Method2CpG,Method1NotCpG,Method2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactCpGTypeStatTest[i])), paste0("Not", colnames(PactCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c(as.character(row.names(PactCpGTypeStatTest)[c(1,3)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(PactCpGTypeStatTest)[i]) #Add CpG type to results
  PactCpGTypeWGMB <- rbind(PactCpGTypeWGMB, ctResults) #Add test statistics to master table
}
```

```{r}
PactCpGTypeWGMB$p.adj <- p.adjust(PactCpGTypeWGMB$p.value, method = "fdr") #Correct p-value using FDR
PactCpGTypeWGMB$comparison <- rep("WGBS vs. MBDBS", times = 3) #Add methods compared
head(PactCpGTypeWGMB) #Confirm changes
```

#### RRBS vs. MBD-BS

```{r}
PactCpGTypeRRMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactCpGTypeStatTest)) { #For each CpG type
  Method1CpG <- PactCpGTypeStatTest[2,i] #Variable for # CpG type for method 1
  Method2CpG <- PactCpGTypeStatTest[3,i] #Variable for # CpG type for method 2
  Method1NotCpG <- sum(PactCpGTypeStatTest[2,-i]) #Variable for # other CpG types for method 1
  Method2NotCpG <- sum(PactCpGTypeStatTest[3,-i]) #Variable for # other CpG types for method 2
  ct <- matrix(c(Method1CpG,Method2CpG,Method1NotCpG,Method2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactCpGTypeStatTest[i])), paste0("Not", colnames(PactCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c(as.character(row.names(PactCpGTypeStatTest)[c(2,3)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(PactCpGTypeStatTest)[i]) #Add CpG type to results
  PactCpGTypeRRMB <- rbind(PactCpGTypeRRMB, ctResults) #Add test statistics to master table
}
```

```{r}
PactCpGTypeRRMB$p.adj <- p.adjust(PactCpGTypeRRMB$p.value, method = "fdr") #Correct p-value using FDR
PactCpGTypeRRMB$comparison <- rep("RRBS vs. MBDBS", times = 3) #Add methods compared
head(PactCpGTypeRRMB) #Confirm changes
```

#### Save statistical test results

```{r}
PactCpGTypeStatResults <- rbind(PactCpGTypeWGRR,
                                PactCpGTypeWGMB,
                                PactCpGTypeRRMB) #Combine tables
tail(PactCpGTypeStatResults) #Confirm changes
```

```{r}
write.table(PactCpGTypeStatResults, "../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union-CpG-Type-StatResults.txt", quote = FALSE, row.names = FALSE) #Save table
```

### Create plot

```{r}
PactCpGTypePercents <- PactCpGType[,c(3,5,7)] #Keep only percent information
head(PactCpGTypePercents) #Confirm changes
```

```{r}
#pdf("../Output/Pact-CpG-Type.pdf", height = 8.5, width = 11)
par(mar = c(3,5,0,1), oma = c(2, 1, 1, 1)) #Change figure boundaries

barplot(t(PactCpGTypePercents), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("WGBS", "RRBS", "MBD-BS"), cex.names = 1.2,
        ylim = c(0, 115)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs. 

for (i in 1:ncol(t(PactCpGTypePercents))){
    xx <- t(PactCpGTypePercents)
    xx[,-i] <- NA
    colnames(xx)[-i] <- NA
    barplot(xx, col = c(plotColors[(3*i)-2], plotColors[(3*i)-1], plotColors[(3*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "Proportion CpGs", line = 3, cex = 1.5) #Add y-axis label
mtext(side = 1, "Sequencing Method", line = 3, cex = 1.5) #Add x-axis label

#egments(x0 = 2, y0 = 106, x1 = 3, y1 = 106, lwd = 1.5) #Add line connecting RRBS and MBD-BSSeq
#segments(x0 = 2, y0 = 106, x1 = 2, y1 = 103, lwd = 1.5) #Add left segment from connecting line to RRBS
#segments(x0 = 3, y0 = 106, x1 = 3, y1 = 103, lwd = 1.5) #Add right segment from connecting line to MBD-BSSeq
#text("p = 0.016", x = 2.5, y = 108, cex = 1.2) #Add p-value

#segments(x0 = 1, y0 = 111, x1 = 3, y1 = 111, lwd = 1.5) #Add line connecting WGBS and MBD-BSSeq
#segments(x0 = 1, y0 = 111, x1 = 1, y1 = 108, lwd = 1.5) #Add left segment from connecting line to WGBS
#segments(x0 = 3, y0 = 111, x1 = 3, y1 = 108, lwd = 1.5) #Add right segment from connecting line to MBD-BSSeq
#text("p = 0.006", x = 2, y = 113, cex = 1.2) #Add p-value

#dev.off() #Turn off plotting device
```

## Interspecies chi-squared tests

The hypothesis is that the distribution of CpG type will be the same between species for the same methods used.

#### WGBS

```{r}
WGBSCpGType <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactCpGTypeStatTest)) { #For each CpG type
  Species1CpG <- McapCpGTypeStatTest[1,i] #Variable for # CpG type for Species 1
  Species2CpG <- PactCpGTypeStatTest[1,i] #Variable for # CpG type for Species 2
  Species1NotCpG <- sum(McapCpGTypeStatTest[1,-i]) #Variable for # other CpG types for Species 1
  Species2NotCpG <- sum(PactCpGTypeStatTest[1,-i]) #Variable for # other CpG types for Species 2
  ct <- matrix(c(Species1CpG,Species2CpG,Species1NotCpG,Species2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactCpGTypeStatTest[i])), paste0("Not", colnames(PactCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c("Mcap", "Pact") #Assign row names: Species 1, Species 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(PactCpGTypeStatTest)[i]) #Add CpG type to results
  WGBSCpGType <- rbind(WGBSCpGType, ctResults) #Add test statistics to master table
}
```

```{r}
WGBSCpGType$p.adj <- p.adjust(WGBSCpGType$p.value, method = "fdr") #Correct p-value using FDR
WGBSCpGType$comparison <- rep("Mcap vs. Pact", times = 3) #Add methods compared
head(WGBSCpGType) #Confirm changes
```

#### RRBS

```{r}
RRBSCpGType <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactCpGTypeStatTest)) { #For each CpG type
  Species1CpG <- McapCpGTypeStatTest[2,i] #Variable for # CpG type for Species 1
  Species2CpG <- PactCpGTypeStatTest[2,i] #Variable for # CpG type for Species 2
  Species1NotCpG <- sum(McapCpGTypeStatTest[2,-i]) #Variable for # other CpG types for Species 1
  Species2NotCpG <- sum(PactCpGTypeStatTest[2,-i]) #Variable for # other CpG types for Species 2
  ct <- matrix(c(Species1CpG,Species2CpG,Species1NotCpG,Species2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactCpGTypeStatTest[i])), paste0("Not", colnames(PactCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c("Mcap", "Pact") #Assign row names: Species 1, Species 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(PactCpGTypeStatTest)[i]) #Add CpG type to results
  RRBSCpGType <- rbind(RRBSCpGType, ctResults) #Add test statistics to master table
}
```

```{r}
RRBSCpGType$p.adj <- p.adjust(RRBSCpGType$p.value, method = "fdr") #Correct p-value using FDR
RRBSCpGType$comparison <- rep("Mcap vs. Pact", times = 3) #Add methods compared
head(RRBSCpGType) #Confirm changes
```

#### MBD-BS



#### Save statistical test results

```{r}
PactCpGTypeStatResults <- rbind(PactCpGTypeWGRR,
                                PactCpGTypeWGMB,
                                PactCpGTypeRRMB) #Combine tables
tail(PactCpGTypeStatResults) #Confirm changes
```

```{r}
write.table(PactCpGTypeStatResults, "../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union-CpG-Type-StatResults.txt", quote = FALSE, row.names = FALSE) #Save table
```




# CpG feature overlaps

## Mcap

### Import file counts

```{r}
McapGenomeFeatures <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-CGMotif-Overlaps-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with CG motif and feature track overlaps
McapGenomeFeatures <- McapGenomeFeatures[-6,] #Remove final row
tail(McapGenomeFeatures) #Check import
```

```{r}
McapGeneOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-mcGenes-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-gene overlaps
McapGeneOverlaps <- McapGeneOverlaps[-13,] #Remove final row
tail(McapGeneOverlaps) #Confirm import
```

```{r}
McapCDSOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-mcCDS-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-CDS overlaps
McapCDSOverlaps <- McapCDSOverlaps[-13,] #Remove final row
tail(McapCDSOverlaps) #Confirm import
```

```{r}
McapIntronsOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-mcIntrons-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
McapIntronsOverlaps <- McapIntronsOverlaps[-13,] #Remove final row
tail(McapIntronsOverlaps) #Confirm import
```

```{r}
McapFlanksOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-mcFlanks-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
McapFlanksOverlaps <- McapFlanksOverlaps[-13,] #Remove final row
tail(McapFlanksOverlaps) #Confirm import
```

```{r}
McapIntergenicOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-mcIntergenic-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Intergenic overlaps
McapIntergenicOverlaps <- McapIntergenicOverlaps[-13,] #Remove final row
tail(McapIntergenicOverlaps) #Confirm import
```

### Create summary tables

```{r}
McapFeatureOverlaps <- data.frame("allCpGs" = rep(0, times = 5),
                                  "MBDBSMeth" = rep(0, times = 5),
                                  "MBDBSsparseMeth" = rep(0, times = 5),
                                  "MBDBSunMeth" = rep(0, times = 5),
                                  "MBDBS" = rep(0, times = 5),
                                  "RRBSMeth" = rep(0, times = 5),
                                  "RRBSsparseMeth" = rep(0, times = 5),
                                  "RRBSunMeth" = rep(0, times = 5),
                                  "RRBS" = rep(0, times = 5),
                                  "WGBSMeth" = rep(0, times = 5),
                                  "WGBSsparseMeth" = rep(0, times = 5),
                                  "WGBSunMeth" = rep(0, times = 5),                                  
                                  "WGBS" = rep(0, times = 5)) #Create blank dataframe with information for various CpG categories and methylation status. Match columns to the order of columns in the overlap count files
row.names(McapFeatureOverlaps) <- c("Genes", "CDS", "Introns", "Flanks", "Intergenic") #Assign row names
head(McapFeatureOverlaps) #Confirm changes
```

```{r}
McapFeatureOverlaps$allCpGs <- c(McapGenomeFeatures$counts[3],
                                 McapGenomeFeatures$counts[1],
                                 McapGenomeFeatures$counts[5],
                                 McapGenomeFeatures$counts[2],
                                 McapGenomeFeatures$counts[4]) #Assign information for CG motif overlaps with genome features. Use 0 for totalLines
head(McapFeatureOverlaps) #Confirm modification
```

```{r}
for (i in 1:length(McapGeneOverlaps$counts)) {
  McapFeatureOverlaps[1,i+1] <- McapGeneOverlaps[i,1]
  McapFeatureOverlaps[2,i+1] <- McapCDSOverlaps[i,1]
  McapFeatureOverlaps[3,i+1] <- McapIntronsOverlaps[i,1]
  McapFeatureOverlaps[4,i+1] <- McapFlanksOverlaps[i,1]
  McapFeatureOverlaps[5,i+1] <- McapIntergenicOverlaps[i,1]
} #For each table with feature overlap information, paste the contents of the count column in the assigned row
head(McapFeatureOverlaps) #Check summary table
```

```{r}
write.table(McapFeatureOverlaps, "../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union-Genomic-Location-Counts.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

```{r}
McapFeatureOverlapsPercents <- McapFeatureOverlaps[-1,] #Duplicate dataframe but remove gene row
for (i in 1:length(McapFeatureOverlaps)) {
  McapFeatureOverlapsPercents[,i] <- (McapFeatureOverlapsPercents[,i] / (sum(McapFeatureOverlapsPercents[,i]))) * 100
} #Divide every entry by sum of the column and multiply by 100 to get percentages. Do not include gene information
head(McapFeatureOverlapsPercents) #Check calculations
```

```{r}
write.table(McapFeatureOverlapsPercents, "../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union-Genomic-Location-Percents.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

### Conduct chi-squared tests of homogeneity

```{r}
McapFeatureAllWG <- chisq.test(McapFeatureOverlaps[,c(1,13)]) #All CpGs vs. WGBS
McapFeatureAllWG #The distribution of CpGs in genome features is not significantly different between all CpGs and WGBS
```

```{r}
McapFeatureAllRR <- chisq.test(McapFeatureOverlaps[,c(1,9)]) #All CpGs vs. RRBS
McapFeatureAllRR #The distribution of CpGs in genome features is not significantly different between all CpGs and RRBS
```

```{r}
McapFeatureAllMB <- chisq.test(McapFeatureOverlaps[,c(1,5)]) #All CpGs vs. MBDBS
McapFeatureAllMB #The distribution of CpGs in genome features is not significantly different between all CpGs and WGBS
```

```{r}
McapFeatureWGRR <- chisq.test(McapFeatureOverlaps[,c(13, 9)]) #WGBS vs. RRBS
McapFeatureWGRR #The distribution of CpGs is not sginificantly different between WGBS and RRBS
```

```{r}
McapFeatureWGMB <- chisq.test(McapFeatureOverlaps[,c(13, 5),]) #WGBS vs. MBD-BSSeq
McapFeatureWGMB #The distribution of CpGs is sginificantly different between WGBS and MBD-BSSeq
```

```{r}
McapFeatureRRMB <- chisq.test(McapFeatureOverlaps[,c(9, 5)]) #RRBS vs. MBD-BSSeq
McapFeatureRRMB #The distribution of CpGs is not significantly different between RRBS vs. MBD-BSSeq
```

### Create plot

```{r}
plotColors2 <- c("grey20", "grey40", "grey60", "grey80", 
                rev(RColorBrewer::brewer.pal(4, "Greens")), 
                rev(RColorBrewer::brewer.pal(4, "Blues")), 
                rev(RColorBrewer::brewer.pal(4, "Reds"))) #Create vector of all, WGBS, RRBS, and MBD colors
barplot(t(McapCpGTypePercents), col = dichromat(c(rev(RColorBrewer::brewer.pal(3, "Greens"))))) #Check greens
barplot(t(McapCpGTypePercents), col = dichromat(c(rev(RColorBrewer::brewer.pal(3, "Blues"))))) #Check blues
barplot(t(McapCpGTypePercents), col = dichromat(c(rev(RColorBrewer::brewer.pal(3, "Reds"))))) #Check reds
```

#### All data

```{r}
#pdf("../Output/Mcap-CpG-Features.pdf", height = 8.5, width = 11)
par(mar = c(3,5,0,1), oma = c(2, 1, 1, 1)) #Change figure boundaries

barplot(t(t(McapFeatureOverlapsPercents[,c(1,13,9,5)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("All", "WGBS", "RRBS", "MBD-BS"), cex.names = 1.2,
        ylim = c(0, 115)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs. 

for (i in 1:4) {
    xx <- t(t(McapFeatureOverlapsPercents[,c(1,13,9,5)]))
    xx[,-i] <- NA
    colnames(xx)[-i] <- NA
    barplot(xx, col = c(plotColors2[(4*i)-3], plotColors2[(4*i)-2], plotColors2[(4*i)-1], plotColors2[(4*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "Proportion CpGs", line = 3, cex = 1.5) #Add y-axis label
mtext(side = 1, "Sequencing Method", line = 3, cex = 1.5) #Add x-axis label

#dev.off() #Turn off plotting device
```

#### CpG feature overlap by methylation status

```{r}
#pdf("../Output/Mcap-CpG-Features-MethStatus.pdf", height = 8.5, width = 11)
par(mar = c(1,5,0,1), oma = c(4, 1, 1, 1), mfcol = c(3,1)) #Change figure boundaries

#Methylated CpGs

barplot(t(t(McapFeatureOverlapsPercents[,c(10,6,2)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("", "", ""),
        ylim = c(0, 115)) #Create base plot. Everything should be white. Do not plot axes. Set y-axis specs.

for (i in 2:4) {
    xx <- t(t(McapFeatureOverlapsPercents[,c(10,6,2)]))
    xx[,-(i-1)] <- NA
    colnames(xx)[-(i-1)] <- NA
    barplot(xx, col = c(plotColors2[(4*i)-3], plotColors2[(4*i)-2], plotColors2[(4*i)-1], plotColors2[(4*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% Methylated", line = 3, cex = 1.5) #Add y-axis label

#Sparsely methylated CpGs

barplot(t(t(McapFeatureOverlapsPercents[,c(11,7,3)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("", "", ""),
        ylim = c(0, 115)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs.

for (i in 2:4) {
    xx <- t(t(McapFeatureOverlapsPercents[,c(11,7,3)]))
    xx[,-(i-1)] <- NA
    colnames(xx)[-(i-1)] <- NA
    barplot(xx, col = c(plotColors2[(4*i)-3], plotColors2[(4*i)-2], plotColors2[(4*i)-1], plotColors2[(4*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% Sparsely Methylated", line = 3, cex = 1.5) #Add y-axis label

#Unmethylated CpGs

barplot(t(t(McapFeatureOverlapsPercents[,c(12,8,4)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("WGBS", "RRBS", "MBD-BS"), cex.names = 2,
        ylim = c(0, 115)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs.

for (i in 2:4) {
    xx <- t(t(McapFeatureOverlapsPercents[,c(12,8,4)]))
    xx[,-(i-1)] <- NA
    colnames(xx)[-(i-1)] <- NA
    barplot(xx, col = c(plotColors2[(4*i)-3], plotColors2[(4*i)-2], plotColors2[(4*i)-1], plotColors2[(4*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% Unmethylated", line = 3, cex = 1.5) #Add y-axis label

#dev.off() #Turn off plotting device
```

## Pact

### Import file counts

```{r}
PactGenomeFeatures <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Pact/Pact-CGMotif-Overlaps-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with CG motif and feature track overlaps
PactGenomeFeatures <- PactGenomeFeatures[-6,] #Remove final row
tail(PactGenomeFeatures) #Check import
```

```{r}
PactGeneOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-paGenes-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-gene overlaps
PactGeneOverlaps <- PactGeneOverlaps[-13,] #Remove final row
tail(PactGeneOverlaps) #Confirm import
```

```{r}
PactCDSOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-paCDS-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-CDS overlaps
PactCDSOverlaps <- PactCDSOverlaps[-13,] #Remove final row
tail(PactCDSOverlaps) #Confirm import
```

```{r}
PactIntronsOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-paIntron-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
PactIntronsOverlaps <- PactIntronsOverlaps[-13,] #Remove final row
tail(PactIntronsOverlaps) #Confirm import
```

```{r}
PactFlanksOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-paFlanks-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
PactFlanksOverlaps <- PactFlanksOverlaps[-13,] #Remove final row
tail(PactFlanksOverlaps) #Confirm import
```

```{r}
PactIntergenicOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-paIntergenic-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Intergenic overlaps
PactIntergenicOverlaps <- PactIntergenicOverlaps[-13,] #Remove final row
tail(PactIntergenicOverlaps) #Confirm import
```

### Create summary tables

```{r}
PactFeatureOverlaps <- data.frame("allCpGs" = rep(0, times = 5),
                                  "MBDBSMeth" = rep(0, times = 5),
                                  "MBDBSsparseMeth" = rep(0, times = 5),
                                  "MBDBSunMeth" = rep(0, times = 5),
                                  "MBDBS" = rep(0, times = 5),
                                  "RRBSMeth" = rep(0, times = 5),
                                  "RRBSsparseMeth" = rep(0, times = 5),
                                  "RRBSunMeth" = rep(0, times = 5),
                                  "RRBS" = rep(0, times = 5),
                                  "WGBSMeth" = rep(0, times = 5),
                                  "WGBSsparseMeth" = rep(0, times = 5),
                                  "WGBSunMeth" = rep(0, times = 5),                                  
                                  "WGBS" = rep(0, times = 5)) #Create blank dataframe with information for various CpG categories and methylation status. Match columns to the order of columns in the overlap count files
row.names(PactFeatureOverlaps) <- c("Genes", "CDS", "Introns", "Flanks", "Intergenic") #Assign row names
head(PactFeatureOverlaps) #Confirm changes
```

```{r}
PactFeatureOverlaps$allCpGs <- c(PactGenomeFeatures$counts[3],
                                 PactGenomeFeatures$counts[1],
                                 PactGenomeFeatures$counts[5],
                                 PactGenomeFeatures$counts[2],
                                 PactGenomeFeatures$counts[4]) #Assign information for CG motif overlaps with genome features. Use 0 for totalLines
head(PactFeatureOverlaps) #Confirm modification
```

```{r}
for (i in 1:length(PactGeneOverlaps$counts)) {
  PactFeatureOverlaps[1,i+1] <- PactGeneOverlaps[i,1]
  PactFeatureOverlaps[2,i+1] <- PactCDSOverlaps[i,1]
  PactFeatureOverlaps[3,i+1] <- PactIntronsOverlaps[i,1]
  PactFeatureOverlaps[4,i+1] <- PactFlanksOverlaps[i,1]
  PactFeatureOverlaps[5,i+1] <- PactIntergenicOverlaps[i,1]
} #For each table with feature overlap information, paste the contents of the count column in the assigned row
head(PactFeatureOverlaps) #Check summary table
```

```{r}
write.table(PactFeatureOverlaps, "../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union-Genomic-Location-Counts.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

```{r}
PactFeatureOverlapsPercents <- PactFeatureOverlaps[-1,] #Duplicate dataframe but remove gene row
for (i in 1:length(PactFeatureOverlaps)) {
  PactFeatureOverlapsPercents[,i] <- (PactFeatureOverlapsPercents[,i] / (sum(PactFeatureOverlapsPercents[,i]))) * 100
} #Divide every entry by sum of the column and multiply by 100 to get percentages. Do not include gene information
head(PactFeatureOverlapsPercents) #Check calculations
```

```{r}
write.table(PactFeatureOverlapsPercents, "../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union-Genomic-Location-Percents.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

### Conduct chi-squared tests of homogeneity

```{r}
PactFeatureAllWG <- chisq.test(PactFeatureOverlaps[,c(1,13)]) #All CpGs vs. WGBS
PactFeatureAllWG #The distribution of CpGs in genome features is not significantly different between all CpGs and WGBS
```

```{r}
PactFeatureAllRR <- chisq.test(PactFeatureOverlaps[,c(1,9)]) #All CpGs vs. RRBS
PactFeatureAllRR #The distribution of CpGs in genome features is not significantly different between all CpGs and RRBS
```

```{r}
PactFeatureAllMB <- chisq.test(PactFeatureOverlaps[,c(1,5)]) #All CpGs vs. MBDBS
PactFeatureAllMB #The distribution of CpGs in genome features is not significantly different between all CpGs and WGBS
```

```{r}
PactFeatureWGRR <- chisq.test(PactFeatureOverlaps[,c(13, 9)]) #WGBS vs. RRBS
PactFeatureWGRR #The distribution of CpGs is not sginificantly different between WGBS and RRBS
```

```{r}
PactFeatureWGMB <- chisq.test(PactFeatureOverlaps[,c(13, 5),]) #WGBS vs. MBD-BSSeq
PactFeatureWGMB #The distribution of CpGs is sginificantly different between WGBS and MBD-BSSeq
```

```{r}
PactFeatureRRMB <- chisq.test(PactFeatureOverlaps[,c(9, 5)]) #RRBS vs. MBD-BSSeq
PactFeatureRRMB #The distribution of CpGs is not significantly different between RRBS vs. MBD-BSSeq
```

### Create plot

#### All data

```{r}
#pdf("../Output/Pact-CpG-Features.pdf", height = 8.5, width = 11)
par(mar = c(3,5,0,1), oma = c(2, 1, 1, 1)) #Change figure boundaries

barplot(t(t(PactFeatureOverlapsPercents[,c(1,13,9,5)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("All", "WGBS", "RRBS", "MBD-BS"), cex.names = 1.2,
        ylim = c(0, 115)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs. 

for (i in 1:4) {
    xx <- t(t(PactFeatureOverlapsPercents[,c(1,13,9,5)]))
    xx[,-i] <- NA
    colnames(xx)[-i] <- NA
    barplot(xx, col = c(plotColors2[(4*i)-3], plotColors2[(4*i)-2], plotColors2[(4*i)-1], plotColors2[(4*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "Proportion CpGs", line = 3, cex = 1.5) #Add y-axis label
mtext(side = 1, "Sequencing Method", line = 3, cex = 1.5) #Add x-axis label

#dev.off() #Turn off plotting device
```

#### CpG feature overlap by methylation status

```{r}
#pdf("../Output/Pact-CpG-Features-MethStatus.pdf", height = 8.5, width = 11)
par(mar = c(1,5,0,1), oma = c(4, 1, 1, 1), mfcol = c(3,1)) #Change figure boundaries

#Methylated CpGs

barplot(t(t(PactFeatureOverlapsPercents[,c(10,6,2)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("", "", ""),
        ylim = c(0, 115)) #Create base plot. Everything should be white. Do not plot axes. Set y-axis specs.

for (i in 2:4) {
    xx <- t(t(PactFeatureOverlapsPercents[,c(10,6,2)]))
    xx[,-(i-1)] <- NA
    colnames(xx)[-(i-1)] <- NA
    barplot(xx, col = c(plotColors2[(4*i)-3], plotColors2[(4*i)-2], plotColors2[(4*i)-1], plotColors2[(4*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% Methylated", line = 3, cex = 1.5) #Add y-axis label

#Sparsely methylated CpGs

barplot(t(t(PactFeatureOverlapsPercents[,c(11,7,3)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("", "", ""),
        ylim = c(0, 115)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs.

for (i in 2:4) {
    xx <- t(t(PactFeatureOverlapsPercents[,c(11,7,3)]))
    xx[,-(i-1)] <- NA
    colnames(xx)[-(i-1)] <- NA
    barplot(xx, col = c(plotColors2[(4*i)-3], plotColors2[(4*i)-2], plotColors2[(4*i)-1], plotColors2[(4*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% Sparsely Methylated", line = 3, cex = 1.5) #Add y-axis label

#Unmethylated CpGs

barplot(t(t(PactFeatureOverlapsPercents[,c(12,8,4)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("WGBS", "RRBS", "MBD-BS"), cex.names = 2,
        ylim = c(0, 115)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs.

for (i in 2:4) {
    xx <- t(t(PactFeatureOverlapsPercents[,c(12,8,4)]))
    xx[,-(i-1)] <- NA
    colnames(xx)[-(i-1)] <- NA
    barplot(xx, col = c(plotColors2[(4*i)-3], plotColors2[(4*i)-2], plotColors2[(4*i)-1], plotColors2[(4*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% Unmethylated", line = 3, cex = 1.5) #Add y-axis label

#dev.off() #Turn off plotting device
```


