---
title: "Chacterizing CpG Methylation"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this script, I'll create summary tables and figures to characterize CpG methylation in *M. capitata* and *P. acuta* using WGBS, RRBS, and MBD-BSSeq. I will use a list of CpGs with at least 5x coverage in any samples.

# Session information

```{r}
sessionInfo()
```

# Summary tables

## CpG distribution

### Mcap

#### Import file counts

```{r}
McapAll <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-5x-bedgraph-counts.txt", header = FALSE, col.names = c("totalLines", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
McapAll <- McapAll[-10,] #Remove last row (total lines for all files)
tail(McapAll) #Confirm import
```

```{r}
McapMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-5x-Meth-counts.txt", header = FALSE, col.names = c("Meth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
McapMeth <- McapMeth[-10,] #Remove last row (total lines for all files)
tail(McapMeth) #Confirm import
```

```{r}
McapSparseMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-5x-sparseMeth-counts.txt", header = FALSE, col.names = c("sparseMeth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
McapSparseMeth <- McapSparseMeth[-10,] #Remove last row (total lines for all files)
tail(McapSparseMeth) #Confirm import
```

```{r}
McapUnMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-5x-unMeth-counts.txt", header = FALSE, col.names = c("unMeth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
McapUnMeth <- McapUnMeth[-10,] #Remove last row (total lines for all files)
tail(McapUnMeth) #Confirm import
```

#### Create summary table

```{r}
McapCpGType <- cbind(McapAll, McapMeth, McapSparseMeth, McapUnMeth) #Mash tables together by column
McapCpGType <- McapCpGType[,-c(2,4,6,8)] #Remove filename columns
tail(McapCpGType) #Confirm table mashing
```

```{r}
McapCpGType$percentMeth <- (McapCpGType$Meth / McapCpGType$totalLines) * 100 #Calculate percent methylated loci
McapCpGType$percentSparseMeth <- (McapCpGType$sparseMeth / McapCpGType$totalLines) * 100 #Calculate percent sparsely methylated loci
McapCpGType$percentUnMeth <- (McapCpGType$unMeth / McapCpGType$totalLines) * 100 #Calculate percent unmethylated loci
McapCpGType <- McapCpGType[,c(1, 2, 5, 3, 6, 4, 7)] #Reorganize columns
tail(McapCpGType) #Confirm calculations
```

```{r}
write.table(McapCpGType, "../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-CpG-Type.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

### Pact

#### Import file counts

```{r}
PactAll <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Pact/Pact-5x-bedgraph-counts.txt", header = FALSE, col.names = c("totalLines", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
PactAll <- PactAll[-10,] #Remove last row (total lines for all files)
tail(PactAll) #Confirm import
```

```{r}
PactMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Pact/Pact-5x-Meth-counts.txt", header = FALSE, col.names = c("Meth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
PactMeth <- PactMeth[-10,] #Remove last row (total lines for all files)
tail(PactMeth) #Confirm import
```

```{r}
PactSparseMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Pact/Pact-5x-sparseMeth-counts.txt", header = FALSE, col.names = c("sparseMeth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
PactSparseMeth <- PactSparseMeth[-10,] #Remove last row (total lines for all files)
tail(PactSparseMeth) #Confirm import
```

```{r}
PactUnMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Pact/Pact-5x-unMeth-counts.txt", header = FALSE, col.names = c("unMeth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
PactUnMeth <- PactUnMeth[-10,] #Remove last row (total lines for all files)
tail(PactUnMeth) #Confirm import
```

#### Create summary table

```{r}
PactCpGType <- cbind(PactAll, PactMeth, PactSparseMeth, PactUnMeth) #Mash tables together by column
PactCpGType <- PactCpGType[,-c(2,4,6,8)] #Remove filename columns
tail(PactCpGType) #Confirm table mashing
```

```{r}
PactCpGType$percentMeth <- (PactCpGType$Meth / PactCpGType$totalLines) * 100 #Calculate percent methylated loci
PactCpGType$percentSparseMeth <- (PactCpGType$sparseMeth / PactCpGType$totalLines) * 100 #Calculate percent sparsely methylated loci
PactCpGType$percentUnMeth <- (PactCpGType$unMeth / PactCpGType$totalLines) * 100 #Calculate percent unmethylated loci
PactCpGType <- PactCpGType[,c(1, 2, 5, 3, 6, 4, 7)] #Reorganize columns
tail(PactCpGType) #Confirm calculations
```

```{r}
write.table(PactCpGType, "../analyses/Characterizing-CpG-Methylation-5x/Pact/Pact-CpG-Type.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

## CpG overlaps

### Mcap

#### Genes

##### Import file counts

```{r}
McapGeneOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-5x-mcGenes-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-gene overlaps
McapGeneOverlaps <- McapGeneOverlaps[-37,] #Remove final row
tail(McapGeneOverlaps) #Confirm import
```

##### Transfer values to summary table

```{r}
McapGeneSummary <- data.frame("totalLines" = rep(0, times = 9),
                              "Meth" = rep(0, times = 9),
                              "sparseMeth" = rep(0, times = 9),
                              "unMeth" = rep(0, times = 9)) #Create empty dataframe for final summary table
head(McapGeneSummary) #Confirm dataframe creation
```

```{r}
nLines <- length(McapGeneSummary$totalLines) #Count number of lines
for(i in 1:nLines) {
  McapGeneSummary[i,1] <- McapGeneOverlaps[i+(3*i),1]
} #For every file with the total number of CpG with data (i + (3*i)), take the value in the first column (total lines) and save it in the first column
head(McapGeneSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapGeneSummary[i,2] <- McapGeneOverlaps[(i+(3*i)-3),1]
} #For every file with the total number of CpG with data ((i + (3*i))-3), take the value in the first column (total lines) and save it in the second column
head(McapGeneSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapGeneSummary[i,3] <- McapGeneOverlaps[(i+(3*i)-2),1]
} #For every file with the total number of CpG with data ((i + (3*i))-2), take the value in the first column (total lines) and save it in the third column
head(McapGeneSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapGeneSummary[i,4] <- McapGeneOverlaps[(i+(3*i)-1),1]
} #For every file with the total number of CpG with data ((i + (3*i))-1), take the value in the first column (total lines) and save it in the fourth column
head(McapGeneSummary) #Confirm values were added
```

##### Calculate percents

```{r}
McapGeneSummary$percentMeth <- (McapGeneSummary$Meth / McapGeneSummary$totalLines) * 100 #Calculate percent methylated
McapGeneSummary$percentSparseMeth <- (McapGeneSummary$sparseMeth / McapGeneSummary$totalLines) * 100 #Calculate percent sparsely methylated
McapGeneSummary$percentUnMeth <- (McapGeneSummary$unMeth / McapGeneSummary$totalLines) * 100 #Calculate percent unmethylated
McapGeneSummary <- McapGeneSummary[,c(1,2,5,3,6,4,7)] #Rearrange columns
tail(McapGeneSummary) #Confirm calculations
```

```{r}
write.table(McapGeneSummary, "../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-Gene-Overlaps.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

#### CDS

##### Import file counts

```{r}
McapCDSOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-5x-mcCDS-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-CDS overlaps
McapCDSOverlaps <- McapCDSOverlaps[-37,] #Remove final row
tail(McapCDSOverlaps) #Confirm import
```

##### Transfer values to summary table

```{r}
McapCDSSummary <- data.frame("totalLines" = rep(0, times = 9),
                              "Meth" = rep(0, times = 9),
                              "sparseMeth" = rep(0, times = 9),
                              "unMeth" = rep(0, times = 9)) #Create empty dataframe for final summary table
head(McapCDSSummary) #Confirm dataframe creation
```

```{r}
nLines <- length(McapCDSSummary$totalLines) #Count number of lines
for(i in 1:nLines) {
  McapCDSSummary[i,1] <- McapCDSOverlaps[i+(3*i),1]
} #For every file with the total number of CpG with data (i + (3*i)), take the value in the first column (total lines) and save it in the first column
head(McapCDSSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapCDSSummary[i,2] <- McapCDSOverlaps[(i+(3*i)-3),1]
} #For every file with the total number of CpG with data ((i + (3*i))-3), take the value in the first column (total lines) and save it in the second column
head(McapCDSSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapCDSSummary[i,3] <- McapCDSOverlaps[(i+(3*i)-2),1]
} #For every file with the total number of CpG with data ((i + (3*i))-2), take the value in the first column (total lines) and save it in the third column
head(McapCDSSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapCDSSummary[i,4] <- McapCDSOverlaps[(i+(3*i)-1),1]
} #For every file with the total number of CpG with data ((i + (3*i))-1), take the value in the first column (total lines) and save it in the fourth column
head(McapCDSSummary) #Confirm values were added
```

##### Calculate percents

```{r}
McapCDSSummary$percentMeth <- (McapCDSSummary$Meth / McapCDSSummary$totalLines) * 100 #Calculate percent methylated
McapCDSSummary$percentSparseMeth <- (McapCDSSummary$sparseMeth / McapCDSSummary$totalLines) * 100 #Calculate percent sparsely methylated
McapCDSSummary$percentUnMeth <- (McapCDSSummary$unMeth / McapCDSSummary$totalLines) * 100 #Calculate percent unmethylated
McapCDSSummary <- McapCDSSummary[,c(1,2,5,3,6,4,7)] #Rearrange columns
tail(McapCDSSummary) #Confirm calculations
```

```{r}
write.table(McapCDSSummary, "../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-CDS-Overlaps.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

#### Introns

##### Import file counts

```{r}
McapIntronsOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-5x-mcIntrons-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
McapIntronsOverlaps <- McapIntronsOverlaps[-37,] #Remove final row
tail(McapIntronsOverlaps) #Confirm import
```

##### Transfer values to summary table

```{r}
McapIntronsSummary <- data.frame("totalLines" = rep(0, times = 9),
                              "Meth" = rep(0, times = 9),
                              "sparseMeth" = rep(0, times = 9),
                              "unMeth" = rep(0, times = 9)) #Create empty dataframe for final summary table
head(McapIntronsSummary) #Confirm dataframe creation
```

```{r}
nLines <- length(McapIntronsSummary$totalLines) #Count number of lines
for(i in 1:nLines) {
  McapIntronsSummary[i,1] <- McapIntronsOverlaps[i+(3*i),1]
} #For every file with the total number of CpG with data (i + (3*i)), take the value in the first column (total lines) and save it in the first column
head(McapIntronsSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapIntronsSummary[i,2] <- McapIntronsOverlaps[(i+(3*i)-3),1]
} #For every file with the total number of CpG with data ((i + (3*i))-3), take the value in the first column (total lines) and save it in the second column
head(McapIntronsSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapIntronsSummary[i,3] <- McapIntronsOverlaps[(i+(3*i)-2),1]
} #For every file with the total number of CpG with data ((i + (3*i))-2), take the value in the first column (total lines) and save it in the third column
head(McapIntronsSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapIntronsSummary[i,4] <- McapIntronsOverlaps[(i+(3*i)-1),1]
} #For every file with the total number of CpG with data ((i + (3*i))-1), take the value in the first column (total lines) and save it in the fourth column
head(McapIntronsSummary) #Confirm values were added
```

##### Calculate percents

```{r}
McapIntronsSummary$percentMeth <- (McapIntronsSummary$Meth / McapIntronsSummary$totalLines) * 100 #Calculate percent methylated
McapIntronsSummary$percentSparseMeth <- (McapIntronsSummary$sparseMeth / McapIntronsSummary$totalLines) * 100 #Calculate percent sparsely methylated
McapIntronsSummary$percentUnMeth <- (McapIntronsSummary$unMeth / McapIntronsSummary$totalLines) * 100 #Calculate percent unmethylated
McapIntronsSummary <- McapIntronsSummary[,c(1,2,5,3,6,4,7)] #Rearrange columns
tail(McapIntronsSummary) #Confirm calculations
```

```{r}
write.table(McapIntronsSummary, "../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-Introns-Overlaps.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

#### Intergenic regions

##### Import file counts

```{r}
McapIntergenicOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-5x-mcIntergenic-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Intergenic overlaps
McapIntergenicOverlaps <- McapIntergenicOverlaps[-37,] #Remove final row
tail(McapIntergenicOverlaps) #Confirm import
```

##### Transfer values to summary table

```{r}
McapIntergenicSummary <- data.frame("totalLines" = rep(0, times = 9),
                              "Meth" = rep(0, times = 9),
                              "sparseMeth" = rep(0, times = 9),
                              "unMeth" = rep(0, times = 9)) #Create empty dataframe for final summary table
head(McapIntergenicSummary) #Confirm dataframe creation
```

```{r}
nLines <- length(McapIntergenicSummary$totalLines) #Count number of lines
for(i in 1:nLines) {
  McapIntergenicSummary[i,1] <- McapIntergenicOverlaps[i+(3*i),1]
} #For every file with the total number of CpG with data (i + (3*i)), take the value in the first column (total lines) and save it in the first column
head(McapIntergenicSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapIntergenicSummary[i,2] <- McapIntergenicOverlaps[(i+(3*i)-3),1]
} #For every file with the total number of CpG with data ((i + (3*i))-3), take the value in the first column (total lines) and save it in the second column
head(McapIntergenicSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapIntergenicSummary[i,3] <- McapIntergenicOverlaps[(i+(3*i)-2),1]
} #For every file with the total number of CpG with data ((i + (3*i))-2), take the value in the first column (total lines) and save it in the third column
head(McapIntergenicSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapIntergenicSummary[i,4] <- McapIntergenicOverlaps[(i+(3*i)-1),1]
} #For every file with the total number of CpG with data ((i + (3*i))-1), take the value in the first column (total lines) and save it in the fourth column
head(McapIntergenicSummary) #Confirm values were added
```

##### Calculate percents

```{r}
McapIntergenicSummary$percentMeth <- (McapIntergenicSummary$Meth / McapIntergenicSummary$totalLines) * 100 #Calculate percent methylated
McapIntergenicSummary$percentSparseMeth <- (McapIntergenicSummary$sparseMeth / McapIntergenicSummary$totalLines) * 100 #Calculate percent sparsely methylated
McapIntergenicSummary$percentUnMeth <- (McapIntergenicSummary$unMeth / McapIntergenicSummary$totalLines) * 100 #Calculate percent unmethylated
McapIntergenicSummary <- McapIntergenicSummary[,c(1,2,5,3,6,4,7)] #Rearrange columns
tail(McapIntergenicSummary) #Confirm calculations
```

```{r}
write.table(McapIntergenicSummary, "../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-Intergenic-Overlaps.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

### Pact

#### Genes

##### Import file counts

```{r}
McapGeneOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-5x-mcGenes-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-gene overlaps
McapGeneOverlaps <- McapGeneOverlaps[-37,] #Remove final row
tail(McapGeneOverlaps) #Confirm import
```

##### Transfer values to summary table

```{r}
McapGeneSummary <- data.frame("totalLines" = rep(0, times = 9),
                              "Meth" = rep(0, times = 9),
                              "sparseMeth" = rep(0, times = 9),
                              "unMeth" = rep(0, times = 9)) #Create empty dataframe for final summary table
head(McapGeneSummary) #Confirm dataframe creation
```

```{r}
nLines <- length(McapGeneSummary$totalLines) #Count number of lines
for(i in 1:nLines) {
  McapGeneSummary[i,1] <- McapGeneOverlaps[i+(3*i),1]
} #For every file with the total number of CpG with data (i + (3*i)), take the value in the first column (total lines) and save it in the first column
head(McapGeneSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapGeneSummary[i,2] <- McapGeneOverlaps[(i+(3*i)-3),1]
} #For every file with the total number of CpG with data ((i + (3*i))-3), take the value in the first column (total lines) and save it in the second column
head(McapGeneSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapGeneSummary[i,3] <- McapGeneOverlaps[(i+(3*i)-2),1]
} #For every file with the total number of CpG with data ((i + (3*i))-2), take the value in the first column (total lines) and save it in the third column
head(McapGeneSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapGeneSummary[i,4] <- McapGeneOverlaps[(i+(3*i)-1),1]
} #For every file with the total number of CpG with data ((i + (3*i))-1), take the value in the first column (total lines) and save it in the fourth column
head(McapGeneSummary) #Confirm values were added
```

##### Calculate percents

```{r}
McapGeneSummary$percentMeth <- (McapGeneSummary$Meth / McapGeneSummary$totalLines) * 100 #Calculate percent methylated
McapGeneSummary$percentSparseMeth <- (McapGeneSummary$sparseMeth / McapGeneSummary$totalLines) * 100 #Calculate percent sparsely methylated
McapGeneSummary$percentUnMeth <- (McapGeneSummary$unMeth / McapGeneSummary$totalLines) * 100 #Calculate percent unmethylated
McapGeneSummary <- McapGeneSummary[,c(1,2,5,3,6,4,7)] #Rearrange columns
tail(McapGeneSummary) #Confirm calculations
```

```{r}
write.table(McapGeneSummary, "../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-Gene-Overlaps.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

#### CDS

##### Import file counts

```{r}
McapCDSOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-5x-mcCDS-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-CDS overlaps
McapCDSOverlaps <- McapCDSOverlaps[-37,] #Remove final row
tail(McapCDSOverlaps) #Confirm import
```

##### Transfer values to summary table

```{r}
McapCDSSummary <- data.frame("totalLines" = rep(0, times = 9),
                              "Meth" = rep(0, times = 9),
                              "sparseMeth" = rep(0, times = 9),
                              "unMeth" = rep(0, times = 9)) #Create empty dataframe for final summary table
head(McapCDSSummary) #Confirm dataframe creation
```

```{r}
nLines <- length(McapCDSSummary$totalLines) #Count number of lines
for(i in 1:nLines) {
  McapCDSSummary[i,1] <- McapCDSOverlaps[i+(3*i),1]
} #For every file with the total number of CpG with data (i + (3*i)), take the value in the first column (total lines) and save it in the first column
head(McapCDSSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapCDSSummary[i,2] <- McapCDSOverlaps[(i+(3*i)-3),1]
} #For every file with the total number of CpG with data ((i + (3*i))-3), take the value in the first column (total lines) and save it in the second column
head(McapCDSSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapCDSSummary[i,3] <- McapCDSOverlaps[(i+(3*i)-2),1]
} #For every file with the total number of CpG with data ((i + (3*i))-2), take the value in the first column (total lines) and save it in the third column
head(McapCDSSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapCDSSummary[i,4] <- McapCDSOverlaps[(i+(3*i)-1),1]
} #For every file with the total number of CpG with data ((i + (3*i))-1), take the value in the first column (total lines) and save it in the fourth column
head(McapCDSSummary) #Confirm values were added
```

##### Calculate percents

```{r}
McapCDSSummary$percentMeth <- (McapCDSSummary$Meth / McapCDSSummary$totalLines) * 100 #Calculate percent methylated
McapCDSSummary$percentSparseMeth <- (McapCDSSummary$sparseMeth / McapCDSSummary$totalLines) * 100 #Calculate percent sparsely methylated
McapCDSSummary$percentUnMeth <- (McapCDSSummary$unMeth / McapCDSSummary$totalLines) * 100 #Calculate percent unmethylated
McapCDSSummary <- McapCDSSummary[,c(1,2,5,3,6,4,7)] #Rearrange columns
tail(McapCDSSummary) #Confirm calculations
```

```{r}
write.table(McapCDSSummary, "../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-CDS-Overlaps.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

#### Introns

##### Import file counts

```{r}
McapIntronsOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-5x-mcIntrons-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
McapIntronsOverlaps <- McapIntronsOverlaps[-37,] #Remove final row
tail(McapIntronsOverlaps) #Confirm import
```

##### Transfer values to summary table

```{r}
McapIntronsSummary <- data.frame("totalLines" = rep(0, times = 9),
                              "Meth" = rep(0, times = 9),
                              "sparseMeth" = rep(0, times = 9),
                              "unMeth" = rep(0, times = 9)) #Create empty dataframe for final summary table
head(McapIntronsSummary) #Confirm dataframe creation
```

```{r}
nLines <- length(McapIntronsSummary$totalLines) #Count number of lines
for(i in 1:nLines) {
  McapIntronsSummary[i,1] <- McapIntronsOverlaps[i+(3*i),1]
} #For every file with the total number of CpG with data (i + (3*i)), take the value in the first column (total lines) and save it in the first column
head(McapIntronsSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapIntronsSummary[i,2] <- McapIntronsOverlaps[(i+(3*i)-3),1]
} #For every file with the total number of CpG with data ((i + (3*i))-3), take the value in the first column (total lines) and save it in the second column
head(McapIntronsSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapIntronsSummary[i,3] <- McapIntronsOverlaps[(i+(3*i)-2),1]
} #For every file with the total number of CpG with data ((i + (3*i))-2), take the value in the first column (total lines) and save it in the third column
head(McapIntronsSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapIntronsSummary[i,4] <- McapIntronsOverlaps[(i+(3*i)-1),1]
} #For every file with the total number of CpG with data ((i + (3*i))-1), take the value in the first column (total lines) and save it in the fourth column
head(McapIntronsSummary) #Confirm values were added
```

##### Calculate percents

```{r}
McapIntronsSummary$percentMeth <- (McapIntronsSummary$Meth / McapIntronsSummary$totalLines) * 100 #Calculate percent methylated
McapIntronsSummary$percentSparseMeth <- (McapIntronsSummary$sparseMeth / McapIntronsSummary$totalLines) * 100 #Calculate percent sparsely methylated
McapIntronsSummary$percentUnMeth <- (McapIntronsSummary$unMeth / McapIntronsSummary$totalLines) * 100 #Calculate percent unmethylated
McapIntronsSummary <- McapIntronsSummary[,c(1,2,5,3,6,4,7)] #Rearrange columns
tail(McapIntronsSummary) #Confirm calculations
```

```{r}
write.table(McapIntronsSummary, "../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-Introns-Overlaps.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

#### Intergenic regions

##### Import file counts

```{r}
McapIntergenicOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-5x-mcIntergenic-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Intergenic overlaps
McapIntergenicOverlaps <- McapIntergenicOverlaps[-37,] #Remove final row
tail(McapIntergenicOverlaps) #Confirm import
```

##### Transfer values to summary table

```{r}
McapIntergenicSummary <- data.frame("totalLines" = rep(0, times = 9),
                              "Meth" = rep(0, times = 9),
                              "sparseMeth" = rep(0, times = 9),
                              "unMeth" = rep(0, times = 9)) #Create empty dataframe for final summary table
head(McapIntergenicSummary) #Confirm dataframe creation
```

```{r}
nLines <- length(McapIntergenicSummary$totalLines) #Count number of lines
for(i in 1:nLines) {
  McapIntergenicSummary[i,1] <- McapIntergenicOverlaps[i+(3*i),1]
} #For every file with the total number of CpG with data (i + (3*i)), take the value in the first column (total lines) and save it in the first column
head(McapIntergenicSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapIntergenicSummary[i,2] <- McapIntergenicOverlaps[(i+(3*i)-3),1]
} #For every file with the total number of CpG with data ((i + (3*i))-3), take the value in the first column (total lines) and save it in the second column
head(McapIntergenicSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapIntergenicSummary[i,3] <- McapIntergenicOverlaps[(i+(3*i)-2),1]
} #For every file with the total number of CpG with data ((i + (3*i))-2), take the value in the first column (total lines) and save it in the third column
head(McapIntergenicSummary) #Confirm values were added
```

```{r}
for(i in 1:nLines) {
  McapIntergenicSummary[i,4] <- McapIntergenicOverlaps[(i+(3*i)-1),1]
} #For every file with the total number of CpG with data ((i + (3*i))-1), take the value in the first column (total lines) and save it in the fourth column
head(McapIntergenicSummary) #Confirm values were added
```

##### Calculate percents

```{r}
McapIntergenicSummary$percentMeth <- (McapIntergenicSummary$Meth / McapIntergenicSummary$totalLines) * 100 #Calculate percent methylated
McapIntergenicSummary$percentSparseMeth <- (McapIntergenicSummary$sparseMeth / McapIntergenicSummary$totalLines) * 100 #Calculate percent sparsely methylated
McapIntergenicSummary$percentUnMeth <- (McapIntergenicSummary$unMeth / McapIntergenicSummary$totalLines) * 100 #Calculate percent unmethylated
McapIntergenicSummary <- McapIntergenicSummary[,c(1,2,5,3,6,4,7)] #Rearrange columns
tail(McapIntergenicSummary) #Confirm calculations
```

```{r}
write.table(McapIntergenicSummary, "../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-Intergenic-Overlaps.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

# CpG distributions

## Import data

### Mcap

#### Sample 1

```{r}
cpgMethylationWGBS1 <- read.delim("../analyses/Characterizing-CpG-Methylation/Meth10.fastp-trim.202003063900_R1_001_bismark_bt2_pe._10x.bedgraph-Mcap", header = FALSE) #Import file with CpG methylation for all loci with 10x coverage
colnames(cpgMethylationWGBS1) <- c("chromosome", "start", "end", "methylation") #Add column names
head(cpgMethylationWGBS1) #Confirm import
```

```{r}
cpgMethylationRRBS1 <- read.delim("../analyses/Characterizing-CpG-Methylation/Meth13.fastp-trim.202003062040_R1_001_bismark_bt2_pe_10x.bedgraph-Mcap", header = FALSE) #Import file with CpG methylation for all loci with 10x coverage
colnames(cpgMethylationRRBS1) <- c("chromosome", "start", "end", "methylation") #Add column names
head(cpgMethylationRRBS1) #Confirm import
```

```{r}
cpgMethylationMBD1 <- read.delim("../analyses/Characterizing-CpG-Methylation/Meth16.fastp-trim.202003062412_R1_001_bismark_bt2_pe._10x.bedgraph-Mcap", header = FALSE) #Import file with CpG methylation for all loci with 10x coverage
colnames(cpgMethylationMBD1) <- c("chromosome", "start", "end", "methylation") #Add column names
head(cpgMethylationMBD1) #Confirm import
```

#### Sample 2

```{r}
cpgMethylationWGBS2 <- read.delim("../analyses/Characterizing-CpG-Methylation/Meth11.fastp-trim.202003065734_R1_001_bismark_bt2_pe._10x.bedgraph-Mcap", header = FALSE) #Import file with CpG methylation for all loci with 10x coverage
colnames(cpgMethylationWGBS2) <- c("chromosome", "start", "end", "methylation") #Add column names
head(cpgMethylationWGBS2) #Confirm import
```

```{r}
cpgMethylationRRBS2 <- read.delim("../analyses/Characterizing-CpG-Methylation/Meth14.fastp-trim.202003064415_R1_001_bismark_bt2_pe_10x.bedgraph-Mcap", header = FALSE) #Import file with CpG methylation for all loci with 10x coverage
colnames(cpgMethylationRRBS2) <- c("chromosome", "start", "end", "methylation") #Add column names
head(cpgMethylationRRBS2) #Confirm import
```

```{r}
cpgMethylationMBD2 <- read.delim("../analyses/Characterizing-CpG-Methylation/Meth17.fastp-trim.202003063731_R1_001_bismark_bt2_pe._10x.bedgraph-Mcap", header = FALSE) #Import file with CpG methylation for all loci with 10x coverage
colnames(cpgMethylationMBD2) <- c("chromosome", "start", "end", "methylation") #Add column names
head(cpgMethylationMBD2) #Confirm import
```

#### Sample 3

```{r}
cpgMethylationWGBS3 <- read.delim("../analyses/Characterizing-CpG-Methylation/Meth12.fastp-trim.202003060645_R1_001_bismark_bt2_pe._10x.bedgraph-Mcap", header = FALSE) #Import file with CpG methylation for all loci with 10x coverage
colnames(cpgMethylationWGBS3) <- c("chromosome", "start", "end", "methylation") #Add column names
head(cpgMethylationWGBS3) #Confirm import
```

```{r}
cpgMethylationRRBS3 <- read.delim("../analyses/Characterizing-CpG-Methylation/Meth15.fastp-trim.202003065503_R1_001_bismark_bt2_pe_10x.bedgraph-Mcap", header = FALSE) #Import file with CpG methylation for all loci with 10x coverage
colnames(cpgMethylationRRBS3) <- c("chromosome", "start", "end", "methylation") #Add column names
head(cpgMethylationRRBS3) #Confirm import
```

```{r}
cpgMethylationMBD3 <- read.delim("../analyses/Characterizing-CpG-Methylation/Meth18.fastp-trim.202003065117_R1_001_bismark_bt2_pe._10x.bedgraph-Mcap", header = FALSE) #Import file with CpG methylation for all loci with 10x coverage
colnames(cpgMethylationMBD3) <- c("chromosome", "start", "end", "methylation") #Add column names
head(cpgMethylationMBD3) #Confirm import
```

## Create individual figures to test options

```{r}
hist(cpgMethylationWGBS1$methylation)
hist(cpgMethylationRRBS1$methylation)
hist(cpgMethylationMBD1$methylation)

hist(cpgMethylationWGBS2$methylation)
hist(cpgMethylationRRBS2$methylation)
hist(cpgMethylationMBD2$methylation)

hist(cpgMethylationWGBS3$methylation)
hist(cpgMethylationRRBS3$methylation)
hist(cpgMethylationMBD3$methylation)
```

```{r}
plot(density(cpgMethylationWGBS1$methylation))
plot(density(cpgMethylationRRBS1$methylation))
plot(density(cpgMethylationMBD1$methylation))

plot(density(cpgMethylationWGBS2$methylation))
plot(density(cpgMethylationRRBS2$methylation))
plot(density(cpgMethylationMBD2$methylation))

plot(density(cpgMethylationWGBS3$methylation))
plot(density(cpgMethylationRRBS3$methylation))
plot(density(cpgMethylationMBD3$methylation))
```

## Create multipanel figure with all samples

```{r}
par(mfrow = c(3,3))
hist(cpgMethylationWGBS1$methylation, main = "")
hist(cpgMethylationRRBS1$methylation, main = "")
hist(cpgMethylationMBD1$methylation, main = "")

hist(cpgMethylationWGBS2$methylation, main = "")
hist(cpgMethylationRRBS2$methylation, main = "")
hist(cpgMethylationMBD2$methylation, main = "")

hist(cpgMethylationWGBS3$methylation, main = "")
hist(cpgMethylationRRBS3$methylation, main = "")
hist(cpgMethylationMBD3$methylation, main = "")
```

