---
title: "Chacterizing CpG Methylation (5x Union Data)"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this script, I'll create summary tables and figures to characterize CpG methylation in *M. capitata* and *P. acuta* 5x union bedgraphs for WGBS, RRBS, and MBD-BS.

# Install packages

```{r}
#install.packages("RColorBrewer")
require(RColorBrewer) #Load RColorBrewer

#install.packages("dichromat")
require(dichromat)

#install.packages("broom")
require(broom)
```

# Session information

```{r}
sessionInfo()
```

# CpG distributions

## Mcap

### Import file counts

```{r}
McapAll <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-averages-counts.txt", header = FALSE, col.names = c("totalLines", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
McapAll <- McapAll[-4,] #Remove last row (total lines for all files)
tail(McapAll) #Confirm import
```

```{r}
McapMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-Meth-counts.txt", header = FALSE, col.names = c("Meth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
McapMeth <- McapMeth[-4,] #Remove last row (total lines for all files)
tail(McapMeth) #Confirm import
```

```{r}
McapSparseMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-sparseMeth-counts.txt", header = FALSE, col.names = c("sparseMeth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
McapSparseMeth <- McapSparseMeth[-4,] #Remove last row (total lines for all files)
tail(McapSparseMeth) #Confirm import
```

```{r}
McapUnMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-unMeth-counts.txt", header = FALSE, col.names = c("unMeth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
McapUnMeth <- McapUnMeth[-4,] #Remove last row (total lines for all files)
tail(McapUnMeth) #Confirm import
```

### Create summary table

```{r}
McapCpGType <- cbind(McapAll, McapMeth, McapSparseMeth, McapUnMeth) #Mash tables together by column
McapCpGType <- McapCpGType[,-c(2,4,6,8)] #Remove filename columns
rownames(McapCpGType) <- c("MBDBS", "RRBS", "WGBS") #Add rownames
McapCpGType <- McapCpGType[c(3,2,1),] #Order rows: WGBS, RRBS, MBDBS
tail(McapCpGType) #Confirm table mashing
```

```{r}
McapCpGType$percentMeth <- (McapCpGType$Meth / McapCpGType$totalLines) * 100 #Calculate percent methylated loci
McapCpGType$percentSparseMeth <- (McapCpGType$sparseMeth / McapCpGType$totalLines) * 100 #Calculate percent sparsely methylated loci
McapCpGType$percentUnMeth <- (McapCpGType$unMeth / McapCpGType$totalLines) * 100 #Calculate percent unmethylated loci
McapCpGType <- McapCpGType[,c(1, 2, 5, 3, 6, 4, 7)] #Reorganize columns
tail(McapCpGType) #Confirm calculations
```

```{r}
write.table(McapCpGType, "../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union-CpG-Type.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

### Conduct chi-squared tests of homogeneity

The null hypothesis is that distributions of different CpG types is the same between sequencing methods

```{r}
McapCpGTypeStatTest <- McapCpGType[,c(2,4,6)] #Separate out count data for statistical testing
head(McapCpGTypeStatTest) #Confirm changes
```

#### WGBS vs. RRBS

```{r}
McapCpGTypeWGRR <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(McapCpGTypeStatTest)) { #For each CpG type
  Method1CpG <- McapCpGTypeStatTest[1,i] #Variable for # CpG type for method 1
  Method2CpG <- McapCpGTypeStatTest[2,i] #Variable for # CpG type for method 2
  Method1NotCpG <- sum(McapCpGTypeStatTest[1,-i]) #Variable for # other CpG types for method 1
  Method2NotCpG <- sum(McapCpGTypeStatTest[2,-i]) #Variable for # other CpG types for method 2
  ct <- matrix(c(Method1CpG,Method2CpG,Method1NotCpG,Method2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(McapCpGTypeStatTest[i])), paste0("Not", colnames(McapCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c(as.character(row.names(McapCpGTypeStatTest)[1:2])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(McapCpGTypeStatTest)[i]) #Add CpG type to results
  McapCpGTypeWGRR <- rbind(McapCpGTypeWGRR, ctResults) #Add test statistics to master table
}
```

```{r}
McapCpGTypeWGRR$p.adj <- p.adjust(McapCpGTypeWGRR$p.value, method = "fdr") #Correct p-value using FDR
McapCpGTypeWGRR$comparison <- rep("WGBS vs. RRBS", times = 3) #Add methods compared
head(McapCpGTypeWGRR) #Confirm changes
```

#### WGBS vs. MBD-BS

```{r}
McapCpGTypeWGMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(McapCpGTypeStatTest)) { #For each CpG type
  Method1CpG <- McapCpGTypeStatTest[1,i] #Variable for # CpG type for method 1
  Method2CpG <- McapCpGTypeStatTest[3,i] #Variable for # CpG type for method 2
  Method1NotCpG <- sum(McapCpGTypeStatTest[1,-i]) #Variable for # other CpG types for method 1
  Method2NotCpG <- sum(McapCpGTypeStatTest[3,-i]) #Variable for # other CpG types for method 2
  ct <- matrix(c(Method1CpG,Method2CpG,Method1NotCpG,Method2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(McapCpGTypeStatTest[i])), paste0("Not", colnames(McapCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c(as.character(row.names(McapCpGTypeStatTest)[c(1,3)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(McapCpGTypeStatTest)[i]) #Add CpG type to results
  McapCpGTypeWGMB <- rbind(McapCpGTypeWGMB, ctResults) #Add test statistics to master table
}
```

```{r}
McapCpGTypeWGMB$p.adj <- p.adjust(McapCpGTypeWGMB$p.value, method = "fdr") #Correct p-value using FDR
McapCpGTypeWGMB$comparison <- rep("WGBS vs. MBDBS", times = 3) #Add methods compared
head(McapCpGTypeWGMB) #Confirm changes
```

#### RRBS vs. MBD-BS

```{r}
McapCpGTypeRRMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(McapCpGTypeStatTest)) { #For each CpG type
  Method1CpG <- McapCpGTypeStatTest[2,i] #Variable for # CpG type for method 1
  Method2CpG <- McapCpGTypeStatTest[3,i] #Variable for # CpG type for method 2
  Method1NotCpG <- sum(McapCpGTypeStatTest[2,-i]) #Variable for # other CpG types for method 1
  Method2NotCpG <- sum(McapCpGTypeStatTest[3,-i]) #Variable for # other CpG types for method 2
  ct <- matrix(c(Method1CpG,Method2CpG,Method1NotCpG,Method2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(McapCpGTypeStatTest[i])), paste0("Not", colnames(McapCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c(as.character(row.names(McapCpGTypeStatTest)[c(2,3)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(McapCpGTypeStatTest)[i]) #Add CpG type to results
  McapCpGTypeRRMB <- rbind(McapCpGTypeRRMB, ctResults) #Add test statistics to master table
}
```

```{r}
McapCpGTypeRRMB$p.adj <- p.adjust(McapCpGTypeRRMB$p.value, method = "fdr") #Correct p-value using FDR
McapCpGTypeRRMB$comparison <- rep("RRBS vs. MBDBS", times = 3) #Add methods compared
head(McapCpGTypeRRMB) #Confirm changes
```

#### Save statistical test results

```{r}
McapCpGTypeStatResults <- rbind(McapCpGTypeWGRR,
                                McapCpGTypeWGMB,
                                McapCpGTypeRRMB) #Combine tables
tail(McapCpGTypeStatResults) #Confirm changes
```

```{r}
write.table(McapCpGTypeStatResults, "../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union-CpG-Type-StatResults.txt", quote = FALSE, row.names = FALSE) #Save table
```

### Create plot

```{r}
McapCpGTypePercents <- McapCpGType[,c(3,5,7)] #Keep only percent information
head(McapCpGTypePercents) #Confirm changes
```

```{r}
plotColors <- c(rev(RColorBrewer::brewer.pal(3, "Greens")), 
                rev(RColorBrewer::brewer.pal(3, "Blues")), 
                rev(RColorBrewer::brewer.pal(3, "Reds"))) #Create vector of WGBS, RRBS, and MBD colors
barplot(t(McapCpGTypePercents), col = dichromat(c(rev(RColorBrewer::brewer.pal(3, "Greens"))))) #Check greens
barplot(t(McapCpGTypePercents), col = dichromat(c(rev(RColorBrewer::brewer.pal(3, "Blues"))))) #Check blues
barplot(t(McapCpGTypePercents), col = dichromat(c(rev(RColorBrewer::brewer.pal(3, "Reds"))))) #Check reds
```

```{r}
#pdf("../Output/Mcap_union-CpG-Type.pdf", height = 8.5, width = 11)
par(mar = c(2,5,0,1), oma = c(1, 1, 0, 12)) #Change figure boundaries

barplot(t(McapCpGTypePercents), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("WGBS", "RRBS", "MBD-BS"), cex.names = 1.5,
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs. 

for (i in 1:ncol(t(McapCpGTypePercents))){
    xx <- t(McapCpGTypePercents)
    xx[,-i] <- NA
    colnames(xx)[-i] <- NA
    barplot(xx, col = c(plotColors[(3*i)-2], plotColors[(3*i)-1], plotColors[(3*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% CpGs", line = 3, cex = 1.5) #Add y-axis label

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE) #Create new plot
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n") #Add new plot on top of current plot
legend(x = 0.54, y = 0.87, 
       xpd = TRUE,
       legend = c(expression("High (">="50%)"), "Moderate (10-50%)", expression("Weak ("<="10%)")),
       pch = 22, 
        col = "black", 
        pt.bg = c("grey20", "grey50", "grey80"),
       bty = "n",
       cex = 1.5, 
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box
text("Methylation Status", x = 0.77, y = 0.879, cex = 1.5) #Add legend title that is aligned with legend

#dev.off() #Turn off plotting device
```

## Pact

### Import file counts

```{r}
PactAll <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-averages-counts.txt", header = FALSE, col.names = c("totalLines", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
PactAll <- PactAll[-4,] #Remove last row (total lines for all files)
tail(PactAll) #Confirm import
```

```{r}
PactMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-Meth-counts.txt", header = FALSE, col.names = c("Meth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
PactMeth <- PactMeth[-4,] #Remove last row (total lines for all files)
tail(PactMeth) #Confirm import
```

```{r}
PactSparseMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-sparseMeth-counts.txt", header = FALSE, col.names = c("sparseMeth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
PactSparseMeth <- PactSparseMeth[-4,] #Remove last row (total lines for all files)
tail(PactSparseMeth) #Confirm import
```

```{r}
PactUnMeth <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-unMeth-counts.txt", header = FALSE, col.names = c("unMeth", "filename")) #Import file using space as a delimiter. Columns are the number of lines and the filename
PactUnMeth <- PactUnMeth[-4,] #Remove last row (total lines for all files)
tail(PactUnMeth) #Confirm import
```

### Create summary table

```{r}
PactCpGType <- cbind(PactAll, PactMeth, PactSparseMeth, PactUnMeth) #Mash tables together by column
PactCpGType <- PactCpGType[,-c(2,4,6,8)] #Remove filename columns
rownames(PactCpGType) <- c("MBDBS", "RRBS", "WGBS") #Add rownames
PactCpGType <- PactCpGType[c(3,2,1),] #Order rows: WGBS, RRBS, MBDBS
tail(PactCpGType) #Confirm table mashing
```

```{r}
PactCpGType$percentMeth <- (PactCpGType$Meth / PactCpGType$totalLines) * 100 #Calculate percent methylated loci
PactCpGType$percentSparseMeth <- (PactCpGType$sparseMeth / PactCpGType$totalLines) * 100 #Calculate percent sparsely methylated loci
PactCpGType$percentUnMeth <- (PactCpGType$unMeth / PactCpGType$totalLines) * 100 #Calculate percent unmethylated loci
PactCpGType <- PactCpGType[,c(1, 2, 5, 3, 6, 4, 7)] #Reorganize columns
tail(PactCpGType) #Confirm calculations
```

```{r}
write.table(PactCpGType, "../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-CpG-Type.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

### Conduct chi-squared tests of homogeneity

The null hypothesis is that distributions of different CpG types is the same between sequencing methods

```{r}
PactCpGTypeStatTest <- PactCpGType[,c(2,4,6)] #Separate out count data for statistical testing
head(PactCpGTypeStatTest) #Confirm changes
```

#### WGBS vs. RRBS

```{r}
PactCpGTypeWGRR <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactCpGTypeStatTest)) { #For each CpG type
  Method1CpG <- PactCpGTypeStatTest[1,i] #Variable for # CpG type for method 1
  Method2CpG <- PactCpGTypeStatTest[2,i] #Variable for # CpG type for method 2
  Method1NotCpG <- sum(PactCpGTypeStatTest[1,-i]) #Variable for # other CpG types for method 1
  Method2NotCpG <- sum(PactCpGTypeStatTest[2,-i]) #Variable for # other CpG types for method 2
  ct <- matrix(c(Method1CpG,Method2CpG,Method1NotCpG,Method2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactCpGTypeStatTest[i])), paste0("Not", colnames(PactCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c(as.character(row.names(PactCpGTypeStatTest)[1:2])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(PactCpGTypeStatTest)[i]) #Add CpG type to results
  PactCpGTypeWGRR <- rbind(PactCpGTypeWGRR, ctResults) #Add test statistics to master table
}
```

```{r}
PactCpGTypeWGRR$p.adj <- p.adjust(PactCpGTypeWGRR$p.value, method = "fdr") #Correct p-value using FDR
PactCpGTypeWGRR$comparison <- rep("WGBS vs. RRBS", times = 3) #Add methods compared
head(PactCpGTypeWGRR) #Confirm changes
```

#### WGBS vs. MBD-BS

```{r}
PactCpGTypeWGMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactCpGTypeStatTest)) { #For each CpG type
  Method1CpG <- PactCpGTypeStatTest[1,i] #Variable for # CpG type for method 1
  Method2CpG <- PactCpGTypeStatTest[3,i] #Variable for # CpG type for method 2
  Method1NotCpG <- sum(PactCpGTypeStatTest[1,-i]) #Variable for # other CpG types for method 1
  Method2NotCpG <- sum(PactCpGTypeStatTest[3,-i]) #Variable for # other CpG types for method 2
  ct <- matrix(c(Method1CpG,Method2CpG,Method1NotCpG,Method2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactCpGTypeStatTest[i])), paste0("Not", colnames(PactCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c(as.character(row.names(PactCpGTypeStatTest)[c(1,3)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(PactCpGTypeStatTest)[i]) #Add CpG type to results
  PactCpGTypeWGMB <- rbind(PactCpGTypeWGMB, ctResults) #Add test statistics to master table
}
```

```{r}
PactCpGTypeWGMB$p.adj <- p.adjust(PactCpGTypeWGMB$p.value, method = "fdr") #Correct p-value using FDR
PactCpGTypeWGMB$comparison <- rep("WGBS vs. MBDBS", times = 3) #Add methods compared
head(PactCpGTypeWGMB) #Confirm changes
```

#### RRBS vs. MBD-BS

```{r}
PactCpGTypeRRMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactCpGTypeStatTest)) { #For each CpG type
  Method1CpG <- PactCpGTypeStatTest[2,i] #Variable for # CpG type for method 1
  Method2CpG <- PactCpGTypeStatTest[3,i] #Variable for # CpG type for method 2
  Method1NotCpG <- sum(PactCpGTypeStatTest[2,-i]) #Variable for # other CpG types for method 1
  Method2NotCpG <- sum(PactCpGTypeStatTest[3,-i]) #Variable for # other CpG types for method 2
  ct <- matrix(c(Method1CpG,Method2CpG,Method1NotCpG,Method2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactCpGTypeStatTest[i])), paste0("Not", colnames(PactCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c(as.character(row.names(PactCpGTypeStatTest)[c(2,3)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(PactCpGTypeStatTest)[i]) #Add CpG type to results
  PactCpGTypeRRMB <- rbind(PactCpGTypeRRMB, ctResults) #Add test statistics to master table
}
```

```{r}
PactCpGTypeRRMB$p.adj <- p.adjust(PactCpGTypeRRMB$p.value, method = "fdr") #Correct p-value using FDR
PactCpGTypeRRMB$comparison <- rep("RRBS vs. MBDBS", times = 3) #Add methods compared
head(PactCpGTypeRRMB) #Confirm changes
```

#### Save statistical test results

```{r}
PactCpGTypeStatResults <- rbind(PactCpGTypeWGRR,
                                PactCpGTypeWGMB,
                                PactCpGTypeRRMB) #Combine tables
tail(PactCpGTypeStatResults) #Confirm changes
```

```{r}
write.table(PactCpGTypeStatResults, "../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union-CpG-Type-StatResults.txt", quote = FALSE, row.names = FALSE) #Save table
```

### Create plot

```{r}
PactCpGTypePercents <- PactCpGType[,c(3,5,7)] #Keep only percent information
head(PactCpGTypePercents) #Confirm changes
```

```{r}
#pdf("../Output/Pact_union-CpG-Type.pdf", height = 8.5, width = 11)
par(mar = c(2,5,0,1), oma = c(1, 1, 0, 12)) #Change figure boundaries

barplot(t(PactCpGTypePercents), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("WGBS", "RRBS", "MBD-BS"), cex.names = 1.5,
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs. 

for (i in 1:ncol(t(PactCpGTypePercents))){
    xx <- t(PactCpGTypePercents)
    xx[,-i] <- NA
    colnames(xx)[-i] <- NA
    barplot(xx, col = c(plotColors[(3*i)-2], plotColors[(3*i)-1], plotColors[(3*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% CpGs", line = 3, cex = 1.5) #Add y-axis label

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE) #Create new plot
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n") #Add new plot on top of current plot
legend(x = 0.54, y = 0.87, 
       xpd = TRUE,
       legend = c(expression("High (">="50%)"), "Moderate (10-50%)", expression("Weak ("<="10%)")),
       pch = 22, 
        col = "black", 
        pt.bg = c("grey20", "grey50", "grey80"),
       bty = "n",
       cex = 1.5, 
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box
text("Methylation Status", x = 0.77, y = 0.879, cex = 1.5) #Add legend title that is aligned with legend

#dev.off() #Turn off plotting device
```

## Interspecies chi-squared tests

The hypothesis is that the distribution of CpG type will be the same between species for the same methods used.

#### WGBS

```{r}
WGBSCpGType <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactCpGTypeStatTest)) { #For each CpG type
  Species1CpG <- McapCpGTypeStatTest[1,i] #Variable for # CpG type for Species 1
  Species2CpG <- PactCpGTypeStatTest[1,i] #Variable for # CpG type for Species 2
  Species1NotCpG <- sum(McapCpGTypeStatTest[1,-i]) #Variable for # other CpG types for Species 1
  Species2NotCpG <- sum(PactCpGTypeStatTest[1,-i]) #Variable for # other CpG types for Species 2
  ct <- matrix(c(Species1CpG,Species2CpG,Species1NotCpG,Species2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactCpGTypeStatTest[i])), paste0("Not", colnames(PactCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c("Mcap", "Pact") #Assign row names: Species 1, Species 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(PactCpGTypeStatTest)[i]) #Add CpG type to results
  WGBSCpGType <- rbind(WGBSCpGType, ctResults) #Add test statistics to master table
}
```

```{r}
WGBSCpGType$p.adj <- p.adjust(WGBSCpGType$p.value, method = "fdr") #Correct p-value using FDR
WGBSCpGType$comparison <- rep("WGBS", times = 3) #Add methods compared
head(WGBSCpGType) #Confirm changes
```

#### RRBS

```{r}
RRBSCpGType <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactCpGTypeStatTest)) { #For each CpG type
  Species1CpG <- McapCpGTypeStatTest[2,i] #Variable for # CpG type for Species 1
  Species2CpG <- PactCpGTypeStatTest[2,i] #Variable for # CpG type for Species 2
  Species1NotCpG <- sum(McapCpGTypeStatTest[2,-i]) #Variable for # other CpG types for Species 1
  Species2NotCpG <- sum(PactCpGTypeStatTest[2,-i]) #Variable for # other CpG types for Species 2
  ct <- matrix(c(Species1CpG,Species2CpG,Species1NotCpG,Species2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactCpGTypeStatTest[i])), paste0("Not", colnames(PactCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c("Mcap", "Pact") #Assign row names: Species 1, Species 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(PactCpGTypeStatTest)[i]) #Add CpG type to results
  RRBSCpGType <- rbind(RRBSCpGType, ctResults) #Add test statistics to master table
}
```

```{r}
RRBSCpGType$p.adj <- p.adjust(RRBSCpGType$p.value, method = "fdr") #Correct p-value using FDR
RRBSCpGType$comparison <- rep("RRBS", times = 3) #Add methods compared
head(RRBSCpGType) #Confirm changes
```

#### MBD-BS

```{r}
MBDBSCpGType <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactCpGTypeStatTest)) { #For each CpG type
  Species1CpG <- McapCpGTypeStatTest[3,i] #Variable for # CpG type for Species 1
  Species2CpG <- PactCpGTypeStatTest[3,i] #Variable for # CpG type for Species 2
  Species1NotCpG <- sum(McapCpGTypeStatTest[3,-i]) #Variable for # other CpG types for Species 1
  Species2NotCpG <- sum(PactCpGTypeStatTest[3,-i]) #Variable for # other CpG types for Species 2
  ct <- matrix(c(Species1CpG,Species2CpG,Species1NotCpG,Species2NotCpG), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactCpGTypeStatTest[i])), paste0("Not", colnames(PactCpGTypeStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c("Mcap", "Pact") #Assign row names: Species 1, Species 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$CpGType <- as.character(colnames(PactCpGTypeStatTest)[i]) #Add CpG type to results
  MBDBSCpGType <- rbind(MBDBSCpGType, ctResults) #Add test statistics to master table
}
```

```{r}
MBDBSCpGType$p.adj <- p.adjust(MBDBSCpGType$p.value, method = "fdr") #CoMBDect p-value using FDR
MBDBSCpGType$comparison <- rep("MBDBS", times = 3) #Add methods compared
head(MBDBSCpGType) #Confirm changes
```

#### Save statistical test results

```{r}
InterspeciesCpGTypeStatResults <- rbind(WGBSCpGType,
                                        RRBSCpGType,
                                        MBDBSCpGType) #Combine tables
tail(InterspeciesCpGTypeStatResults) #Confirm changes
```

```{r}
write.table(InterspeciesCpGTypeStatResults, "../analyses/Characterizing-CpG-Methylation-5x-Union/Interspecies_union-CpG-Type-StatResults.txt", quote = FALSE, row.names = FALSE) #Save table
```

## Create multipanel plot

```{r}
#pdf("../Output/Union-CpG-Type-Multipanel.pdf", height = 8.5, width = 11)
par(mar = c(1,5,1,0), oma = c(2, 1, 0, 11), mfrow = c(2,1)) #Change figure boundaries

#Mcap
barplot(t(McapCpGTypePercents), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("", "", ""),
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs. 

for (i in 1:ncol(t(McapCpGTypePercents))){
    xx <- t(McapCpGTypePercents)
    xx[,-i] <- NA
    colnames(xx)[-i] <- NA
    barplot(xx, col = c(plotColors[(3*i)-2], plotColors[(3*i)-1], plotColors[(3*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% CpGs", line = 3, cex = 1.5) #Add y-axis label
text(expression(paste("a. ", italic("M. capitata"))), x = 0.06, y = 105, cex = 1.2, adj = 0) #Add italics species name

#Pact
barplot(t(PactCpGTypePercents), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("WGBS", "RRBS", "MBD-BS"), cex.names = 1.5,
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs. 

for (i in 1:ncol(t(PactCpGTypePercents))){
    xx <- t(PactCpGTypePercents)
    xx[,-i] <- NA
    colnames(xx)[-i] <- NA
    barplot(xx, col = c(plotColors[(3*i)-2], plotColors[(3*i)-1], plotColors[(3*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% CpGs", line = 3, cex = 1.5) #Add y-axis label
text(expression(paste("b. ", italic("P. acuta"))), x = 0.06, y = 105, cex = 1.2, adj = 0) #Add italics species name

#Add legend
par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE) #Create new plot
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n") #Add new plot on top of current plot
legend(x = 0.62, y = 0.96, 
       xpd = TRUE,
       legend = c("Strong", "Moderate", "Weak"),
       pch = 22, 
        col = "black", 
        pt.bg = c("grey20", "grey50", "grey80"),
       bty = "n",
       cex = 1.5, 
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box
text("Methylation Status", x = 0.85, y = 0.965, cex = 1.5) #Add legend title that is aligned with legend

#dev.off() #Turn off plotting device
```

# CpG feature overlaps

## Mcap

### Import file counts

```{r}
McapGenomeFeatures <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-CGMotif-Overlaps-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with CG motif and feature track overlaps
McapGenomeFeatures <- McapGenomeFeatures[-8,] #Remove final row
tail(McapGenomeFeatures) #Check import
```

```{r}
McapGeneOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-mcGenes-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-gene overlaps
McapGeneOverlaps <- McapGeneOverlaps[-13,] #Remove final row
tail(McapGeneOverlaps) #Confirm import
```

```{r}
McapCDSOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-mcCDS-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-CDS overlaps
McapCDSOverlaps <- McapCDSOverlaps[-13,] #Remove final row
tail(McapCDSOverlaps) #Confirm import
```

```{r}
McapIntronsOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-mcIntrons-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
McapIntronsOverlaps <- McapIntronsOverlaps[-13,] #Remove final row
tail(McapIntronsOverlaps) #Confirm import
```

```{r}
McapFlanksOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-mcFlanks-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
McapFlanksOverlaps <- McapFlanksOverlaps[-13,] #Remove final row
tail(McapFlanksOverlaps) #Confirm import
```

```{r}
McapFlanksUpstreamOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-mcFlanksUpstream-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
McapFlanksUpstreamOverlaps <- McapFlanksUpstreamOverlaps[-13,] #Remove final row
tail(McapFlanksUpstreamOverlaps) #Confirm import
```

```{r}
McapFlanksDownstreamOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-mcFlanksDownstream-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
McapFlanksDownstreamOverlaps <- McapFlanksDownstreamOverlaps[-13,] #Remove final row
tail(McapFlanksDownstreamOverlaps) #Confirm import
```

```{r}
McapIntergenicOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union_5x-mcIntergenic-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Intergenic overlaps
McapIntergenicOverlaps <- McapIntergenicOverlaps[-13,] #Remove final row
tail(McapIntergenicOverlaps) #Confirm import
```

### Create summary tables

```{r}
McapFeatureOverlaps <- data.frame("allCpGs" = rep(0, times = 7),
                                  "MBDBSMeth" = rep(0, times = 7),
                                  "MBDBSsparseMeth" = rep(0, times = 7),
                                  "MBDBSunMeth" = rep(0, times = 7),
                                  "MBDBS" = rep(0, times = 7),
                                  "RRBSMeth" = rep(0, times = 7),
                                  "RRBSsparseMeth" = rep(0, times = 7),
                                  "RRBSunMeth" = rep(0, times = 7),
                                  "RRBS" = rep(0, times = 7),
                                  "WGBSMeth" = rep(0, times = 7),
                                  "WGBSsparseMeth" = rep(0, times = 7),
                                  "WGBSunMeth" = rep(0, times = 7),                                  
                                  "WGBS" = rep(0, times = 7)) #Create blank dataframe with information for various CpG categories and methylation status. Match columns to the order of columns in the overlap count files
row.names(McapFeatureOverlaps) <- c("Genes", "CDS", "Introns", "Flanking Regions", "Upstream Flanks", "Downstream Flanks", "Intergenic") #Assign row names
head(McapFeatureOverlaps) #Confirm changes
```

```{r}
McapFeatureOverlaps$allCpGs <- c(McapGenomeFeatures$counts[5],
                                 McapGenomeFeatures$counts[1],
                                 McapGenomeFeatures$counts[7],
                                 McapGenomeFeatures$counts[3],
                                 McapGenomeFeatures$counts[4],
                                 McapGenomeFeatures$counts[2],
                                 McapGenomeFeatures$counts[6]) #Assign information for CG motif overlaps with genome features. Use 0 for totalLines
head(McapFeatureOverlaps) #Confirm modification
```

```{r}
for (i in 1:length(McapGeneOverlaps$counts)) {
  McapFeatureOverlaps[1,i+1] <- McapGeneOverlaps[i,1]
  McapFeatureOverlaps[2,i+1] <- McapCDSOverlaps[i,1]
  McapFeatureOverlaps[3,i+1] <- McapIntronsOverlaps[i,1]
  McapFeatureOverlaps[4,i+1] <- McapFlanksOverlaps[i,1]
  McapFeatureOverlaps[5,i+1] <- McapFlanksUpstreamOverlaps[i,1]
  McapFeatureOverlaps[6,i+1] <- McapFlanksDownstreamOverlaps[i,1]
  McapFeatureOverlaps[7,i+1] <- McapIntergenicOverlaps[i,1]
} #For each table with feature overlap information, paste the contents of the count column in the assigned row
tail(McapFeatureOverlaps) #Check summary table
```

```{r}
write.table(McapFeatureOverlaps, "../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union-Genomic-Location-Counts.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

### Conduct chi-squared tests of homogeneity

```{r}
McapFeatureStatTest <- data.frame(t(McapFeatureOverlaps[-c(1,4),c(1,13,9,5)])) #Separate out count data for statistical testing. Remove gene and total flank information. Transpose and save as dataframe so methods are row names
head(McapFeatureStatTest) #Confirm changes
```

#### All vs. WGBS

```{r}
McapFeatureAllWG <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(McapFeatureStatTest)) { #For each feature
  AllFeature <- McapFeatureStatTest[1,i] #Variable for # feature overlap with background CpGs
  MethodFeature <- McapFeatureStatTest[2,i] #Variable for # feature overlap with method CpGs
  AllNotFeature <- sum(McapFeatureStatTest[1,-i]) #Variable for # other feature overlap with background CpGs
  MethodNotFeature <- sum(McapFeatureStatTest[2,-i]) #Variable for # other feature overlap with method CpGs
  ct <- matrix(c(AllFeature,MethodFeature,AllNotFeature,MethodNotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(McapFeatureStatTest[i])), paste0("Not", colnames(McapFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(McapFeatureStatTest)[c(1,2)])) #Assign row names: all, method
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(McapFeatureStatTest)[i]) #Add feature to results
  McapFeatureAllWG <- rbind(McapFeatureAllWG, ctResults) #Add test statistics to master table
}
```

```{r}
McapFeatureAllWG$p.adj <- p.adjust(McapFeatureAllWG$p.value, method = "fdr") #Correct p-value using FDR
McapFeatureAllWG$comparison <- rep("All vs. WGBS", times = 5) #Add methods compared
head(McapFeatureAllWG) #Confirm changes
```

#### All vs. RRBS

```{r}
McapFeatureAllRR <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(McapFeatureStatTest)) { #For each feature
  AllFeature <- McapFeatureStatTest[1,i] #Variable for # feature overlap with background CpGs
  MethodFeature <- McapFeatureStatTest[3,i] #Variable for # feature overlap with method CpGs
  AllNotFeature <- sum(McapFeatureStatTest[1,-i]) #Variable for # other feature overlap with background CpGs
  MethodNotFeature <- sum(McapFeatureStatTest[3,-i]) #Variable for # other feature overlap with method CpGs
  ct <- matrix(c(AllFeature,MethodFeature,AllNotFeature,MethodNotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(McapFeatureStatTest[i])), paste0("Not", colnames(McapFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(McapFeatureStatTest)[c(1,3)])) #Assign row names: all, method
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(McapFeatureStatTest)[i]) #Add feature to results
  McapFeatureAllRR <- rbind(McapFeatureAllRR, ctResults) #Add test statistics to master table
}
```

```{r}
McapFeatureAllRR$p.adj <- p.adjust(McapFeatureAllRR$p.value, method = "fdr") #Correct p-value using FDR
McapFeatureAllRR$comparison <- rep("All vs. RRBS", times = 5) #Add methods compared
head(McapFeatureAllRR) #Confirm changes
```

#### All vs. MBD-BS

```{r}
McapFeatureAllMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(McapFeatureStatTest)) { #For each feature
  AllFeature <- McapFeatureStatTest[1,i] #Variable for # feature overlap with background CpGs
  MethodFeature <- McapFeatureStatTest[4,i] #Variable for # feature overlap with method CpGs
  AllNotFeature <- sum(McapFeatureStatTest[1,-i]) #Variable for # other feature overlap with background CpGs
  MethodNotFeature <- sum(McapFeatureStatTest[4,-i]) #Variable for # other feature overlap with method CpGs
  ct <- matrix(c(AllFeature,MethodFeature,AllNotFeature,MethodNotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(McapFeatureStatTest[i])), paste0("Not", colnames(McapFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(McapFeatureStatTest)[c(1,4)])) #Assign row names: all, method
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(McapFeatureStatTest)[i]) #Add feature to results
  McapFeatureAllMB <- rbind(McapFeatureAllMB, ctResults) #Add test statistics to master table
}
```

```{r}
McapFeatureAllMB$p.adj <- p.adjust(McapFeatureAllMB$p.value, method = "fdr") #Correct p-value using FDR
McapFeatureAllMB$comparison <- rep("All vs. MBDBS", times = 5) #Add methods compared
head(McapFeatureAllMB) #Confirm changes
```

#### WGBS vs. RRBS

```{r}
McapFeatureWGRR <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(McapFeatureStatTest)) { #For each feature
  Method1Feature <- McapFeatureStatTest[2,i] #Variable for # feature for method 1
  Method2Feature <- McapFeatureStatTest[3,i] #Variable for # feature for method 2
  Method1NotFeature <- sum(McapFeatureStatTest[2,-i]) #Variable for # other feature for method 1
  Method2NotFeature <- sum(McapFeatureStatTest[3,-i]) #Variable for # other feature for method 2
  ct <- matrix(c(Method1Feature,Method2Feature,Method1NotFeature,Method2NotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(McapFeatureStatTest[i])), paste0("Not", colnames(McapFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(McapFeatureStatTest)[c(2,3)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(McapFeatureStatTest)[i]) #Add feature to results
  McapFeatureWGRR <- rbind(McapFeatureWGRR, ctResults) #Add test statistics to master table
}
```

```{r}
McapFeatureWGRR$p.adj <- p.adjust(McapFeatureWGRR$p.value, method = "fdr") #Correct p-value using FDR
McapFeatureWGRR$comparison <- rep("WGBS vs. RRBS", times = 5) #Add methods compared
head(McapFeatureWGRR) #Confirm changes
```

#### WGBS vs. MBD-BS

```{r}
McapFeatureWGMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(McapFeatureStatTest)) { #For each feature
  Method1Feature <- McapFeatureStatTest[2,i] #Variable for # feature for method 1
  Method2Feature <- McapFeatureStatTest[4,i] #Variable for # feature for method 2
  Method1NotFeature <- sum(McapFeatureStatTest[2,-i]) #Variable for # other feature for method 1
  Method2NotFeature <- sum(McapFeatureStatTest[4,-i]) #Variable for # other feature for method 2
  ct <- matrix(c(Method1Feature,Method2Feature,Method1NotFeature,Method2NotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(McapFeatureStatTest[i])), paste0("Not", colnames(McapFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(McapFeatureStatTest)[c(2,4)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(McapFeatureStatTest)[i]) #Add feature to results
  McapFeatureWGMB <- rbind(McapFeatureWGMB, ctResults) #Add test statistics to master table
}
```

```{r}
McapFeatureWGMB$p.adj <- p.adjust(McapFeatureWGMB$p.value, method = "fdr") #Correct p-value using FDR
McapFeatureWGMB$comparison <- rep("WGBS vs. MBDBS", times = 5) #Add methods compared
head(McapFeatureWGMB) #Confirm changes
```

#### RRBS vs. MBD-BS

```{r}
McapFeatureRRMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(McapFeatureStatTest)) { #For each feature
  Method1Feature <- McapFeatureStatTest[3,i] #Variable for # feature for method 1
  Method2Feature <- McapFeatureStatTest[4,i] #Variable for # feature for method 2
  Method1NotFeature <- sum(McapFeatureStatTest[3,-i]) #Variable for # other feature for method 1
  Method2NotFeature <- sum(McapFeatureStatTest[4,-i]) #Variable for # other feature for method 2
  ct <- matrix(c(Method1Feature,Method2Feature,Method1NotFeature,Method2NotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(McapFeatureStatTest[i])), paste0("Not", colnames(McapFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(McapFeatureStatTest)[c(3,4)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(McapFeatureStatTest)[i]) #Add feature to results
  McapFeatureRRMB <- rbind(McapFeatureRRMB, ctResults) #Add test statistics to master table
}
```

```{r}
McapFeatureRRMB$p.adj <- p.adjust(McapFeatureRRMB$p.value, method = "fdr") #Correct p-value using FDR
McapFeatureRRMB$comparison <- rep("RRBS vs. MBDBS", times = 5) #Add methods compared
head(McapFeatureRRMB) #Confirm changes
```

#### Save statistical test results

```{r}
McapFeatureStatResults <- rbind(McapFeatureAllWG,
                                McapFeatureAllRR,
                                McapFeatureAllMB,
                                McapFeatureWGRR,
                                McapFeatureWGMB,
                                McapFeatureRRMB) #Combine tables
tail(McapFeatureStatResults) #Confirm changes
```

```{r}
write.table(McapFeatureStatResults, "../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union-Genomic-Location-StatResults.txt", quote = FALSE, row.names = FALSE) #Save table
```

### Create plot

```{r}
McapFeatureOverlapsPercents <- McapFeatureOverlaps[-c(1,4),] #Duplicate dataframe but remove gene and total flank rows
for (i in 1:length(McapFeatureOverlaps)) {
  McapFeatureOverlapsPercents[,i] <- (McapFeatureOverlapsPercents[,i] / (sum(McapFeatureOverlapsPercents[,i]))) * 100
} #Divide every entry by sum of the column and multiply by 100 to get percentages. Do not include gene information
head(McapFeatureOverlapsPercents) #Check calculations
```

```{r}
write.table(McapFeatureOverlapsPercents, "../analyses/Characterizing-CpG-Methylation-5x-Union/Mcap/Mcap_union-Genomic-Location-Percents.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

```{r}
plotColors2 <- c("grey20", "grey40", "grey60", "grey80", "grey100", 
                rev(RColorBrewer::brewer.pal(5, "Greens")), 
                rev(RColorBrewer::brewer.pal(5, "Blues")), 
                rev(RColorBrewer::brewer.pal(5, "Reds"))) #Create vector of all, WGBS, RRBS, and MBD colors
barplot(t(t(McapFeatureOverlapsPercents)), col = dichromat(c(rev(RColorBrewer::brewer.pal(5, "Greens"))))) #Check greens
barplot(t(t(McapFeatureOverlapsPercents)), col = dichromat(c(rev(RColorBrewer::brewer.pal(5, "Blues"))))) #Check blues
barplot(t(t(McapFeatureOverlapsPercents)), col = dichromat(c(rev(RColorBrewer::brewer.pal(5, "Reds"))))) #Check reds
```

#### All data

```{r}
#pdf("../Output/Mcap_union-CpG-Features.pdf", height = 8.5, width = 11)
par(mar = c(2,5,0,1), oma = c(1, 1, 0, 11)) #Change figure boundaries

barplot(t(t(McapFeatureOverlapsPercents[,c(1,13,9,5)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("Genome", "WGBS", "RRBS", "MBD-BS"), cex.names = 1.5,
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs. 

for (i in 1:4) {
    xx <- t(t(McapFeatureOverlapsPercents[,c(1,13,9,5)]))
    xx[,-i] <- NA
    colnames(xx)[-i] <- NA
    barplot(xx, col = c(plotColors2[(5*i)-4], plotColors2[(5*i)-3], plotColors2[(5*i)-2], plotColors2[(5*i)-1], plotColors2[(5*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% CpGs", line = 3, cex = 1.5) #Add y-axis label

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE) #Create new plot
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n") #Add new plot on top of current plot
legend(x = 0.57, y = 0.87, 
       xpd = TRUE,
       legend = c("CDS", "Introns", "Upstream Flank", "Downstream Flank", "Intergenic"),
       pch = 22, 
        col = "black", 
        pt.bg = c("grey20", "grey40", "grey60", "grey80", "grey100"),
       bty = "n",
       cex = 1.5, 
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box
text("Genome Feature", x = 0.785, y = 0.879, cex = 1.5) #Add legend title that is aligned with legend

#dev.off() #Turn off plotting device
```

#### CpG feature overlap by methylation status

```{r}
#pdf("../Output/Mcap_union-CpG-Features-MethStatus.pdf", height = 8.5, width = 11)
par(mar = c(1,5,0,1), oma = c(4, 1, 1, 11), mfcol = c(3,1)) #Change figure boundaries

#Strong methylation
barplot(t(t(McapFeatureOverlapsPercents[,c(10,6,2)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("", "", ""),
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Set y-axis specs.

for (i in 2:4) {
    xx <- t(t(McapFeatureOverlapsPercents[,c(10,6,2)]))
    xx[,-(i-1)] <- NA
    colnames(xx)[-(i-1)] <- NA
    barplot(xx, col = c(plotColors2[(5*i)-4], plotColors2[(5*i)-3], plotColors2[(5*i)-2], plotColors2[(5*i)-1], plotColors2[(5*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% Strong", line = 3, cex = 1.5) #Add y-axis label

#Moderate methylation
barplot(t(t(McapFeatureOverlapsPercents[,c(11,7,3)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("", "", ""),
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs.

for (i in 2:4) {
    xx <- t(t(McapFeatureOverlapsPercents[,c(11,7,3)]))
    xx[,-(i-1)] <- NA
    colnames(xx)[-(i-1)] <- NA
    barplot(xx, col = c(plotColors2[(5*i)-4], plotColors2[(5*i)-3], plotColors2[(5*i)-2], plotColors2[(5*i)-1], plotColors2[(5*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% Moderate", line = 3, cex = 1.5) #Add y-axis label

#Weak methylation
barplot(t(t(McapFeatureOverlapsPercents[,c(12,8,4)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("WGBS", "RRBS", "MBD-BS"), cex.names = 2,
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs.

for (i in 2:4) {
    xx <- t(t(McapFeatureOverlapsPercents[,c(12,8,4)]))
    xx[,-(i-1)] <- NA
    colnames(xx)[-(i-1)] <- NA
    barplot(xx, col = c(plotColors2[(5*i)-4], plotColors2[(5*i)-3], plotColors2[(5*i)-2], plotColors2[(5*i)-1], plotColors2[(5*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% Weak", line = 3, cex = 1.5) #Add y-axis label

#Add legend
par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE) #Create new plot
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n") #Add new plot on top of current plot
legend(x = 0.73, y = 0.97, 
       xpd = TRUE,
       legend = c("CDS", "Introns", "Upstream Flank", "Downstream Flank", "Intergenic"),
       pch = 22, 
        col = "black", 
        pt.bg = c("grey20", "grey40", "grey60", "grey80", "grey100"),
       bty = "n",
       cex = 1.5, 
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box
text("Genome Feature", x = 0.87, y = 0.97, cex = 1.5) #Add legend title that is aligned with legend

#dev.off() #Turn off plotting device
```

### Create comparative barplots

```{r}
McapFeatureOverlapMethProps <- data.frame("WBGS" = (McapFeatureOverlaps$WGBSMeth / McapFeatureOverlaps$allCpGs),
                                          "RRBS" = (McapFeatureOverlaps$RRBSMeth / McapFeatureOverlaps$allCpGs),
                                          "MBDBS" = (McapFeatureOverlaps$MBDBSMeth / McapFeatureOverlaps$allCpGs)) #Divide # strong meth loci by all CpGs in a feature to get proportions
McapFeatureOverlapMethProps <- McapFeatureOverlapMethProps * 100 #Multiply by 100 to get percents
row.names(McapFeatureOverlapMethProps) <- row.names(McapFeatureOverlaps) #Assign row names
McapFeatureOverlapMethProps <- McapFeatureOverlapMethProps[-c(1,4),] #Remove genes and total flanks information
head(McapFeatureOverlapMethProps) #Confirm proportions
```

```{r}
#pdf("../Output/Mcap_union-StrongMeth-CpG-Features.pdf", width = 11, height = 8.5) #Save file
par(mar = c(5, 8, 1, 1),  oma = c(1, 5, 1, 1)) #Change figure boundaries

barsMcap <- barplot(t(McapFeatureOverlapMethProps),
                    beside = TRUE, horiz = TRUE,
                    col = c(plotColors2[7], plotColors2[12], plotColors2[17]),
                    xlim = c(0,10),
                    axes = FALSE, names.arg = rep("", times = 5)) #Create a horizontal barplot with comparative proportions. Color based on sequencing method and do not include any axes. Use blanks as y-axis labels. Save barplot information.

axis(side = 2, at = barsMcap[2,], labels = row.names(McapFeatureOverlapMethProps), tick = FALSE, las = 2, col = "grey80", cex.axis = 1.5) #Add y-axis labels at the center of grouped bars. Remove tick marks and change label orientation.

axis(side = 1, at = seq(from = 0, to = 10, by = 2), cex = 1.2, col = "grey80") #Add x-axis
mtext(side = 1, "Strongly Methylated CpGs/All CpGs (%)", line = 3, cex = 1.5) #Add x-axis label

legend("topright",
       xpd = TRUE,
       legend = c("WGBS", "RRBS", "MBD-BS"),
       pch = 22, 
        col = "black", 
        pt.bg = c(plotColors2[7], plotColors2[12], plotColors2[17]),
       bty = "n",
       cex = 1.5,
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box

#dev.off()
```

```{r}
#pdf("../Output/Mcap_union-StrongMeth-CpG-Features-NoFlanks.pdf", width = 11, height = 8.5) #Save file
par(mar = c(5, 8, 1, 1),  oma = c(1, 0, 1, 1)) #Change figure boundaries

barsMcap <- barplot(t(McapFeatureOverlapMethProps[-c(3,4),]),
                    beside = TRUE, horiz = TRUE,
                    col = c(plotColors2[7], plotColors2[12], plotColors2[17]),
                    xlim = c(0,10),
                    axes = FALSE, names.arg = rep("", times = 3)) #Create a horizontal barplot with comparative proportions. Color based on sequencing method and do not include any axes. Use blanks as y-axis labels. Save barplot information.

axis(side = 2, at = barsMcap[2,], labels = row.names(McapFeatureOverlapMethProps[-c(3,4),]), tick = FALSE, las = 2, col = "grey80", cex.axis = 1.5) #Add y-axis labels at the center of grouped bars. Remove tick marks and change label orientation.

axis(side = 1, at = seq(from = 0, to = 10, by = 2), cex = 1.2, col = "grey80") #Add x-axis
mtext(side = 1, "Strongly Methylated CpGs/All CpGs (%)", line = 3, cex = 1.5) #Add x-axis label

legend("topright",
       xpd = TRUE,
       legend = c("WGBS", "RRBS", "MBD-BS"),
       pch = 22, 
        col = "black", 
        pt.bg = c(plotColors2[7], plotColors2[12], plotColors2[17]),
       bty = "n",
       cex = 1.5,
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box

#dev.off()
```

## Pact

### Import file counts

```{r}
PactGenomeFeatures <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Pact/Pact-CGMotif-Overlaps-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with CG motif and feature track overlaps
PactGenomeFeatures <- PactGenomeFeatures[-8,] #Remove final row
tail(PactGenomeFeatures) #Check import
```

```{r}
PactGeneOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-paGenes-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-gene overlaps
PactGeneOverlaps <- PactGeneOverlaps[-13,] #Remove final row
tail(PactGeneOverlaps) #Confirm import
```

```{r}
PactCDSOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-paCDS-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-CDS overlaps
PactCDSOverlaps <- PactCDSOverlaps[-13,] #Remove final row
tail(PactCDSOverlaps) #Confirm import
```

```{r}
PactIntronsOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-paIntron-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
PactIntronsOverlaps <- PactIntronsOverlaps[-13,] #Remove final row
tail(PactIntronsOverlaps) #Confirm import
```

```{r}
PactFlanksOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-paFlanks-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
PactFlanksOverlaps <- PactFlanksOverlaps[-13,] #Remove final row
tail(PactFlanksOverlaps) #Confirm import
```

```{r}
PactFlanksUpstreamOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-paFlanksUpstream-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
PactFlanksUpstreamOverlaps <- PactFlanksUpstreamOverlaps[-13,] #Remove final row
tail(PactFlanksUpstreamOverlaps) #Confirm import
```

```{r}
PactFlanksDownstreamOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-paFlanksDownstream-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
PactFlanksDownstreamOverlaps <- PactFlanksDownstreamOverlaps[-13,] #Remove final row
tail(PactFlanksDownstreamOverlaps) #Confirm import
```

```{r}
PactIntergenicOverlaps <- read.table("../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union_5x-paIntergenic-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Intergenic overlaps
PactIntergenicOverlaps <- PactIntergenicOverlaps[-13,] #Remove final row
tail(PactIntergenicOverlaps) #Confirm import
```

### Create summary tables

```{r}
PactFeatureOverlaps <- data.frame("allCpGs" = rep(0, times = 7),
                                  "MBDBSMeth" = rep(0, times = 7),
                                  "MBDBSsparseMeth" = rep(0, times = 7),
                                  "MBDBSunMeth" = rep(0, times = 7),
                                  "MBDBS" = rep(0, times = 7),
                                  "RRBSMeth" = rep(0, times = 7),
                                  "RRBSsparseMeth" = rep(0, times = 7),
                                  "RRBSunMeth" = rep(0, times = 7),
                                  "RRBS" = rep(0, times = 7),
                                  "WGBSMeth" = rep(0, times = 7),
                                  "WGBSsparseMeth" = rep(0, times = 7),
                                  "WGBSunMeth" = rep(0, times = 7),                                  
                                  "WGBS" = rep(0, times = 7)) #Create blank dataframe with information for various CpG categories and methylation status. Match columns to the order of columns in the overlap count files
row.names(PactFeatureOverlaps) <- c("Genes", "CDS", "Introns", "Flanks", "Upstream Flanks", "Downstream Flanks", "Intergenic") #Assign row names
head(PactFeatureOverlaps) #Confirm changes
```

```{r}
PactFeatureOverlaps$allCpGs <- c(PactGenomeFeatures$counts[5],
                                 PactGenomeFeatures$counts[1],
                                 PactGenomeFeatures$counts[7],
                                 PactGenomeFeatures$counts[3],
                                 PactGenomeFeatures$counts[4],
                                 PactGenomeFeatures$counts[2],
                                 PactGenomeFeatures$counts[6]) #Assign information for CG motif overlaps with genome features. Use 0 for totalLines
head(PactFeatureOverlaps) #Confirm modification
```

```{r}
for (i in 1:length(PactGeneOverlaps$counts)) {
  PactFeatureOverlaps[1,i+1] <- PactGeneOverlaps[i,1]
  PactFeatureOverlaps[2,i+1] <- PactCDSOverlaps[i,1]
  PactFeatureOverlaps[3,i+1] <- PactIntronsOverlaps[i,1]
  PactFeatureOverlaps[4,i+1] <- PactFlanksOverlaps[i,1]
  PactFeatureOverlaps[5,i+1] <- PactFlanksUpstreamOverlaps[i,1]
  PactFeatureOverlaps[6,i+1] <- PactFlanksDownstreamOverlaps[i,1]
  PactFeatureOverlaps[7,i+1] <- PactIntergenicOverlaps[i,1]
} #For each table with feature overlap information, paste the contents of the count column in the assigned row
tail(PactFeatureOverlaps) #Check summary table
```

```{r}
write.table(PactFeatureOverlaps, "../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union-Genomic-Location-Counts.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

### Conduct chi-squared tests of homogeneity

```{r}
PactFeatureStatTest <- data.frame(t(PactFeatureOverlaps[-c(1,4),c(1,13,9,5)])) #Separate out count data for statistical testing. Remove gene and total flank information. Transpose and save as dataframe so methods are row names
head(PactFeatureStatTest) #Confirm changes
```

#### All vs. WGBS

```{r}
PactFeatureAllWG <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactFeatureStatTest)) { #For each feature
  AllFeature <- PactFeatureStatTest[1,i] #Variable for # feature overlap with background CpGs
  MethodFeature <- PactFeatureStatTest[2,i] #Variable for # feature overlap with method CpGs
  AllNotFeature <- sum(PactFeatureStatTest[1,-i]) #Variable for # other feature overlap with background CpGs
  MethodNotFeature <- sum(PactFeatureStatTest[2,-i]) #Variable for # other feature overlap with method CpGs
  ct <- matrix(c(AllFeature,MethodFeature,AllNotFeature,MethodNotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactFeatureStatTest[i])), paste0("Not", colnames(PactFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(PactFeatureStatTest)[c(1,2)])) #Assign row names: all, method
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(PactFeatureStatTest)[i]) #Add feature to results
  PactFeatureAllWG <- rbind(PactFeatureAllWG, ctResults) #Add test statistics to master table
}
```

```{r}
PactFeatureAllWG$p.adj <- p.adjust(PactFeatureAllWG$p.value, method = "fdr") #Correct p-value using FDR
PactFeatureAllWG$comparison <- rep("All vs. WGBS", times = 5) #Add methods compared
head(PactFeatureAllWG) #Confirm changes
```

#### All vs. RRBS

```{r}
PactFeatureAllRR <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactFeatureStatTest)) { #For each feature
  AllFeature <- PactFeatureStatTest[1,i] #Variable for # feature overlap with background CpGs
  MethodFeature <- PactFeatureStatTest[3,i] #Variable for # feature overlap with method CpGs
  AllNotFeature <- sum(PactFeatureStatTest[1,-i]) #Variable for # other feature overlap with background CpGs
  MethodNotFeature <- sum(PactFeatureStatTest[3,-i]) #Variable for # other feature overlap with method CpGs
  ct <- matrix(c(AllFeature,MethodFeature,AllNotFeature,MethodNotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactFeatureStatTest[i])), paste0("Not", colnames(PactFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(PactFeatureStatTest)[c(1,3)])) #Assign row names: all, method
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(PactFeatureStatTest)[i]) #Add feature to results
  PactFeatureAllRR <- rbind(PactFeatureAllRR, ctResults) #Add test statistics to master table
}
```

```{r}
PactFeatureAllRR$p.adj <- p.adjust(PactFeatureAllRR$p.value, method = "fdr") #Correct p-value using FDR
PactFeatureAllRR$comparison <- rep("All vs. RRBS", times = 5) #Add methods compared
head(PactFeatureAllRR) #Confirm changes
```

#### All vs. MBD-BS

```{r}
PactFeatureAllMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactFeatureStatTest)) { #For each feature
  AllFeature <- PactFeatureStatTest[1,i] #Variable for # feature overlap with background CpGs
  MethodFeature <- PactFeatureStatTest[4,i] #Variable for # feature overlap with method CpGs
  AllNotFeature <- sum(PactFeatureStatTest[1,-i]) #Variable for # other feature overlap with background CpGs
  MethodNotFeature <- sum(PactFeatureStatTest[4,-i]) #Variable for # other feature overlap with method CpGs
  ct <- matrix(c(AllFeature,MethodFeature,AllNotFeature,MethodNotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactFeatureStatTest[i])), paste0("Not", colnames(PactFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(PactFeatureStatTest)[c(1,4)])) #Assign row names: all, method
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(PactFeatureStatTest)[i]) #Add feature to results
  PactFeatureAllMB <- rbind(PactFeatureAllMB, ctResults) #Add test statistics to master table
}
```

```{r}
PactFeatureAllMB$p.adj <- p.adjust(PactFeatureAllMB$p.value, method = "fdr") #Correct p-value using FDR
PactFeatureAllMB$comparison <- rep("All vs. MBDBS", times = 5) #Add methods compared
head(PactFeatureAllMB) #Confirm changes
```

#### WGBS vs. RRBS

```{r}
PactFeatureWGRR <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactFeatureStatTest)) { #For each feature
  Method1Feature <- PactFeatureStatTest[2,i] #Variable for # feature for method 1
  Method2Feature <- PactFeatureStatTest[3,i] #Variable for # feature for method 2
  Method1NotFeature <- sum(PactFeatureStatTest[2,-i]) #Variable for # other feature for method 1
  Method2NotFeature <- sum(PactFeatureStatTest[3,-i]) #Variable for # other feature for method 2
  ct <- matrix(c(Method1Feature,Method2Feature,Method1NotFeature,Method2NotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactFeatureStatTest[i])), paste0("Not", colnames(PactFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(PactFeatureStatTest)[c(2,3)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(PactFeatureStatTest)[i]) #Add feature to results
  PactFeatureWGRR <- rbind(PactFeatureWGRR, ctResults) #Add test statistics to master table
}
```

```{r}
PactFeatureWGRR$p.adj <- p.adjust(PactFeatureWGRR$p.value, method = "fdr") #Correct p-value using FDR
PactFeatureWGRR$comparison <- rep("WGBS vs. RRBS", times = 5) #Add methods compared
head(PactFeatureWGRR) #Confirm changes
```

#### WGBS vs. MBD-BS

```{r}
PactFeatureWGMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactFeatureStatTest)) { #For each feature
  Method1Feature <- PactFeatureStatTest[2,i] #Variable for # feature for method 1
  Method2Feature <- PactFeatureStatTest[4,i] #Variable for # feature for method 2
  Method1NotFeature <- sum(PactFeatureStatTest[2,-i]) #Variable for # other feature for method 1
  Method2NotFeature <- sum(PactFeatureStatTest[4,-i]) #Variable for # other feature for method 2
  ct <- matrix(c(Method1Feature,Method2Feature,Method1NotFeature,Method2NotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactFeatureStatTest[i])), paste0("Not", colnames(PactFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(PactFeatureStatTest)[c(2,4)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(PactFeatureStatTest)[i]) #Add feature to results
  PactFeatureWGMB <- rbind(PactFeatureWGMB, ctResults) #Add test statistics to master table
}
```

```{r}
PactFeatureWGMB$p.adj <- p.adjust(PactFeatureWGMB$p.value, method = "fdr") #Correct p-value using FDR
PactFeatureWGMB$comparison <- rep("WGBS vs. MBDBS", times = 5) #Add methods compared
head(PactFeatureWGMB) #Confirm changes
```

#### RRBS vs. MBD-BS

```{r}
PactFeatureRRMB <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactFeatureStatTest)) { #For each feature
  Method1Feature <- PactFeatureStatTest[3,i] #Variable for # feature for method 1
  Method2Feature <- PactFeatureStatTest[4,i] #Variable for # feature for method 2
  Method1NotFeature <- sum(PactFeatureStatTest[3,-i]) #Variable for # other feature for method 1
  Method2NotFeature <- sum(PactFeatureStatTest[4,-i]) #Variable for # other feature for method 2
  ct <- matrix(c(Method1Feature,Method2Feature,Method1NotFeature,Method2NotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactFeatureStatTest[i])), paste0("Not", colnames(PactFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(PactFeatureStatTest)[c(3,4)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(PactFeatureStatTest)[i]) #Add feature to results
  PactFeatureRRMB <- rbind(PactFeatureRRMB, ctResults) #Add test statistics to master table
}
```

```{r}
PactFeatureRRMB$p.adj <- p.adjust(PactFeatureRRMB$p.value, method = "fdr") #Correct p-value using FDR
PactFeatureRRMB$comparison <- rep("RRBS vs. MBDBS", times = 5) #Add methods compared
head(PactFeatureRRMB) #Confirm changes
```

#### Save statistical test results

```{r}
PactFeatureStatResults <- rbind(PactFeatureAllWG,
                                PactFeatureAllRR,
                                PactFeatureAllMB,
                                PactFeatureWGRR,
                                PactFeatureWGMB,
                                PactFeatureRRMB) #Combine tables
tail(PactFeatureStatResults) #Confirm changes
```

```{r}
write.table(PactFeatureStatResults, "../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union-Genomic-Location-StatResults.txt", quote = FALSE, row.names = FALSE) #Save table
```

### Create stacked barplot

```{r}
PactFeatureOverlapsPercents <- PactFeatureOverlaps[-c(1,4),] #Duplicate dataframe but remove gene and total flank rows
for (i in 1:length(PactFeatureOverlaps)) {
  PactFeatureOverlapsPercents[,i] <- (PactFeatureOverlapsPercents[,i] / (sum(PactFeatureOverlapsPercents[,i]))) * 100
} #Divide every entry by sum of the column and multiply by 100 to get percentages. Do not include gene information
head(PactFeatureOverlapsPercents) #Check calculations
```

```{r}
write.table(PactFeatureOverlapsPercents, "../analyses/Characterizing-CpG-Methylation-5x-Union/Pact/Pact_union-Genomic-Location-Percents.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

#### All data

```{r}
#pdf("../Output/Pact_union-CpG-Features.pdf", height = 8.5, width = 11)
par(mar = c(2,5,0,1), oma = c(1, 1, 0, 11)) #Change figure boundaries

barplot(t(t(PactFeatureOverlapsPercents[,c(1,13,9,5)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("Genome", "WGBS", "RRBS", "MBD-BS"), cex.names = 1.5,
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs. 

for (i in 1:4) {
    xx <- t(t(PactFeatureOverlapsPercents[,c(1,13,9,5)]))
    xx[,-i] <- NA
    colnames(xx)[-i] <- NA
    barplot(xx, col = c(plotColors2[(5*i)-4], plotColors2[(5*i)-3], plotColors2[(5*i)-2], plotColors2[(5*i)-1], plotColors2[(5*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% CpGs", line = 3, cex = 1.5) #Add y-axis label

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE) #Create new plot
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n") #Add new plot on top of current plot
legend(x = 0.57, y = 0.87, 
       xpd = TRUE,
       legend = c("CDS", "Introns", "Upstream Flank", "Downstream Flank", "Intergenic"),
       pch = 22, 
        col = "black", 
        pt.bg = c("grey20", "grey40", "grey60", "grey80", "grey100"),
       bty = "n",
       cex = 1.5, 
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box
text("Genome Feature", x = 0.785, y = 0.879, cex = 1.5) #Add legend title that is aligned with legend

#dev.off() #Turn off plotting device
```

#### CpG feature overlap by methylation status

```{r}
#pdf("../Output/Pact_union-CpG-Features-MethStatus.pdf", height = 8.5, width = 11)
par(mar = c(1,5,0,1), oma = c(4, 1, 1, 11), mfcol = c(3,1)) #Change figure boundaries

#Strong methylation
barplot(t(t(PactFeatureOverlapsPercents[,c(10,6,2)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("", "", ""),
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Set y-axis specs.

for (i in 2:4) {
    xx <- t(t(PactFeatureOverlapsPercents[,c(10,6,2)]))
    xx[,-(i-1)] <- NA
    colnames(xx)[-(i-1)] <- NA
    barplot(xx, col = c(plotColors2[(5*i)-4], plotColors2[(5*i)-3], plotColors2[(5*i)-2], plotColors2[(5*i)-1], plotColors2[(5*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% Strong", line = 3, cex = 1.5) #Add y-axis label

#Moderate methylation
barplot(t(t(PactFeatureOverlapsPercents[,c(11,7,3)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("", "", ""),
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs.

for (i in 2:4) {
    xx <- t(t(PactFeatureOverlapsPercents[,c(11,7,3)]))
    xx[,-(i-1)] <- NA
    colnames(xx)[-(i-1)] <- NA
    barplot(xx, col = c(plotColors2[(5*i)-4], plotColors2[(5*i)-3], plotColors2[(5*i)-2], plotColors2[(5*i)-1], plotColors2[(5*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% Moderate", line = 3, cex = 1.5) #Add y-axis label

#Weak methylation
barplot(t(t(PactFeatureOverlapsPercents[,c(12,8,4)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("WGBS", "RRBS", "MBD-BS"), cex.names = 2,
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs.

for (i in 2:4) {
    xx <- t(t(PactFeatureOverlapsPercents[,c(12,8,4)]))
    xx[,-(i-1)] <- NA
    colnames(xx)[-(i-1)] <- NA
    barplot(xx, col = c(plotColors2[(5*i)-4], plotColors2[(5*i)-3], plotColors2[(5*i)-2], plotColors2[(5*i)-1], plotColors2[(5*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% Weak", line = 3, cex = 1.5) #Add y-axis label

#Add legend
par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE) #Create new plot
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n") #Add new plot on top of current plot
legend(x = 0.73, y = 0.97, 
       xpd = TRUE,
       legend = c("CDS", "Introns", "Upstream Flank", "Downstream Flank", "Intergenic"),
       pch = 22, 
        col = "black", 
        pt.bg = c("grey20", "grey40", "grey60", "grey80", "grey100"),
       bty = "n",
       cex = 1.5, 
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box
text("Genome Feature", x = 0.87, y = 0.97, cex = 1.5) #Add legend title that is aligned with legend

#dev.off() #Turn off plotting device
```

### Create comparative barplots

```{r}
PactFeatureOverlapMethProps <- data.frame("WBGS" = (PactFeatureOverlaps$WGBSMeth / PactFeatureOverlaps$allCpGs),
                                          "RRBS" = (PactFeatureOverlaps$RRBSMeth / PactFeatureOverlaps$allCpGs),
                                          "MBDBS" = (PactFeatureOverlaps$MBDBSMeth / PactFeatureOverlaps$allCpGs)) #Divide # strong meth loci by all CpGs in a feature to get proportions
PactFeatureOverlapMethProps <- PactFeatureOverlapMethProps * 100 #Multiply by 100 to get percents
row.names(PactFeatureOverlapMethProps) <- row.names(PactFeatureOverlaps) #Assign row names
PactFeatureOverlapMethProps <- PactFeatureOverlapMethProps[-c(1,4),] #Remove genes and total flanks information
head(PactFeatureOverlapMethProps) #Confirm proportions
```

```{r}
#pdf("../Output/Pact_union-StrongMeth-CpG-Features.pdf", width = 11, height = 8.5) #Save file
par(mar = c(5, 8, 1, 1),  oma = c(1, 5, 1, 1)) #Change figure boundaries

barsPact <- barplot(t(PactFeatureOverlapMethProps),
                    beside = TRUE, horiz = TRUE,
                    col = c(plotColors2[7], plotColors2[12], plotColors2[17]),
                    xlim = c(0,20),
                    axes = FALSE, names.arg = rep("", times = 5)) #Create a horizontal barplot with comparative proportions. Color based on sequencing method and do not include any axes. Use blanks as y-axis labels. Save barplot information.

axis(side = 2, at = barsPact[2,], labels = row.names(PactFeatureOverlapMethProps), tick = FALSE, las = 2, col = "grey80", cex.axis = 1.5) #Add y-axis labels at the center of grouped bars. Remove tick marks and change label orientation.

axis(side = 1, at = seq(from = 0, to = 20, by = 2), cex = 1.2, col = "grey80") #Add x-axis
mtext(side = 1, "Strongly Methylated CpGs/All CpGs (%)", line = 3, cex = 1.5) #Add x-axis label

legend("topright",
       xpd = TRUE,
       legend = c("WGBS", "RRBS", "MBD-BS"),
       pch = 22, 
        col = "black", 
        pt.bg = c(plotColors2[7], plotColors2[12], plotColors2[17]),
       bty = "n",
       cex = 1.5,
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box

#dev.off()
```

```{r}
#pdf("../Output/Pact_union-StrongMeth-CpG-Features-NoFlanks.pdf", width = 11, height = 8.5) #Save file
par(mar = c(5, 8, 1, 1),  oma = c(1, 0, 1, 1)) #Change figure boundaries

barsPact <- barplot(t(PactFeatureOverlapMethProps[-c(3,4),]),
                    beside = TRUE, horiz = TRUE,
                    col = c(plotColors2[7], plotColors2[12], plotColors2[17]),
                    xlim = c(0,20),
                    axes = FALSE, names.arg = rep("", times = 3)) #Create a horizontal barplot with comparative proportions. Color based on sequencing method and do not include any axes. Use blanks as y-axis labels. Save barplot information.

axis(side = 2, at = barsPact[2,], labels = row.names(PactFeatureOverlapMethProps[-c(3,4),]), tick = FALSE, las = 2, col = "grey80", cex.axis = 1.5) #Add y-axis labels at the center of grouped bars. Remove tick marks and change label orientation.

axis(side = 1, at = seq(from = 0, to = 20, by = 2), cex = 1.2, col = "grey80") #Add x-axis
mtext(side = 1, "Strongly Methylated CpGs/All CpGs (%)", line = 3, cex = 1.5) #Add x-axis label

legend("topright",
       xpd = TRUE,
       legend = c("WGBS", "RRBS", "MBD-BS"),
       pch = 22, 
        col = "black", 
        pt.bg = c(plotColors2[7], plotColors2[12], plotColors2[17]),
       bty = "n",
       cex = 1.5,
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box

#dev.off()
```

## Interspecies chi-squared tests

The hypothesis is that the distribution of feature overlaps will be the same between species for the same methods used.

#### All

```{r}
AllFeature <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactFeatureStatTest)) { #For each feature overlap
  Species1Feature <- McapFeatureStatTest[1,i] #Variable for # feature overlap for Species 1
  Species2Feature <- PactFeatureStatTest[1,i] #Variable for # feature overlap for Species 2
  Species1NotFeature <- sum(McapFeatureStatTest[1,-i]) #Variable for # other feature overlaps for Species 1
  Species2NotFeature <- sum(PactFeatureStatTest[1,-i]) #Variable for # other feature overlaps for Species 2
  ct <- matrix(c(Species1Feature,Species2Feature,Species1NotFeature,Species2NotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactFeatureStatTest[i])), paste0("Not", colnames(PactFeatureStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c("Mcap", "Pact") #Assign row names: Species 1, Species 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$Feature <- as.character(colnames(PactFeatureStatTest)[i]) #Add feature overlap to results
  AllFeature <- rbind(AllFeature, ctResults) #Add test statistics to master table
}
```

```{r}
AllFeature$p.adj <- p.adjust(AllFeature$p.value, method = "fdr") #Correct p-value using FDR
AllFeature$comparison <- rep("Genome", times = 5) #Add methods compared
head(AllFeature) #Confirm changes
```

#### WGBS

```{r}
WGBSFeature <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactFeatureStatTest)) { #For each feature overlap
  Species1Feature <- McapFeatureStatTest[2,i] #Variable for # feature overlap for Species 1
  Species2Feature <- PactFeatureStatTest[2,i] #Variable for # feature overlap for Species 2
  Species1NotFeature <- sum(McapFeatureStatTest[2,-i]) #Variable for # other feature overlaps for Species 1
  Species2NotFeature <- sum(PactFeatureStatTest[2,-i]) #Variable for # other feature overlaps for Species 2
  ct <- matrix(c(Species1Feature,Species2Feature,Species1NotFeature,Species2NotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactFeatureStatTest[i])), paste0("Not", colnames(PactFeatureStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c("Mcap", "Pact") #Assign row names: Species 1, Species 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$Feature <- as.character(colnames(PactFeatureStatTest)[i]) #Add feature overlap to results
  WGBSFeature <- rbind(WGBSFeature, ctResults) #Add test statistics to master table
}
```

```{r}
WGBSFeature$p.adj <- p.adjust(WGBSFeature$p.value, method = "fdr") #Correct p-value using FDR
WGBSFeature$comparison <- rep("WGBS", times = 5) #Add methods compared
head(WGBSFeature) #Confirm changes
```

#### RRBS

```{r}
RRBSFeature <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactFeatureStatTest)) { #For each feature overlap
  Species1Feature <- McapFeatureStatTest[3,i] #Variable for # feature overlap for Species 1
  Species2Feature <- PactFeatureStatTest[3,i] #Variable for # feature overlap for Species 2
  Species1NotFeature <- sum(McapFeatureStatTest[3,-i]) #Variable for # other feature overlaps for Species 1
  Species2NotFeature <- sum(PactFeatureStatTest[3,-i]) #Variable for # other feature overlaps for Species 2
  ct <- matrix(c(Species1Feature,Species2Feature,Species1NotFeature,Species2NotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactFeatureStatTest[i])), paste0("Not", colnames(PactFeatureStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c("Mcap", "Pact") #Assign row names: Species 1, Species 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$Feature <- as.character(colnames(PactFeatureStatTest)[i]) #Add feature overlap to results
  RRBSFeature <- rbind(RRBSFeature, ctResults) #Add test statistics to master table
}
```

```{r}
RRBSFeature$p.adj <- p.adjust(RRBSFeature$p.value, method = "fdr") #Correct p-value using FDR
RRBSFeature$comparison <- rep("RRBS", times = 5) #Add methods compared
head(RRBSFeature) #Confirm changes
```

#### MBD-BS

```{r}
MBDBSFeature <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactFeatureStatTest)) { #For each feature overlap
  Species1Feature <- McapFeatureStatTest[4,i] #Variable for # feature overlap for Species 1
  Species2Feature <- PactFeatureStatTest[4,i] #Variable for # feature overlap for Species 2
  Species1NotFeature <- sum(McapFeatureStatTest[4,-i]) #Variable for # other feature overlaps for Species 1
  Species2NotFeature <- sum(PactFeatureStatTest[4,-i]) #Variable for # other feature overlaps for Species 2
  ct <- matrix(c(Species1Feature,Species2Feature,Species1NotFeature,Species2NotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactFeatureStatTest[i])), paste0("Not", colnames(PactFeatureStatTest[i]))) #Assign column names: type, not type
  rownames(ct) <- c("Mcap", "Pact") #Assign row names: Species 1, Species 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$Feature <- as.character(colnames(PactFeatureStatTest)[i]) #Add feature overlap to results
  MBDBSFeature <- rbind(MBDBSFeature, ctResults) #Add test statistics to master table
}
```

```{r}
MBDBSFeature$p.adj <- p.adjust(MBDBSFeature$p.value, method = "fdr") #Correct p-value using FDR
MBDBSFeature$comparison <- rep("MBDBS", times = 5) #Add methods compared
head(MBDBSFeature) #Confirm changes
```

#### Save statistical test results

```{r}
InterspeciesFeatureStatResults <- rbind(AllFeature,
                                        WGBSFeature,
                                        RRBSFeature,
                                        MBDBSFeature) #Combine tables
tail(InterspeciesFeatureStatResults) #Confirm changes
```

```{r}
write.table(InterspeciesFeatureStatResults, "../analyses/Characterizing-CpG-Methylation-5x-Union/Interspecies_union-Feature-StatResults.txt", quote = FALSE, row.names = FALSE) #Save table
```

## Create multipanel plot

```{r}
#pdf("../Output/Union-CpG-Features-Multipanel.pdf", height = 8.5, width = 11)
par(mar = c(1,5,1,0), oma = c(2, 1, 0, 12), mfrow = c(2,1)) #Change figure boundaries

#Mcap
barplot(t(t(McapFeatureOverlapsPercents[,c(1,13,9,5)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("", "", "", ""), cex.names = 1.5,
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs. 

for (i in 1:4) {
    xx <- t(t(McapFeatureOverlapsPercents[,c(1,13,9,5)]))
    xx[,-i] <- NA
    colnames(xx)[-i] <- NA
    barplot(xx, col = c(plotColors2[(5*i)-4], plotColors2[(5*i)-3], plotColors2[(5*i)-2], plotColors2[(5*i)-1], plotColors2[(5*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% CpGs", line = 3, cex = 1.5) #Add y-axis label
text(expression(paste("a. ", italic("M. capitata"))), x = 0.06, y = 105, cex = 1.2, adj = 0) #Add italics species name

#Pact
barplot(t(t(PactFeatureOverlapsPercents[,c(1,13,9,5)])), 
        col= "white", 
        axes = FALSE, 
        names.arg = c("Genome", "WGBS", "RRBS", "MBD-BS"), cex.names = 1.5,
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs. 

for (i in 1:4) {
    xx <- t(t(PactFeatureOverlapsPercents[,c(1,13,9,5)]))
    xx[,-i] <- NA
    colnames(xx)[-i] <- NA
    barplot(xx, col = c(plotColors2[(5*i)-4], plotColors2[(5*i)-3], plotColors2[(5*i)-2], plotColors2[(5*i)-1], plotColors2[(5*i)]), add = TRUE, axes = FALSE, names.arg = c("", "", "", "")) 
} #Make each bar in the stacked barplot a different color gradient. Create a new dataframe with data for the plot and ignore all additional columns and column names. Plot the single bar with the correct color from plotColors. Do not add axes and add blank names for each column.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% CpGs", line = 3, cex = 1.5) #Add y-axis label
text(expression(paste("b. ", italic("P. acuta"))), x = 0.06, y = 105, cex = 1.2, adj = 0) #Add italics species name

#Add legend
par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE) #Create new plot
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n") #Add new plot on top of current plot
legend(x = 0.57, y = 0.91, 
       xpd = TRUE,
       legend = c("CDS", "Introns", "Upstream Flank", "Downstream Flank", "Intergenic"),
       pch = 22, 
        col = "black", 
        pt.bg = c("grey20", "grey40", "grey60", "grey80", "grey100"),
       bty = "n",
       cex = 1.5, 
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box
text("Genome Feature", x = 0.785, y = 0.92, cex = 1.5) #Add legend title that is aligned with legend

#dev.off() #Turn off plotting device
```

