---
title: "Expression_Methylation"
author: "HM Putnam"
date: "6/12/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)

#source("https://bioconductor.org/biocLite.R")
library("pheatmap")
library("RColorBrewer")
library("tidyverse")
library("GSEABase")
library("cowplot")
library("lattice")
library("latticeExtra")
library("ggforce")
library("goseq")
library("ggpubr")

```

Goal is to identify and compare presence and proportion of CpG in genes captured by each method and to compare GOslim terms between each method

Load sample info, annotation, and counts
```{r load data}
#load sample info
sample.info <- read.csv(file="metadata/RNASeq_metadata.csv", header=T, sep=",", row.names=1)
#separate sample info by species
Mcap.sample.info <- subset(sample.info, Species == "Montipora capitata")
#Mcap = Montipora capitata
Pact.sample.info <- subset(sample.info, Species == "Pocillopora acuta")
#Pact = Pocillopora acuta

#Load Mcap Annotation Info
Mcap.annot <-  read.csv(file="genome-feature-files/Mcap-GO-KO-Kegg.tab", header=FALSE, sep="\t") 
colnames(Mcap.annot) <- c("Uniprot", "gene", "eval", "Prot.ID", "Rev", "Prot.Name.Long", "Prot.Name.Short", "Taxa", "Num", "GO1", "GO2", "GO3", "GO4", "GO.IDs","KEGG", "KEGG.Path")  
Mcap.annot <- Mcap.annot %>% 
  distinct(gene, .keep_all = TRUE) #keep only distinct gene ids
Mcap.annot$gene <- gsub("augustus.", "", Mcap.annot$gene) #remove excess characters from gene name
Mcap.annot$gene <- gsub(".t1", "", Mcap.annot$gene) #remove excess characters from gene name

#Identify Mcap full set of GO terms
Mcap.GO <- Mcap.annot[,c(2,14)] #identify GOids
splitted <- strsplit(as.character(Mcap.GO$GO.IDs), "; ") #slit into multiple GO ids
Mcap.GOs <- data.frame(v1 = rep.int(Mcap.GO$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all GOs with each gene
Mcap.GO.terms <- Mcap.GOs %>% distinct(v2, .keep_all = TRUE) #keep only distinct GO terms
length(unique(Mcap.GO.terms$v2)) #view number of go terms

#Load Pact Annotation Info
Pact.annot <-  read.csv(file="genome-feature-files/Pact-GO-KO-Kegg.tab", header=FALSE, sep="\t") 
colnames(Pact.annot) <- c("Uniprot", "gene", "eval", "Prot.ID", "Rev", "Prot.Name.Long", "Prot.Name.Short", "Taxa", "Num", "GO1", "GO2", "GO3", "GO4", "GO.IDs","KEGG", "KEGG.Path")  
Pact.annot <- Pact.annot %>% 
  distinct(gene, .keep_all = TRUE) #keep only distinct gene ids

#Identify Mcap full set of GO terms
Pact.GO <- Pact.annot[,c(2,14)] #identify GOids
splitted <- strsplit(as.character(Pact.GO$GO.IDs), "; ") #slit into multiple GO ids
Pact.GOs <- data.frame(v1 = rep.int(Pact.GO$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all GOs with each gene
Pact.GO.terms <- Pact.GOs %>% distinct(v2, .keep_all = TRUE) #keep only distinct GO terms
length(unique(Pact.GO.terms$v2)) #view number of go terms

#Load Mcap gene gff
Mcap.genes <- read.csv(file="genome-feature-files/Mcap.GFFannotation.gff", header=FALSE, sep="\t") 
Mcap.genes <- filter(Mcap.genes, V3 == "gene")
#load sample info
colnames(Mcap.genes) <- c("scaffold", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
Mcap.genes$gene <- gsub("ID=", "", Mcap.genes$gene)
Mcap.genes$gene <- gsub(";.*", "", Mcap.genes$gene)

Mcap.gene.len.df <- Mcap.genes 
Mcap.gene.len.df$length <- Mcap.gene.len.df$gene.stop-Mcap.gene.len.df$gene.start


#write output of gene gff only
write.table(Mcap.genes, file="genome-feature-files/Mcap.GFFannotation.genes.gff", sep="\t", 
            col.names = FALSE, row.names=FALSE, quote=FALSE)
Mcap.genes <- Mcap.genes[,c(1,4,5,9)]

#Load Pact gene gff
Pact.genes <- read.csv(file="genome-feature-files/Pact_Structural_annotation_abintio.gff", header=FALSE, sep="\t", skip=22) #load sample info
Pact.genes <- filter(Pact.genes, V3 == "gene")
Pact.genes$V1 <- gsub("cov", "_cov", Pact.genes$V1)
write.table(Pact.genes, file="genome-feature-files/Pact.GFFannotation.genes.gff", sep="\t", 
            col.names = FALSE, row.names=FALSE, quote=FALSE)
Pact.genes <- Pact.genes[,c(1,4,5,9)]
colnames(Pact.genes) <- c("scaffold",  "gene.start","gene.stop", "gene")

Pact.gene.len.df <- Pact.genes 
Pact.gene.len.df$length <- Pact.gene.len.df$gene.stop-Pact.gene.len.df$gene.start

# #load expression data
# #Montipora capitata counts
# Mcap.1101 <- read.csv(file="RNASeq/PolyA/1101_gene_abund.tab", header=TRUE, sep="\t") #Load expression matrix
# colnames(Mcap.1101)[c(8,9)] <- c("Mcap.1101.FPKM", "Mcap.1101.TPM")
# Mcap.1548 <- read.csv(file="RNASeq/PolyA/1548_gene_abund.tab", header=TRUE, sep="\t") #Load expression matrix
# colnames(Mcap.1548)[c(8,9)] <- c("Mcap.1548.FPKM", "Mcap.1548.TPM")
# Mcap.1628 <- read.csv(file="RNASeq/PolyA/1628_gene_abund.tab", header=TRUE, sep="\t") #Load expression matrix
# colnames(Mcap.1628)[c(8,9)] <- c("Mcap.1628.FPKM", "Mcap.1628.TPM")
# 
# Mcap.counts <- left_join(Mcap.1101,Mcap.1548, by="Gene.ID" )
# Mcap.counts <- left_join(Mcap.counts,Mcap.1628, by="Gene.ID" )
# Mcap.counts <- Mcap.counts[,c(1,3,5,6,8,9,16,17, 24,25)]
# Mcap.counts$mean.FPKM <- rowMeans(Mcap.counts[,c(5,7,9)])
# Mcap.counts$mean.TPM <- rowMeans(Mcap.counts[,c(6,8,10)])
# colnames(Mcap.counts) <- c("gene", "scaffold", "start", "stop",  "Mcap.1101.FPKM", "Mcap.1101.TPM", "Mcap.1548.FPKM", "Mcap.1548.TPM",
#                            "Mcap.1628.FPKM", "Mcap.1628.TPM", "mean.FPKM", "mean.TPM" )
# range(Mcap.counts$mean.FPKM)
# mean(Mcap.counts$mean.FPKM)
# range(Mcap.counts$mean.TPM)
# mean(Mcap.counts$mean.TPM)
# 
# #Pocillopora acuta counts
# Pact.1041 <- read.csv(file="RNASeq/PolyA/1041_gene_abund.tab", header=TRUE, sep="\t") #Load expression matrix
# colnames(Pact.1041)[c(8,9)] <- c("Pact.1041.FPKM", "Pact.1041.TPM")
# Pact.1471 <- read.csv(file="RNASeq/PolyA/1471_gene_abund.tab", header=TRUE, sep="\t") #Load expression matrix
# colnames(Pact.1471)[c(8,9)] <- c("Pact.1471.FPKM", "Pact.1471.TPM")
# Pact.1637 <- read.csv(file="RNASeq/PolyA/1637_gene_abund.tab", header=TRUE, sep="\t") #Load expression matrix
# colnames(Pact.1637)[c(8,9)] <- c("Pact.1637.FPKM", "Pact.1637.TPM")
# 
# Pact.counts <- left_join(Pact.1041,Pact.1471, by="Gene.ID" )
# Pact.counts <- left_join(Pact.counts,Pact.1637, by="Gene.ID" )
# Pact.counts <- Pact.counts[,c(1,3,5,6,8,9,16,17, 24,25)]
# Pact.counts$mean.FPKM <- rowMeans(Pact.counts[,c(5,7,9)])
# Pact.counts$mean.TPM <- rowMeans(Pact.counts[,c(6,8,10)])
# colnames(Pact.counts) <- c("gene", "scaffold", "start", "stop",  "Pact.1041.FPKM", "Pact.1041.TPM", "Pact.1548.FPKM", "Pact.1548.TPM",
#                            "Pact.1628.FPKM", "Pact.1628.TPM", "mean.FPKM", "mean.TPM" )
# range(Pact.counts$mean.FPKM)
# mean(Pact.counts$mean.FPKM)
# range(Pact.counts$mean.TPM)
# mean(Pact.counts$mean.TPM)


# Load ortholog data
Ortho.data <- read.csv(file="analyses/coral_ortho/Fastas/OrthoFinder/Results_Jul06/Orthologues/Orthologues_Mcap.protein/Mcap.protein__v__Pocillopora_acuta_protein.tsv", header=TRUE, sep="\t")

#remove extra gene id characters
Ortho.data$Mcap.protein <- gsub("augustus.", "", Ortho.data$Mcap.protein)
Ortho.data$Mcap.protein <- gsub(".t1", "", Ortho.data$Mcap.protein)

Ortho.data.1.1 <- Ortho.data %>% 
  filter(!grepl(',', Mcap.protein))
Ortho.data.1.1 <- Ortho.data.1.1 %>% 
  filter(!grepl(',', Pocillopora_acuta_protein))
Ortho.data.1.1$ID <- seq(from=1, to=nrow(Ortho.data.1.1), by=1)

Mcap.ortho <- Ortho.data.1.1[,c(4,2)]
colnames(Mcap.ortho) <- c("ID", "gene")
Pact.ortho <- Ortho.data.1.1[,c(4,3)]
colnames(Pact.ortho) <- c("ID", "gene")

```

Count CpG per gene sensu Dimond and Roberts 
```{bash}
#Mcap

perl -e '$count=0; $len=0; while(<>) {s/\r?\n//; s/\t/ /g; if (s/^>//) { if ($. != 1) {print "\n"} s/ |$/\t/; $count++; $_ .= "\t";} else {s/ //g; $len += length($_)} print $_;} print "\n"; warn "\nConverted $count FASTA records in $. lines to tabular format\nTotal sequence length: $len\n\n";' \ data/fasta/Montipora_capitata_coding.fasta > data/fasta/Mcap_fasta2tab

perl -e '$col = 2;' -e 'while (<>) { s/\r?\n//; @F = split /\t/, $_; $len = length($F[$col]); print "$_\t$len\n" } warn "\nAdded column with length of column $col for $. lines.\n\n";' \ data/fasta/Mcap_fasta2tab > data/fasta/Mcap_tab_1

awk '{print $2}' data/fasta/Mcap_tab_1 > data/fasta/Mcap_tab_2

echo "CG" | awk -F\[Cc][Gg] '{print NF-1}' data/fasta/Mcap_tab_2 > data/fasta/Mcap_CG

awk '{print $1}' data/fasta/Mcap_tab_1 > data/fasta/Mcap_gene_names

paste data/fasta/Mcap_gene_names data/fasta/Mcap_CG > data/fasta/Mcap_CpG_per_gene


#Pact

perl -e '$count=0; $len=0; while(<>) {s/\r?\n//; s/\t/ /g; if (s/^>//) { if ($. != 1) {print "\n"} s/ |$/\t/; $count++; $_ .= "\t";} else {s/ //g; $len += length($_)} print $_;} print "\n"; warn "\nConverted $count FASTA records in $. lines to tabular format\nTotal sequence length: $len\n\n";' \ data/fasta/Pocillopora_acuta_coding.fasta > data/fasta/Pact_fasta2tab

perl -e '$col = 2;' -e 'while (<>) { s/\r?\n//; @F = split /\t/, $_; $len = length($F[$col]); print "$_\t$len\n" } warn "\nAdded column with length of column $col for $. lines.\n\n";' \ data/fasta/Pact_fasta2tab > data/fasta/Pact_tab_1

awk '{print $2}' data/fasta/Pact_tab_1 > data/fasta/Pact_tab_2

echo "CG" | awk -F\[Cc][Gg] '{print NF-1}' data/fasta/Pact_tab_2 > data/fasta/Pact_CG

awk '{print $1}' data/fasta/Pact_tab_1 > data/fasta/Pact_gene_names

paste data/fasta/Pact_gene_names data/fasta/Pact_CG > data/fasta/Pact_CpG_per_gene

```

Count CpG per gene via intersectBed with genes
```{bash}

#Mcap
intersectBed \
-wb \
-a  genome-feature-files/Mcap_CpG.gff \
-b genome-feature-files/Mcap.GFFannotation.genes.gff \
> genome-feature-files/Mcap_CpG_per_gene

awk '{print $NF}' genome-feature-files/Mcap_CpG_per_gene > genome-feature-files/Mcap_CpG_per_gene1

#Pact
intersectBed \
-wb \
-a  genome-feature-files/Pact_CpG.gff \
-b genome-feature-files/Pact.GFFannotation.genes.gff \
> genome-feature-files/Pact_CpG_per_gene

awk '{print $NF}' genome-feature-files/Pact_CpG_per_gene > genome-feature-files/Pact_CpG_per_gene1


```

Obtain the genes from both genomes that have CpG data with 5x coverage
```{bash}
#Intersect Mcap CpG with genes
for f in data/Mcap_tab/*5x.tab
do
intersectBed \
-wb \
-a ${f} \
-b genome-feature-files/Mcap.GFFannotation.genes.gff \
> ${f}_gene
done

#Intersect Pact CpG with genes
for f in data/Pact_tab/*5x.tab
do
intersectBed \
-wb \
-a ${f} \
-b genome-feature-files/Pact.GFFannotation.genes.gff \
> ${f}_gene
done
```

Load M. capitata (Mcap) Methylation data and ID genes merge with one to one Orthologs
```{r}
#load methylation data with minimum of 5x coverage of CpGs that has intersected with genes
#1101 1548 1628
WGBS.1101 <- read.csv("data/Mcap_tab/Meth10_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(WGBS.1101) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
WGBS.1101$rep <- "WGBS.1101"
WGBS.1548 <- read.csv("data/Mcap_tab/Meth11_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(WGBS.1548) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
WGBS.1548$rep <- "WGBS.1548"
WGBS.1628 <- read.csv("data/Mcap_tab/Meth12_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(WGBS.1628) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
WGBS.1628$rep <- "WGBS.1628"

#Determine the number of CpGs per gene for the orthologs
# WGBS samples
WGBS.data <- rbind(WGBS.1101,WGBS.1548,WGBS.1628) #combine samples into df by method
WGBS.data$num <- 1 #add a column of 1's so they can be used to sum the number of CpGs per gene per replicate
WGBS.CpG.by.Gene <- WGBS.data %>%
  group_by(gene, rep) %>%
  summarise(CpG.Count = sum(num)) #Sum CpG per gene per rep
WGBS.CpG.by.Gene <- WGBS.CpG.by.Gene %>%
  group_by(gene) %>%
  summarise(mean.CpG.Count = mean(CpG.Count)) #mean CpG per gene

WGBS.CpG.by.Gene <- merge(WGBS.CpG.by.Gene, Mcap.genes, by="gene", all = TRUE) #merge with all genes to have full list available

Mcap.CpG.possible <- read.csv("genome-feature-files/Mcap_CpG_per_gene1", header=FALSE, sep="\t") 
Mcap.CpG.possible$count <- 1 
colnames(Mcap.CpG.possible) <- c("gene", "count")
Mcap.CpG.possible.sum <- Mcap.CpG.possible %>%
  group_by(gene) %>%
  summarise(CpG.Poss = sum(count)) #Sum CpG per gene
WGBS.CpG.by.Gene <- merge(WGBS.CpG.by.Gene, Mcap.CpG.possible.sum, by="gene", all = TRUE) #merge to keep orthologs and possible CpG per gene
WGBS.CpG.by.Gene <- merge(WGBS.CpG.by.Gene, Mcap.ortho, by="gene", all = FALSE) #merge to keep orthologs only
WGBS.CpG.by.Gene$species <- "M.capitata" #add species
WGBS.CpG.by.Gene$method <- "WGBS" #add method
WGBS.CpG.by.Gene <- arrange(WGBS.CpG.by.Gene,scaffold, gene.start) #order by orthogroup

#RRBS samples
RRBS.1101 <- read.csv("data/Mcap_tab/Meth13_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(RRBS.1101) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
RRBS.1101$rep <- "RRBS.1101"
RRBS.1548 <- read.csv("data/Mcap_tab/Meth14_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(RRBS.1548) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
RRBS.1548$rep <- "RRBS.1548"
RRBS.1628 <- read.csv("data/Mcap_tab/Meth15_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(RRBS.1628) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
RRBS.1628$rep <- "RRBS.1628"

#Determine the number of CpGs per gene for the orthologs
# RRBS samples
RRBS.data <- rbind(RRBS.1101,RRBS.1548,RRBS.1628) #combine samples into df by method
RRBS.data$num <- 1 #add a column of 1's so they can be used to sum the number of CpGs per gene
RRBS.CpG.by.Gene <- RRBS.data %>%
  group_by(gene, rep) %>%
  summarise(CpG.Count = sum(num)) #Sum CpG per gene per rep
RRBS.CpG.by.Gene <- RRBS.CpG.by.Gene %>%
  group_by(gene) %>%
  summarise(mean.CpG.Count = mean(CpG.Count)) #mean CpG per gene
RRBS.CpG.by.Gene <- merge(RRBS.CpG.by.Gene, Mcap.genes, by="gene", all = TRUE) #merge with all genes to have full list available
RRBS.CpG.by.Gene <- merge(RRBS.CpG.by.Gene, Mcap.CpG.possible.sum, by="gene", all = TRUE) #merge to keep orthologs and possible CpG per gene
RRBS.CpG.by.Gene <- merge(RRBS.CpG.by.Gene, Mcap.ortho, by="gene", all = FALSE) #merge to keep orthologs only
RRBS.CpG.by.Gene$species <- "M.capitata" #add species
RRBS.CpG.by.Gene$method <- "RRBS" #add method
RRBS.CpG.by.Gene <- arrange(RRBS.CpG.by.Gene,scaffold, gene.start) #order by orthogroup

#load MBDBS data
MBDBS.1101 <- read.csv("data/Mcap_tab/Meth16_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(MBDBS.1101) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
MBDBS.1101$rep <- "MBDBS.1101"
MBDBS.1548 <- read.csv("data/Mcap_tab/Meth17_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(MBDBS.1548) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
MBDBS.1548$rep <- "MBDBS.1548"
MBDBS.1628 <- read.csv("data/Mcap_tab/Meth18_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(MBDBS.1628) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
MBDBS.1628$rep <- "MBDBS.1628"

#Determine the number of CpGs per gene for the orthologs
# MBDBS samples
MBDBS.data <- rbind(MBDBS.1101,MBDBS.1548,MBDBS.1628) #combine samples into df by method
MBDBS.data$num <- 1 #add a column of 1's so they can be used to sum the number of CpGs per gene
MBDBS.CpG.by.Gene <- MBDBS.data %>%
  group_by(gene, rep) %>%
  summarise(CpG.Count = sum(num)) #Sum CpG per gene per rep
MBDBS.CpG.by.Gene <- MBDBS.CpG.by.Gene %>%
  group_by(gene) %>%
  summarise(mean.CpG.Count = mean(CpG.Count)) #mean CpG per gene
MBDBS.CpG.by.Gene <- merge(MBDBS.CpG.by.Gene, Mcap.genes, by="gene", all = TRUE) #merge with all genes to have full list available
MBDBS.CpG.by.Gene <- merge(MBDBS.CpG.by.Gene, Mcap.CpG.possible.sum, by="gene", all = TRUE) #merge to keep orthologs and possible CpG per gene
MBDBS.CpG.by.Gene <- merge(MBDBS.CpG.by.Gene, Mcap.ortho, by="gene", all = FALSE) #merge to keep orthologs only
MBDBS.CpG.by.Gene$species <- "M.capitata" #add species
MBDBS.CpG.by.Gene$method <- "MBDBS" #add method
MBDBS.CpG.by.Gene <- arrange(MBDBS.CpG.by.Gene,scaffold, gene.start) #order by orthogroup

#combine CpG per gene for orthologs dataframe
Ortho.data.counts <- rbind(WGBS.CpG.by.Gene, RRBS.CpG.by.Gene, MBDBS.CpG.by.Gene)
# NA remains for those genes that are orthologs but have no CpG data
Ortho.data.counts$prop <- Ortho.data.counts$mean.CpG.Count/Ortho.data.counts$CpG.Poss
range(na.omit(Ortho.data.counts$prop))
which(Ortho.data.counts$prop ==0)

pdf("Output/Mcap_CpGperGeneProportion.1to1.Orthologs.pdf", height=3, width=8)
MC <- ggplot(Ortho.data.counts, aes(ID, method, fill=prop, width = 1)) + 
  geom_tile() +
  scale_fill_gradient2(low="cyan", mid="magenta", high="yellow", na.value = "black", midpoint=0.5)+
  theme(axis.text.x = element_text(angle = 90, size=1)) +
  ggtitle("A M. capitata")
MC
dev.off()

```

Load P. acuta (Pact) Methylation data and ID genes merge with one to one Orthologs
```{r}
#load methylation data with minimum of 5x coverage of CpGs that has intersected with genes
#1041 1471 1637
WGBS.1041 <- read.csv("data/Pact_tab/Meth1_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(WGBS.1041) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
WGBS.1041$rep <- "WGBS.1041"
WGBS.1471 <- read.csv("data/Pact_tab/Meth2_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(WGBS.1471) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
WGBS.1471$rep <- "WGBS.1471"
WGBS.1637 <- read.csv("data/Pact_tab/Meth3_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(WGBS.1637) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
WGBS.1637$rep <- "WGBS.1637"

#Determine the number of CpGs per gene for the orthologs
# WGBS samples
Pact.WGBS.data <- rbind(WGBS.1041,WGBS.1471,WGBS.1637) #combine samples into df by method
Pact.WGBS.data$num <- 1 #add a column of 1's so they can be used to sum the number of CpGs per gene per replicate
Pact.WGBS.CpG.by.Gene <-Pact.WGBS.data %>%
  group_by(gene, rep) %>%
  summarise(CpG.Count = sum(num)) #Sum CpG per gene per rep
Pact.WGBS.CpG.by.Gene <- Pact.WGBS.CpG.by.Gene %>%
  group_by(gene) %>%
  summarise(mean.CpG.Count = mean(CpG.Count)) #mean CpG per gene

Pact.WGBS.CpG.by.Gene <- merge(Pact.WGBS.CpG.by.Gene, Pact.genes, by="gene", all = TRUE) #merge with all genes to have full list available

Pact.CpG.possible <- read.csv("genome-feature-files/Pact_CpG_per_gene1", header=FALSE, sep="\t") 
Pact.CpG.possible$count <- 1 
colnames(Pact.CpG.possible) <- c("gene", "count")
Pact.CpG.possible.sum <- Pact.CpG.possible %>%
  group_by(gene) %>%
  summarise(CpG.Poss = sum(count)) #Sum CpG per gene
Pact.WGBS.CpG.by.Gene <- merge(Pact.WGBS.CpG.by.Gene, Pact.CpG.possible.sum, by="gene", all = TRUE) #merge to keep orthologs and possible CpG per gene
Pact.WGBS.CpG.by.Gene <- merge(Pact.WGBS.CpG.by.Gene, Pact.ortho, by="gene", all = FALSE) #merge to keep orthologs only
Pact.WGBS.CpG.by.Gene$species <- "P.acuta" #add species
Pact.WGBS.CpG.by.Gene$method <- "WGBS" #add method
Pact.WGBS.CpG.by.Gene <- arrange(Pact.WGBS.CpG.by.Gene,scaffold, gene.start) #order by orthogroup

#RRBS samples
RRBS.1041 <- read.csv("data/Pact_tab/Meth4_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(RRBS.1041) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
RRBS.1041$rep <- "RRBS.1041"
RRBS.1471 <- read.csv("data/Pact_tab/Meth5_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(RRBS.1471) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
RRBS.1471$rep <- "RRBS.1471"
RRBS.1637 <- read.csv("data/Pact_tab/Meth6_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(RRBS.1637) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
RRBS.1637$rep <- "RRBS.1637"

#Determine the number of CpGs per gene for the orthologs
# RRBS samples
Pact.RRBS.data <- rbind(RRBS.1041,RRBS.1471,RRBS.1637) #combine samples into df by method
Pact.RRBS.data$num <- 1 #add a column of 1's so they can be used to sum the number of CpGs per gene
Pact.RRBS.CpG.by.Gene <- Pact.RRBS.data %>%
  group_by(gene, rep) %>%
  summarise(CpG.Count = sum(num)) #Sum CpG per gene per rep
Pact.RRBS.CpG.by.Gene <- Pact.RRBS.CpG.by.Gene %>%
  group_by(gene) %>%
  summarise(mean.CpG.Count = mean(CpG.Count)) #mean CpG per gene
Pact.RRBS.CpG.by.Gene <- merge(Pact.RRBS.CpG.by.Gene, Pact.genes, by="gene", all = TRUE) #merge with all genes to have full list available
Pact.RRBS.CpG.by.Gene <- merge(Pact.RRBS.CpG.by.Gene, Pact.CpG.possible.sum, by="gene", all = TRUE) #merge to keep orthologs and possible CpG per gene
Pact.RRBS.CpG.by.Gene <- merge(Pact.RRBS.CpG.by.Gene, Pact.ortho, by="gene", all = FALSE) #merge to keep orthologs only
Pact.RRBS.CpG.by.Gene$species <- "P.acuta" #add species
Pact.RRBS.CpG.by.Gene$method <- "RRBS" #add method
Pact.RRBS.CpG.by.Gene <- arrange(Pact.RRBS.CpG.by.Gene,scaffold, gene.start) #order by orthogroup

#load MBDBS data
MBDBS.1041 <- read.csv("data/Pact_tab/Meth7_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(MBDBS.1041) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
MBDBS.1041$rep <- "MBDBS.1041"
MBDBS.1471 <- read.csv("data/Pact_tab/Meth8_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(MBDBS.1471) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
MBDBS.1471$rep <- "MBDBS.1471"
MBDBS.1637 <- read.csv("data/Pact_tab/Meth9_R1_001_val_1_bismark_bt2_pe._5x.tab_gene", header=FALSE, sep="\t")
colnames(MBDBS.1637) <- c("scaffold", "start", "stop", "permeth","meth", "unmeth", "intersection", "AUGUSTUS", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
MBDBS.1637$rep <- "MBDBS.1637"

#Determine the number of CpGs per gene for the orthologs
# MBDBS samples
Pact.MBDBS.data <- rbind(MBDBS.1041,MBDBS.1471,MBDBS.1637) #combine samples into df by method
Pact.MBDBS.data$num <- 1 #add a column of 1's so they can be used to sum the number of CpGs per gene
Pact.MBDBS.CpG.by.Gene <- Pact.MBDBS.data %>%
  group_by(gene, rep) %>%
  summarise(CpG.Count = sum(num)) #Sum CpG per gene per rep
Pact.MBDBS.CpG.by.Gene <- Pact.MBDBS.CpG.by.Gene %>%
  group_by(gene) %>%
  summarise(mean.CpG.Count = mean(CpG.Count)) #mean CpG per gene
Pact.MBDBS.CpG.by.Gene <- merge(Pact.MBDBS.CpG.by.Gene, Pact.genes, by="gene", all = TRUE) #merge with all genes to have full list available
Pact.MBDBS.CpG.by.Gene <- merge(Pact.MBDBS.CpG.by.Gene, Pact.CpG.possible.sum, by="gene", all = TRUE) #merge to keep orthologs and possible CpG per gene
Pact.MBDBS.CpG.by.Gene <- merge(Pact.MBDBS.CpG.by.Gene, Pact.ortho, by="gene", all = FALSE) #merge to keep orthologs only
Pact.MBDBS.CpG.by.Gene$species <- "P.acuta" #add species
Pact.MBDBS.CpG.by.Gene$method <- "MBDBS" #add method
Pact.MBDBS.CpG.by.Gene <- arrange(Pact.MBDBS.CpG.by.Gene,scaffold, gene.start) #order by orthogroup

#combine CpG per gene for orthologs dataframe
Pact.Ortho.data.counts <- rbind(Pact.WGBS.CpG.by.Gene, Pact.RRBS.CpG.by.Gene, Pact.MBDBS.CpG.by.Gene)
# NA remains for those genes that are orthologs but have no CpG data
Pact.Ortho.data.counts$prop <- Pact.Ortho.data.counts$mean.CpG.Count/Pact.Ortho.data.counts$CpG.Poss
range(na.omit(Pact.Ortho.data.counts$prop))
which(Pact.Ortho.data.counts$prop ==0)

pdf("Output/Pact_CpGperGeneProportion.1to1.Orthologs.pdf", height=3, width=8)
PA <- ggplot(Pact.Ortho.data.counts, aes(ID, method, fill=prop, width = 1)) + 
  geom_tile() +
  scale_fill_gradient2(low="cyan", mid="magenta", high="yellow", na.value = "black", midpoint=0.5)+
  theme(axis.text.x = element_text(angle = 90, size=1)) +
  ggtitle("B P. acuta")
PA
dev.off()

Fig <- ggarrange(MC,PA, ncol = 1, nrow = 2)
ggsave("Output/CpGperGeneProportion.1to1.Orthologs.pdf", Fig, width=6, height=6)


```

Determine Pact GO terms for the genes found with each method
```{r}
#Determine GO terms for the genes found with each method
# WGBS samples
WGBS.genes <- as.data.frame(unique(WGBS.data$gene)) #keep unique genes only
colnames(WGBS.genes) <- c("gene") #name gene column
WGBS.gene.annot <- left_join(WGBS.genes, Mcap.annot, by="gene") #combine genes and annotation
nrow(WGBS.genes)
WGBS.GO <- WGBS.gene.annot[,c(1,14)] #identify GOids
splitted <- strsplit(as.character(WGBS.GO$GO.IDs), "; ") #slit into multiple GO ids
WGBS.GOs <- data.frame(v1 = rep.int(WGBS.GO$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all GOs with each gene
WGBS.GO.terms <- WGBS.GOs %>% distinct(v2, .keep_all = TRUE) #keep only distinct GO terms
length(unique(WGBS.GO.terms$v2)) #view number of go terms

# RRBS samples
RRBS.genes <- as.data.frame(unique(RRBS.data$gene)) #keep unique genes only
colnames(RRBS.genes) <- c("gene") #name gene column
RRBS.gene.annot <- left_join(RRBS.genes, Mcap.annot, by="gene") #combine genes and annotation
nrow(RRBS.genes)
RRBS.GO <- RRBS.gene.annot[,c(1,14)] #identify GOids
splitted <- strsplit(as.character(RRBS.GO$GO.IDs), "; ") #slit into multiple GO ids
RRBS.GOs <- data.frame(v1 = rep.int(RRBS.GO$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all GOs with each gene
RRBS.GO.terms <- RRBS.GOs %>% distinct(v2, .keep_all = TRUE) #keep only distinct GO terms
length(unique(RRBS.GO.terms$v2)) #view number of go terms   

# MBDBS samples
MBDBS.genes <- as.data.frame(unique(MBDBS.data$gene)) #keep unique genes only
colnames(MBDBS.genes) <- c("gene") #name gene column
MBDBS.gene.annot <- left_join(MBDBS.genes, Mcap.annot, by="gene") #combine genes and annotation
nrow(MBDBS.genes)
MBDBS.GO <- MBDBS.gene.annot[,c(1,14)] #identify GOids
splitted <- strsplit(as.character(MBDBS.GO$GO.IDs), "; ") #slit into multiple GO ids
MBDBS.GOs <- data.frame(v1 = rep.int(MBDBS.GO$gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all GOs with each gene
MBDBS.GO.terms <- MBDBS.GOs %>% distinct(v2, .keep_all = TRUE) #keep only distinct GO terms
length(unique(MBDBS.GO.terms$v2)) #view number of go terms

```

Mcap GO Slim Term comparison in genes identifed by each method
```{r}
#GO Term union df
GO.ALL <- left_join(WGBS.GO.terms, RRBS.GO.terms, by="v1", all=TRUE)
GO.ALL <- left_join(GO.ALL , MBDBS.GO.terms, by="v1", all=TRUE)

# GO Slim
fl <- "http://www.geneontology.org/ontology/subsets/goslim_generic.obo"
slim <- getOBOCollection(fl)

WGBS.Ids <- na.omit(as.character(WGBS.GO.terms$v2))
WGBS.Collection <- GOCollection(WGBS.Ids)
WGBS.slims <- goSlim(WGBS.Collection, slim, "BP", evidenceCode="TAS")
WGBS.slims <- WGBS.slims[order(WGBS.slims$Term),]
WGBS.slims$method <- "WGBS"
WGBS.slims <- WGBS.slims[-5,]

RRBS.Ids <- na.omit(as.character(RRBS.GO.terms$v2))
RRBS.Collection <- GOCollection(RRBS.Ids)
RRBS.slims <- goSlim(RRBS.Collection, slim, "BP", evidenceCode="TAS")
RRBS.slims <- RRBS.slims[order(RRBS.slims$Term),]
RRBS.slims$method <- "RRBS"
RRBS.slims <- RRBS.slims[-5,]

MBDBS.Ids <- na.omit(as.character(MBDBS.GO.terms$v2))
MBDBS.Collection <- GOCollection(MBDBS.Ids)
MBDBS.slims <- goSlim(MBDBS.Collection, slim, "BP", evidenceCode="TAS")
MBDBS.slims <- MBDBS.slims[order(MBDBS.slims$Term),]
MBDBS.slims$method <- "MBDBS"
MBDBS.slims <- MBDBS.slims[-5,]

All.slims <- rbind(WGBS.slims,RRBS.slims,MBDBS.slims)
cat.cols <- c("springgreen4","orangered1", "skyblue3")

#plot percent goslim per method   
pdf("Output/Mcap_GOSlim_compare.pdf")
ggplot(All.slims, aes(fill=method, y=Percent, x=Term)) + 
    geom_bar(position="dodge", stat="identity") +
    scale_fill_manual(values=cat.cols) +
    theme(axis.text.x = element_text(angle = 90, size=4)) +
    theme(legend.position = "none") +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  ggtitle("A) M. capitata")
dev.off()
```




```{r}
#GO Term union df
Pact.GO.ALL <- left_join(Pact.WGBS.GO.terms, Pact.RRBS.GO.terms, by="v1", all=TRUE)
Pact.GO.ALL <- left_join(Pact.GO.ALL , Pact.MBDBS.GO.terms, by="v1", all=TRUE)

Pact.WGBS.Ids <- na.omit(as.character(Pact.WGBS.GO.terms$v2))
Pact.WGBS.Collection <- GOCollection(Pact.WGBS.Ids)
Pact.WGBS.slims <- goSlim(Pact.WGBS.Collection, slim, "BP", evidenceCode="TAS")
Pact.WGBS.slims <- Pact.WGBS.slims[order(Pact.WGBS.slims$Term),]
Pact.WGBS.slims$method <- "WGBS"
Pact.WGBS.slims <- Pact.WGBS.slims[-5,]

Pact.RRBS.Ids <- na.omit(as.character(Pact.RRBS.GO.terms$v2))
Pact.RRBS.Collection <- GOCollection(Pact.RRBS.Ids)
Pact.RRBS.slims <- goSlim(Pact.RRBS.Collection, slim, "BP", evidenceCode="TAS")
Pact.RRBS.slims <- Pact.RRBS.slims[order(Pact.RRBS.slims$Term),]
Pact.RRBS.slims$method <- "RRBS"
Pact.RRBS.slims <- Pact.RRBS.slims[-5,]

Pact.MBDBS.Ids <- na.omit(as.character(Pact.MBDBS.GO.terms$v2))
Pact.MBDBS.Collection <- GOCollection(Pact.MBDBS.Ids)
Pact.MBDBS.slims <- goSlim(Pact.MBDBS.Collection, slim, "BP", evidenceCode="TAS")
Pact.MBDBS.slims <- Pact.MBDBS.slims[order(Pact.MBDBS.slims$Term),]
Pact.MBDBS.slims$method <- "MBDBS"
Pact.MBDBS.slims <- Pact.MBDBS.slims[-5,]

Pact.All.slims <- rbind(Pact.WGBS.slims,Pact.RRBS.slims,Pact.MBDBS.slims)

#plot percent goslim per method   
pdf("Output/Pact_GOSlim_compare.pdf")
ggplot(Pact.All.slims, aes(fill=method, y=Percent, x=Term)) + 
    geom_bar(position="dodge", stat="identity") +
    scale_fill_manual(values=cat.cols) +
    theme(axis.text.x = element_text(angle = 90, size=4)) +
    theme(legend.position = "none") +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"))
dev.off() 

```






