---
title: "Identifying Genome Features"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this script, I'll create summary tables and figures for genome feature identification of various CpG categories. I'll also run statistical analyses to determine if distributions differ between these datasets.

# Install packages

```{r}
#install.packages("RColorBrewer")
require(RColorBrewer) #Load RColorBrewer

#install.packages("dichromat")
require(dichromat)
```

# Session information

```{r}
sessionInfo()
```

# DMC data

## Mcap

### Import file counts

```{r}
McapGenomeFeatures <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-CGMotif-Overlaps-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with CG motif and feature track overlaps
McapGenomeFeatures <- McapGenomeFeatures[-8,] #Remove final row
tail(McapGenomeFeatures) #Check import
```

## Pact

### Import line counts

```{r}
PactGenomeFeatures <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Pact/Pact-CGMotif-Overlaps-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with CG motif and feature track overlaps
PactGenomeFeatures <- PactGenomeFeatures[-8,] #Remove final row
tail(PactGenomeFeatures) #Check import
```

```{r}
PactDMCGeneOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paGenes-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-gene overlaps
PactDMCGeneOverlaps <- PactDMCGeneOverlaps[-4,] #Remove final row
tail(PactDMCGeneOverlaps) #Confirm import
```

```{r}
PactDMCCDSOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paCDS-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-CDS overlaps
PactDMCCDSOverlaps <- PactDMCCDSOverlaps[-4,] #Remove final row
tail(PactDMCCDSOverlaps) #Confirm import
```

```{r}
PactDMCIntronsOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paIntron-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
PactDMCIntronsOverlaps <- PactDMCIntronsOverlaps[-4,] #Remove final row
tail(PactDMCIntronsOverlaps) #Confirm import
```

```{r}
PactDMCFlanksOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paFlanks-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-flank overlaps
PactDMCFlanksOverlaps <- PactDMCFlanksOverlaps[-4,] #Remove final row
tail(PactDMCFlanksOverlaps) #Confirm import
```

```{r}
PactDMCUpstreamOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paFlanksUpstream-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-upstream flanks overlaps
PactDMCUpstreamOverlaps <- PactDMCUpstreamOverlaps[-4,] #Remove final row
tail(PactDMCUpstreamOverlaps) #Confirm import
```

```{r}
PactDMCDownstreamOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paFlanksDownstream-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-downstream flanks overlaps
PactDMCDownstreamOverlaps <- PactDMCDownstreamOverlaps[-4,] #Remove final row
tail(PactDMCDownstreamOverlaps) #Confirm import
```

```{r}
PactDMCIntergenicOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paIntergenic-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Intergenic overlaps
PactDMCIntergenicOverlaps <- PactDMCIntergenicOverlaps[-4,] #Remove final row
tail(PactDMCIntergenicOverlaps) #Confirm import
```

### Create summary tables

```{r}
PactDMCFeatureOverlaps <- cbind(PactDMCGeneOverlaps,
                                PactDMCCDSOverlaps,
                                PactDMCIntronsOverlaps,
                                PactDMCFlanksOverlaps,
                                PactDMCUpstreamOverlaps,
                                PactDMCDownstreamOverlaps,
                                PactDMCIntergenicOverlaps) #Use cbind to combine individual dataframes
PactDMCFeatureOverlaps <- PactDMCFeatureOverlaps[,c(seq(1,14,2))] #Only keep count columns
PactDMCFeatureOverlaps <- data.frame(t(PactDMCFeatureOverlaps)) #Transpose dataframe
row.names(PactDMCFeatureOverlaps) <- c("Genes", "CDS", "Introns", "Flanks", "Upstream Flanks", "Downstream Flanks", "Intergenic") #Add row names
colnames(PactDMCFeatureOverlaps) <- c("RvM", "WvM", "WvR") #Add column names based on original files
PactDMCFeatureOverlaps$allCpGs <- PactGenomeFeatures[c(5,1,7,3,4,2,6),1] #Add genome information
head(PactDMCFeatureOverlaps) #Confirm changes
```

```{r}
write.table(PactDMCFeatureOverlaps, "../analyses/Identifying-genomic-locations/Pact/Pact-DMC-Feature-Overlap-counts.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

```{r}
PactDMCFeatureOverlapsPercents <- PactDMCFeatureOverlaps[-c(1,4),] #Remove gene overlap information and save as a new dataframe
for (i in 1:length(PactDMCFeatureOverlapsPercents)) {
  PactDMCFeatureOverlapsPercents[,i] <- (PactDMCFeatureOverlapsPercents[,i] / sum(PactDMCFeatureOverlapsPercents[,i])) * 100
} #Calculate percentages
head(PactDMCFeatureOverlapsPercents) #Check calculations
```

```{r}
write.table(PactDMCFeatureOverlapsPercents, "../analyses/Identifying-genomic-locations/Pact/Pact-DMC-Feature-Overlap-percents.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

# Upset plot data

## Mcap

### Import file counts

```{r}
McapUpsetGeneOverlaps <- read.table("../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-mcGenes-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-gene overlaps
McapUpsetGeneOverlaps <- McapUpsetGeneOverlaps[-9,] #Remove final row
tail(McapUpsetGeneOverlaps) #Confirm import
```

```{r}
McapUpsetCDSOverlaps <- read.table("../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-mcCDS-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-CDS overlaps
McapUpsetCDSOverlaps <- McapUpsetCDSOverlaps[-9,] #Remove final row
tail(McapUpsetCDSOverlaps) #Confirm import
```

```{r}
McapUpsetIntronsOverlaps <- read.table("../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-mcIntrons-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
McapUpsetIntronsOverlaps <- McapUpsetIntronsOverlaps[-9,] #Remove final row
tail(McapUpsetIntronsOverlaps) #Confirm import
```

```{r}
McapUpsetFlanksOverlaps <- read.table("../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-mcFlanks-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-flank overlaps
McapUpsetFlanksOverlaps <- McapUpsetFlanksOverlaps[-9,] #Remove final row
tail(McapUpsetFlanksOverlaps) #Confirm import
```

```{r}
McapUpsetUpstreamOverlaps <- read.table("../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-mcFlanksUpstream-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-upstream flank overlaps
McapUpsetUpstreamOverlaps <- McapUpsetUpstreamOverlaps[-9,] #Remove final row
tail(McapUpsetUpstreamOverlaps) #Confirm import
```

```{r}
McapUpsetDownstreamOverlaps <- read.table("../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-mcFlanksDownstream-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-downstream flank overlaps
McapUpsetDownstreamOverlaps <- McapUpsetDownstreamOverlaps[-9,] #Remove final row
tail(McapUpsetDownstreamOverlaps) #Confirm import
```

```{r}
McapUpsetIntergenicOverlaps <- read.table("../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-mcIntergenic-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Intergenic overlaps
McapUpsetIntergenicOverlaps <- McapUpsetIntergenicOverlaps[-9,] #Remove final row
tail(McapUpsetIntergenicOverlaps) #Confirm import
```

### Create summary tables

```{r}
McapUpsetFeatureOverlaps <- cbind(McapUpsetGeneOverlaps,
                                McapUpsetCDSOverlaps,
                                McapUpsetIntronsOverlaps,
                                McapUpsetFlanksOverlaps,
                                McapUpsetUpstreamOverlaps,
                                McapUpsetDownstreamOverlaps,
                                McapUpsetIntergenicOverlaps) #Use cbind to combine individual dataframes
McapUpsetFeatureOverlaps <- McapUpsetFeatureOverlaps[,c(seq(1,14,2))] #Only keep count columns
colnames(McapUpsetFeatureOverlaps) <- c("Genes", "CDS", "Introns", "Flanks", "Upstream Flanks", "Downstream Flanks", "Intergenic") #Add column names
row.names(McapUpsetFeatureOverlaps) <- c("MBD",
                                         "RRBS.MBD",
                                         "RRBS",
                                         "WGBS.MBD",
                                         "WGBS.RRBS.MBD",
                                         "WGBS.RRBS",
                                         "WGBS",
                                         "None") #Add rownames based on original files
McapUpsetFeatureOverlaps <- data.frame(t(McapUpsetFeatureOverlaps)) #Transpose dataframe
McapUpsetFeatureOverlaps$allCpGs <- McapGenomeFeatures[c(5,1,7,3,4,2,6),1] #Add genome information
McapUpsetFeatureOverlaps <- McapUpsetFeatureOverlaps[,c(9,5,6,4,2,7,3,1,8)] #Reorganize columns
tail(McapUpsetFeatureOverlaps) #Confirm changes
```

```{r}
write.table(McapUpsetFeatureOverlaps, "../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-Feature-Overlap-Counts.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

```{r}
McapUpsetFeatureOverlapsPercents <- McapUpsetFeatureOverlaps[-c(1,4),] #Duplicate but without gene and total flank information
for (i in 1:length(McapUpsetFeatureOverlapsPercents)) {
  McapUpsetFeatureOverlapsPercents[,i] <- (McapUpsetFeatureOverlapsPercents[,i] / sum(McapUpsetFeatureOverlapsPercents[,i])) * 100
} #Calculate percentages
head(McapUpsetFeatureOverlapsPercents) #Confirm calculations
```

```{r}
write.table(McapUpsetFeatureOverlapsPercents, "../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-Feature-Overlap-Percents.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

### Create plot

```{r}
#pdf("../Output/Mcap-Upset-Feature-Overlaps.pdf", height = 8.5, width = 11)
par(mar = c(3,5,0,1), oma = c(0, 1, 1, 11)) #Change figure boundaries

barplot(t(t(McapUpsetFeatureOverlapsPercents)), 
        col= c("grey20", "grey40", "grey60", "grey80", "grey100"), 
        axes = FALSE, 
        names.arg = c("G", "WRM", "WR", "WM", "RM", "W", "R", "M", "None"), cex.names = 1.2,
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% CpGs", line = 3, cex = 1.5) #Add y-axis label

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE) #Create new plot
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n") #Add new plot on top of current plot
legend(x = 0.57, y = 0.835, 
       xpd = TRUE,
       legend = c("CDS", "Introns", "Upstream Flank", "Downstream Flank", "Intergenic"),
       pch = 22, 
        col = "black", 
        pt.bg = c("grey20", "grey40", "grey60", "grey80", "grey100"),
       bty = "n",
       cex = 1.5, 
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box
text("Genome Feature", x = 0.785, y = 0.835, cex = 1.5) #Add legend title that is aligned with legend

#dev.off() #Turn off plotting device
```

## Pact

### Import file counts

```{r}
PactUpsetGeneOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-paGenes-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-gene overlaps
PactUpsetGeneOverlaps <- PactUpsetGeneOverlaps[-9,] #Remove final row
tail(PactUpsetGeneOverlaps) #Confirm import
```

```{r}
PactUpsetCDSOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-paCDS-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-CDS overlaps
PactUpsetCDSOverlaps <- PactUpsetCDSOverlaps[-9,] #Remove final row
tail(PactUpsetCDSOverlaps) #Confirm import
```

```{r}
PactUpsetIntronsOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-paIntron-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
PactUpsetIntronsOverlaps <- PactUpsetIntronsOverlaps[-9,] #Remove final row
tail(PactUpsetIntronsOverlaps) #Confirm import
```

```{r}
PactUpsetFlanksOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-paFlanks-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-flank overlaps
PactUpsetFlanksOverlaps <- PactUpsetFlanksOverlaps[-9,] #Remove final row
tail(PactUpsetFlanksOverlaps) #Confirm import
```

```{r}
PactUpsetUpstreamOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-paFlanksUpstream-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-upstream flank overlaps
PactUpsetUpstreamOverlaps <- PactUpsetUpstreamOverlaps[-9,] #Remove final row
tail(PactUpsetUpstreamOverlaps) #Confirm import
```

```{r}
PactUpsetDownstreamOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-paFlanksDownstream-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-downstream flank overlaps
PactUpsetDownstreamOverlaps <- PactUpsetDownstreamOverlaps[-9,] #Remove final row
tail(PactUpsetDownstreamOverlaps) #Confirm import
```

```{r}
PactUpsetIntergenicOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-paIntergenic-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Intergenic overlaps
PactUpsetIntergenicOverlaps <- PactUpsetIntergenicOverlaps[-9,] #Remove final row
tail(PactUpsetIntergenicOverlaps) #Confirm import
```

### Create summary tables

```{r}
PactUpsetFeatureOverlaps <- cbind(PactUpsetGeneOverlaps,
                                  PactUpsetCDSOverlaps,
                                  PactUpsetIntronsOverlaps,
                                  PactUpsetFlanksOverlaps,
                                  PactUpsetUpstreamOverlaps,
                                  PactUpsetDownstreamOverlaps,
                                  PactUpsetIntergenicOverlaps) #Use cbind to combine individual dataframes
PactUpsetFeatureOverlaps <- PactUpsetFeatureOverlaps[,c(seq(1,14,2))] #Only keep count columns
colnames(PactUpsetFeatureOverlaps) <- c("Genes", "CDS", "Introns", "Flanks", "Upstream Flanks", "Downstream Flanks", "Intergenic") #Add column names
row.names(PactUpsetFeatureOverlaps) <- c("MBD",
                                         "RRBS.MBD",
                                         "RRBS",
                                         "WGBS.MBD",
                                         "WGBS.RRBS.MBD",
                                         "WGBS.RRBS",
                                         "WGBS",
                                         "None") #Add rownames based on original files
PactUpsetFeatureOverlaps <- data.frame(t(PactUpsetFeatureOverlaps)) #Transpose dataframe
PactUpsetFeatureOverlaps$allCpGs <- PactGenomeFeatures[c(5,1,7,3,4,2,6),1] #Add genome information
PactUpsetFeatureOverlaps <- PactUpsetFeatureOverlaps[,c(9,5,6,4,2,7,3,1,8)] #Reorganize columns
tail(PactUpsetFeatureOverlaps) #Confirm changes
```

```{r}
write.table(PactUpsetFeatureOverlaps, "../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-Feature-Overlap-Counts.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

```{r}
PactUpsetFeatureOverlapsPercents <- PactUpsetFeatureOverlaps[-c(1,4),] #Duplicate but without gene and total flank information
for (i in 1:length(PactUpsetFeatureOverlapsPercents)) {
  PactUpsetFeatureOverlapsPercents[,i] <- (PactUpsetFeatureOverlapsPercents[,i] / sum(PactUpsetFeatureOverlapsPercents[,i])) * 100
} #Calculate percentages
head(PactUpsetFeatureOverlapsPercents) #Confirm calculations
```

```{r}
write.table(PactUpsetFeatureOverlapsPercents, "../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-Feature-Overlap-Percents.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

### Create plot

```{r}
#pdf("../Output/Pact-Upset-Feature-Overlaps.pdf", height = 8.5, width = 11)
par(mar = c(3,5,0,1), oma = c(0, 1, 1, 11)) #Change figure boundaries

barplot(t(t(PactUpsetFeatureOverlapsPercents)), 
        col= c("grey20", "grey40", "grey60", "grey80", "grey100"), 
        axes = FALSE, 
        names.arg = c("G", "WRM", "WR", "WM", "RM", "W", "R", "M", "None"), cex.names = 1.2,
        ylim = c(0, 110)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "% CpGs", line = 3, cex = 1.5) #Add y-axis label

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE) #Create new plot
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n") #Add new plot on top of current plot
legend(x = 0.57, y = 0.835, 
       xpd = TRUE,
       legend = c("CDS", "Introns", "Upstream Flank", "Downstream Flank", "Intergenic"),
       pch = 22, 
        col = "black", 
        pt.bg = c("grey20", "grey40", "grey60", "grey80", "grey100"),
       bty = "n",
       cex = 1.5, 
       x.intersp = 0.7, xjust = 0) #Place a legend in the top right of the figure with no box
text("Genome Feature", x = 0.785, y = 0.835, cex = 1.5) #Add legend title that is aligned with legend

#dev.off() #Turn off plotting device
```
