---
title: "Identifying Genome Features"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this script, I'll create summary tables and figures for genome feature identification of various CpG categories. I'll also run statistical analyses to determine if distributions differ between these datasets.

# Install packages

```{r}
#install.packages("RColorBrewer")
require(RColorBrewer) #Load RColorBrewer

#install.packages("dichromat")
require(dichromat)
```

# Session information

```{r}
sessionInfo()
```

# DMC data

## Mcap

## Pact

### Import line counts

```{r}
PactGenomeFeatures <- read.table("../analyses/Characterizing-CpG-Methylation-5x/Pact/Pact-CGMotif-Overlaps-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with CG motif and feature track overlaps
PactGenomeFeatures <- PactGenomeFeatures[-8,] #Remove final row
tail(PactGenomeFeatures) #Check import
```

```{r}
PactDMCGeneOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paGenes-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-gene overlaps
PactDMCGeneOverlaps <- PactDMCGeneOverlaps[-3,] #Remove final row
tail(PactDMCGeneOverlaps) #Confirm import
```

```{r}
PactDMCCDSOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paCDS-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-CDS overlaps
PactDMCCDSOverlaps <- PactDMCCDSOverlaps[-3,] #Remove final row
tail(PactDMCCDSOverlaps) #Confirm import
```

```{r}
PactDMCIntronsOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paIntron-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
PactDMCIntronsOverlaps <- PactDMCIntronsOverlaps[-3,] #Remove final row
tail(PactDMCIntronsOverlaps) #Confirm import
```

```{r}
PactDMCFlanksOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paFlanks-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-flank overlaps
PactDMCFlanksOverlaps <- PactDMCFlanksOverlaps[-3,] #Remove final row
tail(PactDMCFlanksOverlaps) #Confirm import
```

```{r}
PactDMCUpstreamOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paFlanksUpstream-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-upstream flanks overlaps
PactDMCUpstreamOverlaps <- PactDMCUpstreamOverlaps[-3,] #Remove final row
tail(PactDMCUpstreamOverlaps) #Confirm import
```

```{r}
PactDMCDownstreamOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paFlanksDownstream-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-downstream flanks overlaps
PactDMCDownstreamOverlaps <- PactDMCDownstreamOverlaps[-3,] #Remove final row
tail(PactDMCDownstreamOverlaps) #Confirm import
```

```{r}
PactDMCIntergenicOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-DMC-data-paIntergenic-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Intergenic overlaps
PactDMCIntergenicOverlaps <- PactDMCIntergenicOverlaps[-3,] #Remove final row
tail(PactDMCIntergenicOverlaps) #Confirm import
```

### Create summary tables

```{r}
PactDMCFeatureOverlaps <- cbind(PactDMCGeneOverlaps,
                                PactDMCCDSOverlaps,
                                PactDMCIntronsOverlaps,
                                PactDMCFlanksOverlaps,
                                PactDMCUpstreamOverlaps,
                                PactDMCDownstreamOverlaps,
                                PactDMCIntergenicOverlaps) #Use cbind to combine individual dataframes
PactDMCFeatureOverlaps <- PactDMCFeatureOverlaps[,c(seq(1,14,2))] #Only keep count columns
PactDMCFeatureOverlaps <- cbind(PactGenomeFeatures[,1], data.frame(t(PactDMCFeatureOverlaps))) #Transpose dataframe and add genome information
row.names(PactDMCFeatureOverlaps) <- c("Genes", "CDS", "Introns", "Flanks", "Upstream Flanks", "Downstream Flanks", "Intergenic") #Add row names
colnames(PactDMCFeatureOverlaps) <- c("allCpGs", "RvM", "WvM") #Add column names based on original files
head(PactDMCFeatureOverlaps) #Confirm changes
```

```{r}
write.table(PactDMCFeatureOverlaps, "../analyses/Identifying-genomic-locations/Pact/Pact-DMC-Feature-Overlap-counts.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

```{r}
PactDMCFeatureOverlapsPercents <- PactDMCFeatureOverlaps[-c(1,4),] #Remove gene overlap information and save as a new dataframe
for (i in 1:length(PactDMCFeatureOverlapsPercents)) {
  PactDMCFeatureOverlapsPercents[,i] <- (PactDMCFeatureOverlapsPercents[,i] / sum(PactDMCFeatureOverlapsPercents[,i])) * 100
} #Calculate percentages
head(PactDMCFeatureOverlapsPercents) #Check calculations
```

```{r}
write.table(PactDMCFeatureOverlapsPercents, "../analyses/Identifying-genomic-locations/Pact/Pact-DMC-Feature-Overlap-percents.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save summary table
```

### Conduct chi-squared tests of homogeneity

The null hypothesis is that distributions of method-associated DMC are the same as the distribution of CpGs in the genome

```{r}
PactDMCFeatureStatTest <- data.frame(t(PactDMCFeatureOverlaps[-c(1,4),])) #Transpose dataframe for statistical testing and remove gene and total flank information
head(PactDMCFeatureStatTest) #Confirm changes
```

#### All vs. RvM

```{r}
PactFeatureAllRM <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactDMCFeatureStatTest)) { #For each feature
  AllFeature <- PactDMCFeatureStatTest[1,i] #Variable for # feature overlap with background CpGs
  MethodFeature <- PactDMCFeatureStatTest[2,i] #Variable for # feature overlap with method CpGs
  AllNotFeature <- sum(PactDMCFeatureStatTest[1,-i]) #Variable for # other feature overlap with background CpGs
  MethodNotFeature <- sum(PactDMCFeatureStatTest[2,-i]) #Variable for # other feature overlap with method CpGs
  ct <- matrix(c(AllFeature,MethodFeature,AllNotFeature,MethodNotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactDMCFeatureStatTest[i])), paste0("Not", colnames(PactDMCFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(PactDMCFeatureStatTest)[c(1,2)])) #Assign row names: all, DMC type
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(PactDMCFeatureStatTest)[i]) #Add feature to results
  PactFeatureAllRM <- rbind(PactFeatureAllRM, ctResults) #Add test statistics to master table
}
```

```{r}
PactFeatureAllRM$p.adj <- p.adjust(PactFeatureAllRM$p.value, method = "fdr") #Correct p-value using FDR
PactFeatureAllRM$comparison <- rep("All vs. RvM", times = 5) #Add categories compared
head(PactFeatureAllRM) #Confirm changes
```

#### All vs. WvM

```{r}
PactFeatureAllWM <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactDMCFeatureStatTest)) { #For each feature
  AllFeature <- PactDMCFeatureStatTest[1,i] #Variable for # feature overlap with background CpGs
  MethodFeature <- PactDMCFeatureStatTest[3,i] #Variable for # feature overlap with method CpGs
  AllNotFeature <- sum(PactDMCFeatureStatTest[1,-i]) #Variable for # other feature overlap with background CpGs
  MethodNotFeature <- sum(PactDMCFeatureStatTest[3,-i]) #Variable for # other feature overlap with method CpGs
  ct <- matrix(c(AllFeature,MethodFeature,AllNotFeature,MethodNotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactDMCFeatureStatTest[i])), paste0("Not", colnames(PactDMCFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(PactDMCFeatureStatTest)[c(1,3)])) #Assign row names: all, DMC type
  print(ct) #ConfiWM table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(PactDMCFeatureStatTest)[i]) #Add feature to results
  PactFeatureAllWM <- rbind(PactFeatureAllWM, ctResults) #Add test statistics to master table
}
```

```{r}
PactFeatureAllWM$p.adj <- p.adjust(PactFeatureAllWM$p.value, method = "fdr") #Correct p-value using FDR
PactFeatureAllWM$comparison <- rep("All vs. WvM", times = 5) #Add categories compared
head(PactFeatureAllWM) #ConfiWM changes
```

#### RvM vs. WvM

```{r}
PactFeatureRMWM <- data.frame() #Create empty dataframe to store chi-squared results
for(i in 1:ncol(PactDMCFeatureStatTest)) { #For each feature
  Method1Feature <- PactDMCFeatureStatTest[2,i] #Variable for # feature for method 1
  Method2Feature <- PactDMCFeatureStatTest[3,i] #Variable for # feature for method 2
  Method1NotFeature <- sum(PactDMCFeatureStatTest[2,-i]) #Variable for # other feature for method 1
  Method2NotFeature <- sum(PactDMCFeatureStatTest[3,-i]) #Variable for # other feature for method 2
  ct <- matrix(c(Method1Feature,Method2Feature,Method1NotFeature,Method2NotFeature), ncol = 2) #Create contingency table
  colnames(ct) <- c(as.character(colnames(PactDMCFeatureStatTest[i])), paste0("Not", colnames(PactDMCFeatureStatTest[i]))) #Assign column names: feature, not feature
  rownames(ct) <- c(as.character(row.names(PactDMCFeatureStatTest)[c(2,3)])) #Assign row names: method 1, method 2
  print(ct) #Confirm table is correct
  ctResults <- data.frame(broom::tidy(chisq.test(ct))) #Create dataframe storing chi-sq stats results. Use broom::tidy to extract results from test output
  ctResults$FeatureType <- as.character(colnames(PactDMCFeatureStatTest)[i]) #Add feature to results
  PactFeatureRMWM <- rbind(PactFeatureRMWM, ctResults) #Add test statistics to master table
}
```

```{r}
PactFeatureRMWM$p.adj <- p.adjust(PactFeatureRMWM$p.value, method = "fdr") #Correct p-value using FDR
PactFeatureRMWM$comparison <- rep("RvM vs. WvM", times = 5) #Add methods compared
head(PactFeatureRMWM) #Confirm changes
```

#### Save statistical test results

```{r}
PactDMCFeatureStatResults <- rbind(PactFeatureAllRM,
                                PactFeatureAllWM,
                                PactFeatureRMWM) #Combine tables
tail(PactDMCFeatureStatResults) #Confirm changes
```

```{r}
write.table(PactDMCFeatureStatResults, "../analyses/Identifying-genomic-locations/Pact/Pact-DMC-Genomic-Location-StatResults.txt", quote = FALSE, row.names = FALSE) #Save table
```

# Upset plot data

## Mcap

### Import file counts

```{r}
McapUpsetGeneOverlaps <- read.table("../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-mcGenes-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-gene overlaps
McapUpsetGeneOverlaps <- McapUpsetGeneOverlaps[-9,] #Remove final row
tail(McapUpsetGeneOverlaps) #Confirm import
```

```{r}
McapUpsetCDSOverlaps <- read.table("../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-mcCDS-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-CDS overlaps
McapUpsetCDSOverlaps <- McapUpsetCDSOverlaps[-9,] #Remove final row
tail(McapUpsetCDSOverlaps) #Confirm import
```

```{r}
McapUpsetIntronsOverlaps <- read.table("../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-mcIntrons-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
McapUpsetIntronsOverlaps <- McapUpsetIntronsOverlaps[-9,] #Remove final row
tail(McapUpsetIntronsOverlaps) #Confirm import
```

```{r}
McapUpsetFlanksOverlaps <- read.table("../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-mcFlanks-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
McapUpsetFlanksOverlaps <- McapUpsetFlanksOverlaps[-9,] #Remove final row
tail(McapUpsetFlanksOverlaps) #Confirm import
```

```{r}
McapUpsetIntergenicOverlaps <- read.table("../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-mcIntergenic-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Intergenic overlaps
McapUpsetIntergenicOverlaps <- McapUpsetIntergenicOverlaps[-9,] #Remove final row
tail(McapUpsetIntergenicOverlaps) #Confirm import
```

### Create summary tables

```{r}
McapUpsetFeatureOverlaps <- cbind(McapUpsetGeneOverlaps,
                                McapUpsetCDSOverlaps,
                                McapUpsetIntronsOverlaps,
                                McapUpsetFlanksOverlaps,
                                McapUpsetIntergenicOverlaps) #Use cbind to combine individual dataframes
McapUpsetFeatureOverlaps <- McapUpsetFeatureOverlaps[,c(seq(1,10,2))] #Only keep count columns
colnames(McapUpsetFeatureOverlaps) <- c("Genes", "CDS", "Introns", "Flanks", "Intergenic") #Add column names
row.names(McapUpsetFeatureOverlaps) <- c("MBD",
                                        "RRBS.MBD",
                                        "RRBS",
                                        "WGBS.MBD",
                                        "WGBS.RRBS.MBD",
                                        "WGBS.RRBS",
                                        "WGBS",
                                        "None") #Add rownames based on original files
McapUpsetFeatureOverlaps <- data.frame(t(McapUpsetFeatureOverlaps)) #Transpose dataframe
McapUpsetFeatureOverlaps <- McapUpsetFeatureOverlaps[,c(5,6,4,2,7,3,1,8)] #Reorganize columns
tail(McapUpsetFeatureOverlaps) #Confirm changes
```

```{r}
write.table(McapUpsetFeatureOverlaps, "../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-Feature-Overlap-Counts.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

```{r}
McapUpsetFeatureOverlapsPercents <- McapUpsetFeatureOverlaps[-1,] #Duplicate but without gene information
for (i in 1:length(McapUpsetFeatureOverlapsPercents)) {
  McapUpsetFeatureOverlapsPercents[,i] <- (McapUpsetFeatureOverlapsPercents[,i] / sum(McapUpsetFeatureOverlapsPercents[,i])) * 100
} #Calculate percentages
head(McapUpsetFeatureOverlapsPercents) #Confirm calculations
```

```{r}
write.table(McapUpsetFeatureOverlapsPercents, "../analyses/Identifying-genomic-locations/Mcap/Mcap-upset-data-Feature-Overlap-Percents.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

### Create plot

```{r}
#pdf("../Output/Mcap-Upset-Feature-Overlaps.pdf", height = 8.5, width = 11)
par(mar = c(3,5,0,1), oma = c(2, 1, 1, 1)) #Change figure boundaries

barplot(t(t(McapUpsetFeatureOverlapsPercents)), 
        col= c("grey20", "grey40", "grey60", "grey80"), 
        axes = FALSE, 
        names.arg = c("WRM", "WR", "WM", "RM", "W", "R", "M", "None"), cex.names = 1.2,
        ylim = c(0, 115)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "Proportion CpGs", line = 3, cex = 1.5) #Add y-axis label
mtext(side = 1, "Sequencing Method", line = 3, cex = 1.5) #Add x-axis label

#dev.off() #Turn off plotting device
```

## Pact

### Import file counts

```{r}
PactUpsetGeneOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-paGenes-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-gene overlaps
PactUpsetGeneOverlaps <- PactUpsetGeneOverlaps[-9,] #Remove final row
tail(PactUpsetGeneOverlaps) #Confirm import
```

```{r}
PactUpsetCDSOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-paCDS-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-CDS overlaps
PactUpsetCDSOverlaps <- PactUpsetCDSOverlaps[-9,] #Remove final row
tail(PactUpsetCDSOverlaps) #Confirm import
```

```{r}
PactUpsetIntronsOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-paIntron-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
PactUpsetIntronsOverlaps <- PactUpsetIntronsOverlaps[-9,] #Remove final row
tail(PactUpsetIntronsOverlaps) #Confirm import
```

```{r}
PactUpsetFlanksOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-paFlanks-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Introns overlaps
PactUpsetFlanksOverlaps <- PactUpsetFlanksOverlaps[-9,] #Remove final row
tail(PactUpsetFlanksOverlaps) #Confirm import
```

```{r}
PactUpsetIntergenicOverlaps <- read.table("../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-paIntergenic-counts.txt", header = FALSE, col.names = c("counts", "filename")) #Import file with all file-Intergenic overlaps
PactUpsetIntergenicOverlaps <- PactUpsetIntergenicOverlaps[-9,] #Remove final row
tail(PactUpsetIntergenicOverlaps) #Confirm import
```

### Create summary tables

```{r}
PactUpsetFeatureOverlaps <- cbind(PactUpsetGeneOverlaps,
                                PactUpsetCDSOverlaps,
                                PactUpsetIntronsOverlaps,
                                PactUpsetFlanksOverlaps,
                                PactUpsetIntergenicOverlaps) #Use cbind to combine individual dataframes
PactUpsetFeatureOverlaps <- PactUpsetFeatureOverlaps[,c(seq(1,10,2))] #Only keep count columns
colnames(PactUpsetFeatureOverlaps) <- c("Genes", "CDS", "Introns", "Flanks", "Intergenic") #Add column names
row.names(PactUpsetFeatureOverlaps) <- c("MBD",
                                        "RRBS.MBD",
                                        "RRBS",
                                        "WGBS.MBD",
                                        "WGBS.RRBS.MBD",
                                        "WGBS.RRBS",
                                        "WGBS",
                                        "None") #Add rownames based on original files
PactUpsetFeatureOverlaps <- data.frame(t(PactUpsetFeatureOverlaps)) #Transpose dataframe
PactUpsetFeatureOverlaps <- PactUpsetFeatureOverlaps[,c(5,6,4,2,7,3,1,8)] #Reorganize columns
tail(PactUpsetFeatureOverlaps) #Confirm changes
```

```{r}
write.table(PactUpsetFeatureOverlaps, "../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-Feature-Overlap-Counts.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

```{r}
PactUpsetFeatureOverlapsPercents <- PactUpsetFeatureOverlaps[-1,] #Duplicate but without gene information
for (i in 1:length(PactUpsetFeatureOverlapsPercents)) {
  PactUpsetFeatureOverlapsPercents[,i] <- (PactUpsetFeatureOverlapsPercents[,i] / sum(PactUpsetFeatureOverlapsPercents[,i])) * 100
} #Calculate percentages
head(PactUpsetFeatureOverlapsPercents) #Confirm calculations
```

```{r}
write.table(PactUpsetFeatureOverlapsPercents, "../analyses/Identifying-genomic-locations/Pact/Pact-upset-data-Feature-Overlap-Percents.txt", sep = "\t", quote = FALSE, row.names = TRUE) #Save file
```

### Create plot

```{r}
#pdf("../Output/Pact-Upset-Feature-Overlaps.pdf", height = 8.5, width = 11)
par(mar = c(3,5,0,1), oma = c(2, 1, 1, 1)) #Change figure boundaries

barplot(t(t(PactUpsetFeatureOverlapsPercents)), 
        col= c("grey20", "grey40", "grey60", "grey80"), 
        axes = FALSE, 
        names.arg = c("WRM", "WR", "WM", "RM", "W", "R", "M", "None"), cex.names = 1.2,
        ylim = c(0, 115)) #Create base plot. Everything should be white. Do not plot axes. Include sequencing type as labels and set size. Set y-axis specs.

axis(side = 2, at = seq(0, 100, by = 10), las = 2, col = "grey80", cex.axis = 1.2) #Add y-axis
mtext(side = 2, "Proportion CpGs", line = 3, cex = 1.5) #Add y-axis label
mtext(side = 1, "Sequencing Method", line = 3, cex = 1.5) #Add x-axis label

#dev.off() #Turn off plotting device
```
