---
title: "Median Methylation"
author: "Yaamini Venkataraman"
date: "7/9/2020"
output: html_document
---

In this script I will calculate median methylation of genes with 5x coverage in a sample.

# Set up R Markdown Document

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Session information

```{r}
sessionInfo()
```

# *M. capitata*

```{bash}
mkdir ../analyses/Characterizing-CpG-Methylation-5x/Mcap/Median-Methylation-Calculations #Create directory for output
```

## Associate CpG loci with genes

The overlap files generated in [this Jupyter notebook](https://github.com/hputnam/Meth_Compare/blob/master/scripts/Characterizing-CpG-Methylation-5x.ipynb) are missing gene ID information. I can get that from the previously-generated gene track previously generated using `intersectBed`.

```{bash}
head ../genome-feature-files/Mcap.GFFannotation.gene.gff #Look at format of Mcap gene track. Gene ID is the last column
```

```{bash}
#For each sample-gene overlap file (ends in *bedgraph.bed-mcGenes)
#Use intersectBed to find where loci and genes intersect, allowing loci to be mapped to genes
#wb: Print all lines in the second file
#a: sample-gene overlap file
#b: gene track
#Save output in a new file that has the same base name and ends in -geneID

for f in ../analyses/Characterizing-CpG-Methylation-5x/Mcap/*bedgraph.bed-mcGenes
do
  /usr/local/bin/intersectBed \
  -wb \
  -a ${f} \
  -b ../genome-feature-files/Mcap.GFFannotation.gene.gff \
  > ${f}-geneID
done
```

```{bash}
find ../analyses/Characterizing-CpG-Methylation-5x/Mcap/*geneID | wc -l #Confirm that 9 files were created (one per sample)
```

```{bash}
mv ../analyses/Characterizing-CpG-Methylation-5x/Mcap/*geneID ../analyses/Characterizing-CpG-Methylation-5x/Mcap/Median-Methylation-Calculations #Move file to subdirectory for median methylation calculations
```

```{bash}
head ../analyses/Characterizing-CpG-Methylation-5x/Mcap/Median-Methylation-Calculations/Meth10_R1_001_val_1_bismark_bt2_pe._5x.bedgraph.bed-mcGenes-geneID #Confirm files are in the new subdirectory and look at output
```

## Import sample overlaps with gene ID information

```{r}
setwd("../analyses/Characterizing-CpG-Methylation-5x/Mcap/") #Set working directory within the notebook chunk for list.files to find the necessary files
filesToImport <- list.files(pattern = "*geneID") #Create a file list for all 9 files to import. Only import overlaps for full samples (not divided by methylation status)
list2env(lapply(setNames(filesToImport,
                         make.names(gsub("_R1_001_val_1_bismark_bt2_pe._5x.bedgraph.bed", "", filesToImport))),
                read.delim),
         envir = .GlobalEnv) #Import files to the .GlobalEnv with list2env. Use lapply to setNames of the files by taking all the common parts of their names out. Read files with read.delim (can only use defaults = header = TRUE, sep = "\t"). Files will be named Meth#-mcGenes
head(Meth10.mcGenes) #Confirm import
```

## Format dataframes

```{r}
colnames(Meth10.mcGenes) <- c("chr", "start", "stop", "geneID", "percentMeth") #Modify column names
head(Meth10.mcGenes) #Confirm column naming
```

## Calculate median methylation by `geneID`

```{r}
percentMeth10-mcGene <- aggregate(percentMeth ~ geneID, data = Meth10.mcGenes, FUN = median) #Use aggregate to group geneID and calculate median percent methylation
head(percentMeth10-mcGene) #Confirm percent methylation calculation
```





