# figures

- [**Figure 1**](https://github.com/hputnam/Meth_Compare/blob/master/output/figures/Fig_1.jpg): Experimental design. Three biological replicate coral samples were obtained from both *M. capitata* and *P. acuta*. DNA was extracted from each coral sample and split for use in Whole Genome Bisulfite Sequencing (WGBS), Reduced Representation Bisulfite Sequencing (RRBS), and Methyl-CpG Binding Domain Bisulfite Sequencing (MBDBS) library preparation methods. Three libraries were generated for each of the three methods, yielding nine libraries for each species and 18 libraries total.
- [**Figure 2**](https://github.com/hputnam/Meth_Compare/blob/master/output/figures/Fig_2.jpg): Circos plots showing the mean percent methylation of CpGs with 5x cover for each of the three methods on the largest scaffolds of each genome. The outer track shows the scaffold locations and dots indicate the percent methylation as indicated by the y-axes from 0-100% for each of the inner tracks.
- [**Figure 3**](https://github.com/hputnam/Meth_Compare/blob/master/output/figures/Fig_3.jpg): Matrix of pairwise scatter plots for  shared CpG loci (i.e., CpG covered at > 5x across all samples) for (**A**) *M. capitata* (n = 4,666 common loci) and (**B**) *P. acuta* (n = 93,714 common loci). The red lines represent linear regression fits and the green lines are polynomial regression fits. Pearson correlation coefficients for each pairwise comparison are presented in the upper right boxes. Methods are color coded on the X and Y axes (WGBS = green, MBDBS = purple, and RRBS = orange) and replicate samples are indicated on the diagonal along with histograms of % CpG methylation.
- [**Figure 4**](https://github.com/hputnam/Meth_Compare/blob/master/output/figures/Fig_4.jpg): CpG site coverage across library preparation methods. Mean fraction of CpG sites in the genome covered at different sequencing depths (read depths) by (**A**) MBDBS libraries, (**B**) RRBS libraries, and (**C**) WGBS libraries with standard deviations shown by shaded areas (see [Supplementary Table 2](https://github.com/hputnam/Meth_Compare/tree/master/output/supplemental-material) for number of reads in each sample). 
- [**Figure 5**](https://github.com/hputnam/Meth_Compare/blob/master/output/figures/Fig_5.jpg): Proportion of CpGs detected by sequencing methods in coding sequences (CDS), introns, 1 Kb flanking regions upstream (Upstream Flank) or downstream of genes (Downstream Flank), and intergenic regions for (**A**) *M. capitata* and (**B**) *P. acuta*. Principal Coordinate Analyses associated with perMANOVA and beta-dispersion tests related to [Supplementary Table 6](https://github.com/hputnam/Meth_Compare/tree/master/output/supplemental-material) that show differences in proportion of CpGs in various genomic locations (CDS, introns, upstream flanks, downstream flanks, and intergenic regions) for (**C**) *M. capitata* and **D**) *P. acuta*. WGBS is represented by green circles, RRBS by purple triangles, and MBDBS by orange diamonds. Percent variation explained by each PCoA axis is included in the axis label. Ellipses depict 95% confidence intervals for each sequencing method. All eigenvectors are significant at the Î± = 0.05 level.
